<h1>Differential Expression Atlas help</h1>

<!-- Beginning of main text section -->
<section class="grid_18 alpha">

<h2><a name="about">About the Differential Atlas</a></h2>

<p>The Differential Atlas lets you ask questions about which genes are
up&ndash; or down&ndash;regulated in different experimental conditions.  It is
based on a set of highly curated differential expression RNA-Seq and microarray
experiments from <a
	href="http://www.ebi.ac.uk/arrayexpress">ArrayExpress</a> that have been
re-processed using our in&ndash;house differential expression statistical
analysis pipeline.</p>



<h2><a name="at-a-glance">Differential Atlas at&ndash;a&ndash;glance</a></h2>

<p><a href="resources/images/help/differential-atlas-overview.pdf"><img src="resources/images/help/diffExptPageHelp.png" alt="Differential Atlas experiment page screenshot" align="bottom" /></a></p>


<h2><a name="searching">Searching the Differential Atlas</a></h2>


<h3><a name="minPvalLogFC">Specifying the maximum adjusted <i>p</i>-value and minimum log<sub>2</sub> fold-change</a></h3>

<p>By default, the maximum adjusted <i>p</i>-value is 0.05, and the minimum 
log<sub>2</sub> fold-change is 1. This means only genes with an adjusted
<i>p</i>-value below 0.05 and log<sub>2</sub> fold-change above 1 (or below -1)
are displayed.</p>

<p>You may change the two cutoff values to whatever you like, to
relax constraints or make them stricter. Changing these values will affect the
number of genes displayed. Generally, the higher the chosen adjusted
<i>p</i>-value, and the lower the minimum log<sub>2</sub> fold-change, the more
genes will be displayed.</p>

<p>Type any value from 0 to 1 in the <strong>Adjusted <i>p</i>-value cutoff</strong>
box to use as the maximum false discovery rate (FDR) -adjusted <i>p</i>-value
cutoff. Only genes that show differential expression with an adjusted
<i>p</i>-value at or below this cutoff will be returned.</p>

<p>Type any numerical value in the <strong>Log<sub>2</sub> fold-change
cutoff</strong> box to use as the minimum absolute log<sub>2</sub> fold-change.
Only genes that have a larger absolute log<sub>2</sub> fold-change will be
displayed. When we say "absolute" value, we mean the value without a "+" or
"-" sign in front. For example, if you choose an absolute value of 2, then
genes with log<sub>2</sub> fold-change of greater than +2 <strong>or</strong>
less than -2 are shown.</p>


<h3><a name="geneSearch">Searching with genes</a></h3>

<p>You can search with <a href="http://ensembl.org">Ensembl</a> gene symbols
(e.g. desat1), identifiers (e.g. FBgn0086687) or <a title="Ensembl Glossary"
	class="externallink"
	href="http://www.ensembl.org/Help/Glossary">biotypes</a> (e.g.
protein_coding), <a class="externallink"
	href="http://www.uniprot.org">UniProt</a> accessions (e.g. P35542), <a
	class="externallink" href="http://www.geneontology.org">GO</a> ("pheromone
biosynthetic process") or <a href="http://www.ebi.ac.uk/interpro/">InterPro</a>
terms (e.g. "Fatty acid desaturase, type 1"). A <b>space&ndash;separated</b>
list of gene attributes will bring back genes that match <b>at least one</b> of
the attributes in the query.</p>

<p>Use the radio buttons to the right of the Gene Query box to limit the search
to up&ndash;regulated genes, down&ndash;regulated genes, or both.</p>

<p>If the "Exact match" box is checked, only genes with annotations that fully
match your query will be returned. For example, to see if the
<a
	href="http://www.ensembl.org/Drosophila_melanogaster/Gene/Summary?g=FBgn0086687">desat1</a>
gene is differentially expressed, check the "Exact match" box, type "desat1" and click "Search".  If the
box is not checked, genes with annotations that contain words that fully match
your query will be returned. For example, to find results for genes
with "desaturase" in their annotations (e.g. "stearoyl-CoA 9-desaturase
activity", "Fatty acid/sphingolipid desaturase", &hellip;), uncheck the "Exact
match" box, type "desaturase" and click search.</p>



<h3><a name="comparisonSearch">Searching with comparisons</a></h3>

<p>Many experiments contain more than one comparison. If you do
not select any comparisons, results for all comparisons in the experiment will be
displayed. To see the results for specific comparison(s) select the ones you wish
to see from the Comparison box:</p>

<p><img src="resources/images/help/selectContrast.png" alt="Comparison selection dropdown" align="bottom" /></p>

<p>Select one or more comparisons to search with, or start typing to see
suggestions. To see details of the groups of samples used in each comparison,
click the <img alt="Experiment Design button"
src="resources/images/help/expDes_btn.png" align="bottom" /> button.</p>



<h4><i>Specific</i> vs. <i>non-specific</i> search</h4>

<p>If there is more than one comparison in an experiment, the default
Differential Atlas search reports genes with more "specific" differential
expression (with the <i>Specific</i> option selected by default). What we mean
by this is outlined below.<p>



<h5> If no comparisons are selected&hellip;</h5>

<p>If <b>no comparisons</b> are selected, the <i>Specific</i> search will rank
genes so that genes that are differentially expressed in just <b>one</b>
    comparison come first, followed by genes differentially expressed in <b>two</b>
    comparisons, then three and so on, reporting genes that are differentially
expressed in all available comparisons at the end of the list of results. Within
each group of N comparisons, genes are ordered by largest average absolute (i.e.
ignoring positive or negative status) log<sub>2</sub> fold-change.</p>

<p>Only log<sub>2</sub> fold-changes for differential expression in the
direction selected are considered for specificity. For example, if you search
for only up&ndash;regulated genes (by clicking the "up" radio button next to
the Gene Query box), only genes with positive log<sub>2</sub> fold-changes are
used in the calculations for specificity.</p>



<h5>If at least one comparison is selected&hellip;</h5>

<p>If at least one comparison is selected, the <i>Specific</i> search will
promote genes with <b>larger average absolute log<sub>2</sub> fold-change</b> in the
<b>selected</b> comparison(s), and at the same time penalize genes with the same
or larger log<sub>2</sub> fold-change in the <b>un&ndash;selected</b> comparison(s).
This is done using the "fold difference" between the selected and
un&ndash;selected absolute log<sub>2</sub> fold-changes:</p>

<ul>
	<li>Fold difference is calculated by dividing the <b>average
		absolute log<sub>2</sub> fold-change</b> of the selected comparison(s) by the
	largest absolute log<sub>2</sub> fold-change of the un&ndash;selected
        comparisons.</li>
	
	<li>If <b>none</b> of the un&ndash;selected comparisons' absolute log<sub>2</sub>
	fold-changes are at or above the selected log<sub>2</sub> fold-change
	cutoff, the average of the selected comparisons' absolute log<sub>2</sub> fold-changes
	is divided by the cutoff instead.</li>
</ul>

<p>Genes with the <b>greatest fold difference</b> in absolute log<sub>2</sub> fold-changes
between selected and un&ndash;selected comparisons will be pushed towards the top
of the list. Genes that have a larger absolute log<sub>2</sub> fold-change in the
un&ndash;selected comparisons are not shown.</p>



<h5>Non&ndash;specific search</h5>

<p>If the <i>Specific</i> option is not selected, the Differential Atlas search
will <b>ignore</b> specificity of differential expression when ordering the
results. In particular:</p>

<ul>
	
	<li>If no comparisons are selected, genes with largest absolute
	log<sub>2</sub> fold-changes in <b>all</b> comparisons will be reported first.</li>

	<li>If at least one comparison is selected, genes with larger
	absolute log<sub>2</sub> fold-changes across the <b>selected</b> comparisons are reported first.
	Log<sub>2</sub> fold-changes in the un&ndash;selected comparisons are ignored in
	the ordering of results.</li>

</ul>



<h2><a name="heatmap">Heatmap results display</a></h2>

<p>Results for your query are shown in a heatmap table. The heatmap rows are
labelled with gene symbols (and design elements in the case of microarray
data). The columns are labelled with comparison descriptions. Mouseover the
column headings to see the full description.</p>

<p>The heatmap ranks genes by absolute (i.e. ignoring up- or down-regulated
status) log<sub>2</sub> fold-change, with the largest at the top. If some genes
have identical log<sub>2</sub> fold-changes, these are then ordered by adjusted
<i>p</i>-value, with the gene with the lowest adjusted <i>p</i>-value first.
Greater colour intensity means larger absolute log<sub>2</sub> fold-change, as
illustrated by the gradient bars (e.g. <img alt="heatmap colour gradient"
src="resources/images/help/gradient.png" align="bottom" />) displayed above the
heatmap. Blue boxes indicate the gene is down&ndash;regulated, while red ones
mean up&ndash;regulated. Mouseover a filled box in the heatmap to see the
adjusted <i>p</i>-value and log<sub>2</sub> fold-change (and <i>t</i>-statistic
for microarray data). To see the log<sub>2</sub> fold-changes for all genes,
click the <img alt="Display log2 fold-changes"
src="resources/images/help/display_logFCs.png" align="bottom" /> button. This
also reveals the maximum and minimum log<sub>2</sub> fold-changes in the
experiment either side of the gradient bars. The gradient shows intensities
corresponding to up to the <b>top 50</b> differentially expressed genes
currently displayed (rather than all genes returned by the query).</p>

<p>If you mouseover a gene name in the heatmap, synonyms, <a
	href="http://www.geneontology.org/">Gene Ontology</a> terms, and <a
	href="http://www.ebi.ac.uk/interpro/">Interpro</a> terms are displayed:</p>

<p><img src="resources/images/help/mouseoverDiff.png" alt="Differential Atlas gene mouseover" align="bottom" /></p>

<p>Any terms that match your Gene Query term(s) are highlighted in yellow here.
Click the gene name to see more information about that gene, including other
Expression Atlas experiments in which it was found.</p>

<h3><a name="plots">MA plots and gene set overlap summaries</a></h3>

<p>Two types of plots can be visualised by clicking one of the <img
src="resources/images/help/histogram_btn.png" alt="histogram button"
align="bottom" /> buttons in the header of the heatmap table. This brings up a
menu like that shown below. Different numbers of plots may be available for
different experiments.</p>

<p><img src="resources/images/help/plots_menu.png" alt="plots menu"
align="bottom" /></p>

<p>An MA plot is generated for each comparison in every experiment. To see
it, select the <img
src="resources/images/help/plots_menu_MA.png"
alt="MA plot menu item" align="bottom" /> item from the menu. This plot displays
the
average expression level for each gene (normalized microarray intensity level or
RNA-Seq log<sub>2</sub> counts-per-million) on the <i>x</i>-axis against
log<sub>2</sub> fold-change on the <i>y</i>-axis. This plot gives an overview
of the relationship between expression level and fold-change, and can be used
to assess any systematic biases in the data. Genes that were called as
differentially expressed at FDR &lt; 0.05 are shown in red in the plot.</p>

<p>Gene set overlap analysis is performed using the <a
href="http://www.bioconductor.org/packages/release/bioc/html/piano.html">Piano</a>
package from <a href="http://www.bioconductor.org/">Bioconductor</a>. For each
    comparison, overlap between terms from <a
href="http://www.geneontology.org/">GO</a>, <a
href="http://www.ebi.ac.uk/interpro/">InterPro</a>, and <a
href="http://www.reactome.org/">Reactome</a> and the set of
differentially expressed genes is performed, using <a
href="http://en.wikipedia.org/wiki/Fisher%27s_exact_test">Fisher's exact
test</a> with multiple testing correction (FDR &lt; 0.1). Gene set overlap plots, are available
    only when statistically significant overlap of terms was detected. This means that for some experiments, the menu will not display plots for all three of the
aforementioned resources. To see a plot, select one of the following options from the menu:</p>

<ul>
	<li><img
src="resources/images/help/plots_menu_GO.png"
alt="GO plot menu item" align="bottom" /> to see <a href="http://www.geneontology.org/">GO</a> term overlap,</li>
	<li><img
src="resources/images/help/plots_menu_iPro.png"
alt="InterPro plot menu item" align="bottom" /> to see <a href="http://www.ebi.ac.uk/interpro">InterPro</a> term overlap,</li>
	<li><img
src="resources/images/help/plots_menu_react.png"
alt="Reactome plot menu item" align="bottom" /> to see <a href="http://www.reactome.org">Reactome</a> pathway term overlap.</li>
</ul>

<p>Doing so will bring up a plot like the one below, showing a maximum of top 10 overlapping terms (nodes) from a list sorted by the
    effect size (i.e. the number of observed divided by the number of expected genes annotated with a given term
    within the differentially expressed set of genes).
    The terms are linked by edges representing genes shared between them - the more more genes shared between the two terms, the thicker the edge.
    The size of each node represents the proportion of
differentially expressed genes annotated with each term. Please see the <a
href="http://www.bioconductor.org/packages/release/bioc/html/piano.html">Piano</a>
documentation for more details.</p>

<p><img src="resources/images/help/go_plot.png" alt="GO term overlap plot" align="bottom" /></p>

<h3><a name="ensembl">Visualising differential expression at Ensembl</a></h3>

<p>If you select a gene name and a comparison heading from the heatmap, the
Ensembl Genome Browser <strong>Open</strong> button, to the left of the heatmap
table, will become clickable.</p>

<p><img src="resources/images/help/ensembl_gene_contrast_select.png" alt="screenshots showing how to visualise data in Ensembl" align="bottom" /></p>

<p>Clicking the <strong>Open</strong> button will take you to the <a
href="http://www.ensembl.org/index.html">Ensembl</a> or <a
href="http://ensemblgenomes.org/">Ensembl Genomes</a> browser, which will
display the log<sub>2</sub> fold-change for the comparison you selected in the
context of the genomic location of the gene you selected. For more details
about how to use the genome browsers, please see the relevant documentation for
<a href="http://www.ensembl.org/info/index.html">Ensembl</a> or <a
	href="http://ensemblgenomes.org/info/faqs">Ensembl Genomes</a>.</p>

<h2><a name="downloadResults">Downloading query results</a></h2>

<p>The <b>top 50</b> differentially expressed genes resulting from your search
are displayed on the page. Click on the <img alt="Download button"
src="resources/images/help/download_button.png" align="bottom" /> button to
download the <b>full results</b> of your query in tab&ndash;delimited format
with no ordering.</p>



<h2><a name="downloadData">Downloading other experimental data</a></h2>

<p>You can download RNA-seq raw
counts (click the <img
src="resources/images/help/dlRawCounts_btn.png"
alt="raw counts download button" align="bottom" /> button), normalized microarray intensity data (click the <img
src="resources/images/help/dlNorm_btn.png"
alt="normalized data download button" align="bottom" /> button), and all statistical
analytics results for all comparisons in the experiment (click the <img
src="resources/images/help/dlAna_btn.png"
alt="analytics download button" align="bottom" /> button). All data is
provided as tab-delimited text files. You can also download data for each
experiment ready to load into <a href="http://www.r-project.org">R</a> by
clicking the <img src="resources/images/help/Rhelp_button.png" alt="R data
download button" align="bottom" /> on any experiment page. See our <a
href="help/r-data-objects.html">help page</a> about Expression Atlas data in R
for more details.</p>


<h2><a name="qc">QC reports</a></h2>

<p>Click the <img alt="array QC report button"
src="resources/images/help/qc_button.png" align="bottom" /> button to see the
results of quality assessment for the experiment data files.</p>

<p>For microarray experiments, this report is
generated by the <a
href="http://www.bioconductor.org/packages/release/bioc/html/arrayQualityMetrics.html">arrayQualityMetrics</a>
package from <a href="http://www.bioconductor.org">Bioconductor</a> in <a
href="http://www.r-project.org">R</a>. Please see the arrayQualityMetrics <a
href="http://www.bioconductor.org/packages/release/bioc/vignettes/arrayQualityMetrics/inst/doc/arrayQualityMetrics.pdf">documentation</a>
for more details about the methods used. Breifly, outlier arrays are detected
using distance measures, box plots, and MA plots. Any array that is found to be
an outlier by all three of these methods is excluded from further analysis.</p>

<p>For RNA-seq experiments, the QC report is generated by the <a
href="https://code.google.com/p/irap/">iRAP</a> pipeline. Please see the iRAP
<a href="http://biorxiv.org/content/early/2014/06/06/005991">documentation</a>
for more details on the methods used.</p>


<h2><a name="anaMethods">Analysis methods page</a></h2>

<p>The Analysis Methods page (e.g. for <a
	href="experiments/E-MTAB-1066/analysis-methods"><i>D.
		melanogaster</i> CDK8 and CycC mutants</a> or <a
	href="experiments/E-GEOD-38400/analysis-methods"><i>Arabidopsis</i>
	lncRNA-mediated transcriptional silencing mutants</a>) lists the data
analysis methods applied to the raw experimental data (<a
	href="http://nar.oxfordjournals.org/content/38/6/1767.full">FASTQ</a> files
or microarray intensity data), to obtain the differential expression statistics
shown in the Differential Atlas. You can access it by clicking the <img
alt="Analysis Methods button" src="resources/images/help/am_btn.png"
align="bottom" /> button.</p>


<h2><a name="exptDesign">Experiment design page</a></h2>

<p>The Experiment Design page (e.g. for <a
	href="experiments/E-MTAB-1066/experiment-design"><i>D.
		melanogaster</i> CDK8 and CycC mutants</a> is accessible via the <img
alt="Experiment Design" src="resources/images/help/expDes_btn.png"
align="bottom" /> button. Here you can see RNA-Seq processing run accessions
(from <a href="http://www.ebi.ac.uk/ena/">ENA</a>) or microarray assay
accessions, along with their corresponding biological sample characteristics
and experimental variables. If you select a comparison from the "Comparison"
dropdown menu, the runs/assays used in that comparison will be highlighted in the
table. Runs/assays not used in the selected comparison are not highlighted.
Runs/assays assigned to the "Reference" group are highlighted one colour, and
those assigned to the "Test" group are highlighted in another. The up&ndash; or
down&ndash;regulated expression calls shown in the heatmap are always from the
perspective of the "Test" group.</p>

<p><img src="resources/images/help/diffExptDesign.png" alt="Differential Atlas experiment design page" align="bottom" /></p>

<h3><a name="sortExptDesign">Sorting the experiment design</a></h3>

<p>You can sort the experiment design table by clicking on a column. Click the
column again to reverse the sort order. To then sort by another column while
retaining the ordering of the first column, hold down the <b>shift</b> key and
click another column. The first selected column forms the primary sort order
and the shift+click&ndash;ed columns form secondary, tertiary, etc. sort orders
in turn. If you shift+click a column to sort by it and then wish to undo this,
just shift+click it again until the column is unsorted (<img alt="unsorted
arrows" src="resources/images/help/unsortedArrows.png" align="center" />).</p>

<p>The experiment design table is also searchable &mdash; entering a keyword
automatically selects the subsection of the table that matches the keyword.
Click on the <img alt="Download button"
src="resources/images/help/download_button.png" align="bottom" /> button on the
right&ndash;hand side to download the full results of the current query (with
no ordering applied) in tab&ndash;delimited format.</p>


<h2><a name="feedbackSection">Providing feedback</a></h2>

<p>We value your feedback on what Expression Atlas is doing right, what does
not work or what could be achieved more intuitively. We would also be grateful
for any feedback on the analysis methods we have adopted &mdash; we are
passionate about not only about intuitive data presentation, the highest level
of experimental metadata curation, but also about the quality and biological
validity of the data presented in the Differential Atlas.</p>

<p>To send us your feedback, please fill in the form accessed by clicking on
the <i>Feedback</i> link in the top&ndash;right of the page:</p>

<p><img alt="Feedback Pop-up" src="resources/images/help/feedback.png" align="bottom" /></p>

</section>

<aside class="grid_6 omega">

<h3>Jump to...</h3>

<ul>
	<li><a href="help/differential-atlas.html#about">About the Differential Atlas</a></li>
	<li><a href="help/differential-atlas.html#at-a-glance">Differential Atlas at&ndash;a&ndash;glance</a></li>
	<li><a href="help/differential-atlas.html#searching">Searching the Differential Atlas</a>
		<ul>
			<li><a href="help/differential-atlas.html#minPvalLogFC">Specifying the maximum adjusted <i>p</i>-value and minimum log<sub>2</sub> fold-change</a></li>
			<li><a href="help/differential-atlas.html#geneSearch">Searching with genes</a></li>
			<li><a href="help/differential-atlas.html#comparisonSearch">Searching with comparisons</a></li>
		</ul>
	</li>
	<li><a href="help/differential-atlas.html#heatmap">Heatmap results display</a></li>
		<ul>
			<li><a href="help/differential-atlas.html#plots">MA plots and gene set overlap summaries</a></li>
			<li><a href="help/differential-atlas.html#ensembl">Visualising differential expression at Ensembl</a></li>
		</ul>
	</li>
	<li><a href="help/differential-atlas.html#downloadResults">Downloading query results</a></li>
	<li><a href="help/differential-atlas.html#downloadData">Downloading other experimental data</a></li>
	<li><a href="help/differential-atlas.html#qc">QC reports</a></li>
	<li><a href="help/differential-atlas.html#anaMethods">Analysis methods page</a></li>
	<li><a href="help/differential-atlas.html#exptDesign">Experiment design page</a>
		<ul>
			<li><a href="help/differential-atlas.html#sortExptDesign">Sorting the experiment design</a></li>
		</ul>
	</li>
	<li><a href="help/differential-atlas.html#feedbackSection">Providing feedback</a></li>
</ul>
</aside>

