<section>
	<h1>Frequently Asked Questions (FAQs)</h1>

<!-- Beginning of main text section -->
	<div class="grid_18">

		<h2><a name="general">General questions</a></h2>

		<h3><a name="whatIsAtlas">What is the Expression Atlas</a></h3>
		<p>The Expression Atlas database provides information
		on gene expression patterns under different biological conditions.
		It consists of two components: the 'differential' and 'baseline'
		atlases. The Differential Atlas, containing both microarray and
		sequencing data, allows users to query which genes are
		up-/down-regulated in different experimental conditions, e.g. 'in
		Arabidopsis shoots, what genes are upregulated in plants treated by
		X?'  The Baseline Atlas, containing exclusively RNA-seq data, displays
		expression levels of gene products under 'normal' conditions
		(e.g. normal human tissues).</p>

		<h3><a name="exptChosen">How are experiments chosen to be in the Expression Atlas?</a></h3>
		<p>Experiments in the Expression Atlas are selected from the <a
		href="http://www.ebi.ac.uk/arrayexpress">ArrayExpress</a> database of
		functional genomics experiments. Different criteria are used for
		selecting microarray and high throughput sequencing experiments.
		Microarray data must be gene expression data (rather than ChiP-chip or
		CGH for example), it must have a sufficent number of replicates to allow robust
		statistical analysis, and it should be possible to re-annotate the array design against Ensembl.  For RNA-seq data the sequences must be
		of high quality, there should a good quality reference genome build
		available, there must be sufficent replicates for statistical analysis
		for differential experiments and the tissues must be 'normal' if the
		experiment is be used in the Baseline Atlas.<p>

		<h3><a name="getOrigData">How can I obtain the original data for an
		experiment?</a></h3>
		<p>The original raw and processed data files for experiments in
		the Expression Atlas can be found in the <a
		href="http://www.ebi.ac.uk/arrayexpress">ArrayExpress</a>
		database. To link to the data in ArrayExpress click on the experiment accession number
		e.g. E-GEOD-15325 or E-MTAB-599, on any experiment page in the
		Expression Atlas.  In ArrayExpress the original submitted data files can be downloaded as zip
		archives, and sample annotation is available in MAGE-TAB format text
		files. See the <a
		href="https://www.ebi.ac.uk/arrayexpress/help/quick_start.html">ArrayExpress
		Quick Start</a> guide for further information.

		<h3><a name="wholeExpt">It doesn't look like the whole of the original
		experiment is in the Expression Atlas, why not?</a></h3>
		<p>We sometimes only include part of an experiment
		in the Expression Atlas because (1) there are not sufficient replicates of all the sample groups
		within an experiment, or (2) the hybridization or sequencing was not of
		high enough quality.  If there are still enough assays in the experiment after the removal of those with too
		few replicates or low quality then we continue
		processing the experiment for the Expression Atlas.</p>

		<h3><a name="arrayQuality">How is microarray data quality controlled?</a></h3>

		<p>Microarray data quality is assessed using the <a
		href="http://www.bioconductor.org/packages/release/bioc/html/arrayQualityMetrics.html">arrayQualityMetrics</a>
		package in <a href="http://www.r-project.org/">R</a>. Outlier arrays are
		detected using distance measures, boxplots, and MA plots. If an array is
		classed as an outlier by all three methods, it is excluded from further
		analysis. Please see the arrayQualityMetrics <a
		href="http://www.bioconductor.org/packages/release/bioc/vignettes/arrayQualityMetrics/inst/doc/arrayQualityMetrics.pdf">documentation</a>
		for more details on the methods used.</p>

		<h3><a name="rnaseqQuality">How is RNA-seq data quality controlled?</a></h3>

		<p>RNA-seq reads are discarded based on several criteria. First, reads with
		quality scores less than Q10 are removed. Second, the reads are mapped against
		a contamination reference genome (<i>E. coli</i> for animal data, fungal and
		microbial non-redundant reference for plants). Any reads that map to the
		contamination reference are removed. Third, reads with "uncalled" characters
		(i.e. "N"s) are discarded. Lastly, for paired-end libraries, any reads whose
		mate was lost in the previous three steps are also discarded. Please see the <a
		href="http://nunofonseca.github.io/irap/">iRAP</a> <a
		href="http://biorxiv.org/content/early/2014/06/06/005991">documentation</a> for
		more details on the methods used.</p>

		<h3><a name="microarrayAnalysis">How is microarray data analysed?</a></h3>

		<p>Raw single-channel microarray intensities are normalized using <a
			href="http://biostatistics.oxfordjournals.org/content/4/2/249.long">RMA</a>
		via the <a
			href="http://www.bioconductor.org/packages/release/bioc/html/oligo.html">oligo</a>
		package from <a href="http://www.bioconductor.org/">Bioconductor</a> (<a
			href="http://www.affymetrix.com/">Affymetrix</a> data) or using quantile
		normalization via the <a
			href="http://www.bioconductor.org/packages/release/bioc/html/limma.html">limma</a>
		package (<a href="http://www.genomics.agilent.com/">Agilent</a> data).
		Two-channel <a href="http://www.genomics.agilent.com/">Agilent</a> data is
		normalized using LOESS via
		the <a
			href="http://www.bioconductor.org/packages/release/bioc/html/limma.html">limma</a>
		package. Pairwise comparisons are performed using a moderated <i>t</i>-test for
		each gene using <a
			href="http://www.bioconductor.org/packages/release/bioc/html/limma.html">limma</a>.</p>

		<h3><a name="rnaseqAnalysis">How is RNA-seq data analysed?</a></h3>

		<p>RNA-seq data is analysed using the <a
			href="http://nunofonseca.github.io/irap/">iRAP</a> pipeline. <a
		href="http://nar.oxfordjournals.org/content/38/6/1767.long">FASTQ</a> files are
		mapped to the reference genome from <a
		href="http://www.ensembl.org/index.html">Ensembl</a> using <a
		href="http://tophat.cbcb.umd.edu/">TopHat</a>. Raw counts are generated
		using <a
		href="http://www-huber.embl.de/users/anders/HTSeq/doc/count.html">htseq-count</a>.
		FPKMs are calculated from the raw counts by <a
					href="http://nunofonseca.github.io/irap/">iRAP</a> pipeline.
			Pairwise comparisons are performed using a conditioned test based on the negative
		binomial distribution, using <a
		href="http://www.bioconductor.org/packages/release/bioc/html/DESeq.html">DESeq</a>.</p>

		<h3><a name="analysisMethods">How do I see how an individual experiment was analysed?</a></h3>

		<p>On any experiment page, you can see a breakdown of the analysis steps from raw data
		to the results you see in Expression Atlas by clicking the <img alt="analysis
		methods button" src="resources/images/help/am_btn.png" align="bottom" /> button
		to the top right of the page.</p>

		<h3><a name="contact">How can I contact you?</a></h3>

		<p>If you have any questions, problems or suggestions we would love to hear
		from you. Please email us at <a
			href="mailto:arrayexpress-atlas@ebi.ac.uk">arrayexpress-atlas@ebi.ac.uk</a>.</p>

		<h3><a name="mailingList">How can I keep up with the latest Expression Atlas news?</a></h3>

		<p>If you would like to stay up-to-date with news about our latest releases and
		developments, please subscribe to the <a
		href="http://listserver.ebi.ac.uk/mailman/listinfo/arrayexpress-atlas">Expression
		Atlas mailing list</a>.</p>

		<h3><a name="anatomogram">Can I use your anatomograms on my website?</a></h3>

		<p>Yes! The anatomical diagrams (anatomograms) that you see alongside the
		baseline data are available from <a
			href="https://github.com/gxa/atlas/tree/master/web/src/main/webapp/resources/svg">GitHub</a>.
			The anatomograms are licenced under <a
				href="https://creativecommons.org/licenses/by/4.0/">Creative Commons
			Attribution (CC BY)</a>; if you'd like to use them, all we ask is that you
		attribute <a href="http://www.ebi.ac.uk/gxa">Expression Atlas</a> when you do.</p>



		<h2><a name="searching">Searching</a></h2>


		<h3><a name="tissueSearch">How do I find out which genes are expressed in my favourite tissue?</a></h3>

		<p>Use the <strong>Experimental conditions</strong> search box on the <a
			href="/gxa">home page</a> to search all of Expression Atlas for organism parts e.g. <a
			href="/gxa/query?geneQuery=&exactMatch=true&_exactMatch=on&condition=kidney">kidney</a>.
		Your query is expanded using <a href="http://www.ebi.ac.uk/efo">EFO</a>, so
		that this search will also return results matching synonyms and child terms of
		<a
		href="http://www.ebi.ac.uk/efo/search?query=kidney&submitSearch=Search">kidney</a>
		in EFO.  You will see results about baseline and differential expression of
		genes in the organism part(s) you searched for. You can narrow down the search
		to specific genes by also typing gene identifiers in the <strong>Gene
		query</strong> search
		box.</p>

		<p>Use the <strong>Organism part</strong> box on a Baseline experiment page (e.g. <a
			href="experiments/E-MTAB-513">Illumina Body Map</a>) to search within a
		single experiment. See the <a href="help/baseline-atlas.html">Baseline Atlas
			help</a> for more information about searching within a Baseline experiment.</p>


		<h3><a name="multiquery">How do I search for multiple conditions at once?</a></h3>

		<p>Searching with space-separated experimental variables or sample
		characteristics finds experiments with any one of the terms you entered, or all
		of them. For example, searching with <strong>liver heart</strong> will find
		all experiments with both liver <strong>and</strong> heart as well as ones with
		<strong>either</strong> liver <strong>or</strong> heart.</p>

		<p>If you want to only find experiments with <strong>all</strong> terms you
		entered, separate them with <strong>AND</strong>. For example, searching for
		<strong>"wild type" AND Col-0</strong> will find only experiments annotated
		with wild type <strong>and</strong> Col-0, but not those with only one of the
		terms.</p>


		<h3><a name="geneSearch">How do I search for my favourite gene?</a></h3>

		<p>Use the <strong>Gene query</strong> search box on the <a href="/gxa">home
			page</a> to search all of Expression Atlas for genes. You will see results
		about baseline and differential expression for the gene or genes you searched
		for. You can narrow down the search to specific experimental conditions by also
		typing something in the <strong>Experimental conditions</strong> search
		box.</p>

		<p>Use the <strong>Gene query</strong> search box on an experiment page (e.g. <a
			href="experiments/E-MTAB-513">Illumina Body Map</a>) to search for genes
		within a single experiment.</p>

		<h3><a name="multiGeneQuery">How do I search for multiple genes at once?</a></h3>

		<p>Enter gene IDs and/or keywords separated by spaces. Put multi-word search
		terms in quotes, e.g. &quot;transcription factor binding&quot;.</p>


		<h3><a name="geneIDs">What gene identifiers can I use to search?</a></h3>

		<p>You may use the following identifiers to search using the <strong>Gene query</strong> box:</p>

		<ul>
			<li><a href="http://www.ebi.ac.uk/ena/">EMBL</a> e.g. <strong>AC000015</strong></li>
			<li><a href="http://www.ensembl.org/index.html">Ensembl</a> Gene e.g. <strong>ENSG00000171658</strong></li>
			<li><a href="http://www.ensembl.org/index.html">Ensembl</a> Protein e.g. <strong>ENSP00000000233</strong></li>
			<li><a href="http://www.ensembl.org/index.html">Ensembl</a> Transcript e.g. <strong>ENST00000000233</strong></li>
			<li><a href="http://www.geneontology.org/">Gene Ontology</a> ID e.g. <strong>GO:0008134</strong></li>
			<li><a href="http://www.geneontology.org/">Gene Ontology</a> term e.g. <strong>&quot;transcription factor binding&quot;</strong> (please put multi-word query terms in quotes)</li>
			<li><a href="http://www.ebi.ac.uk/interpro/">Interpro</a> e.g. <strong>IPR017892</strong></li>
			<li><a href="http://www.ncbi.nlm.nih.gov/refseq/">RefSeq</a> e.g. <strong>NM_212505</strong></li>
			<li><a href="http://www.uniprot.org/">UniProt</a> e.g. <strong>Q8IZT6</strong></li>
			<li><a href="http://www.uniprot.org/">UniProt</a> Metabolic Enzyme e.g. <strong>O15269</strong></li>
			<li>gene or protein name e.g. <strong>WNT1</strong> or <strong>Wnt-1</strong></li>
			<li>keyword e.g. <strong>transcription</strong></li>
			<li>gene name synonym e.g. <strong>Calmbp1</strong></li>
			<li><a href="http://www.ensembl.org">Ensembl</a> gene biotype e.g. <strong>protein_coding</strong> or <strong>non_coding</strong>. See the <a href="http://www.ensembl.org/Help/Glossary?id=275">glossary</a> at Ensembl for more details.</li>
		</ul>


		<h3><a name="exptLink">How do I find a particular experiment?</a></h3>

		<p>All experiments currently in Expression Atlas are listed on the <a
			href="experiments">Experiments page</a>. Click on the <strong>Description</strong>
		to see the experiment in Expression Atlas. Click on the <a
			href="http://www.ebi.ac.uk/arrayexpress">ArrayExpress</a> accession number
		(e.g. <strong>E-MTAB-1066</strong>) to see the experiment in ArrayExpress. If you know the
		ArrayExpress accession of the experiment you want to see, you can link to the
		experiment in Expression Atlas
		using the following format:</p>

		<p><code>http://www.ebi.ac.uk/gxa/experiments/<strong>&lt;ArrayExpress accession&gt;</strong></code></p>

		<p>E.g. <a
			href="experiments/E-MTAB-1066">http://www.ebi.ac.uk/gxa/experiments/E-MTAB-1066</a></p>


		<h3><a name="api">Can I search Expression Atlas programmatically?</a></h3>

		<p>We are working to implement a formal API to query Expression Atlas. Please
		check back soon for updates.</p>

		<p>In the meantime, if you would like to query <strong>differential</strong>
		Expression Atlas data programmatically, you can construct queries using URLs
		like the ones in the table below. Please be aware that the format of these URLs
		is subject to change. If your queries stop working, please check back here for
		the latest standard or email us at <a
		href="mailto:arrayexpress-atlas@ebi.ac.uk">arrayexpress-atlas@ebi.ac.uk</a>.<p>

		<p>
		<table style="border: 1px #cdd4d2 solid; width: 95%" >
			<thead>
				<tr style="background-color: #E1EEE9">
					<th style="border-right: 1px #cdd4d2 solid; border-bottom: 1px #cdd4d2 solid; padding: 0.5em 1em;">Query</th>
					<th style="border-right: 1px #cdd4d2 solid; border-bottom: 1px #cdd4d2 solid;">URL</th>
					<th style="border-right: 1px #cdd4d2 solid; border-bottom: 1px #cdd4d2 solid;">Result</th>
				</tr>
			</thead>
			<tbody>
				<tr>
					<td style="text-align: center; border-right: 1px #cdd4d2 solid; border-bottom: 1px #cdd4d2 solid;">In what conditions is <a href="http://www.ebi.ac.uk/gxa/genes/ENSG00000066279">ASPM</a> differentially expressed?</td>
					<td style="text-align: center; border-right: 1px #cdd4d2 solid; border-bottom: 1px #cdd4d2 solid;">http://www.ebi.ac.uk/gxa/query.tsv?geneQuery=ASPM</td>
					<td style="text-align: center; border-right: 1px #cdd4d2 solid; border-bottom: 1px #cdd4d2 solid;">A <a href="http://en.wikipedia.org/wiki/Tab-separated_values">tab-delimited</a> text file containing results of comparisons where ASPM is differentially expressed with adjusted <i>p</i>-value &lt; 0.05.</td>
				</tr>
				<tr>
					<td style="text-align: center; border-right: 1px #cdd4d2 solid; border-bottom: 1px #cdd4d2 solid;">What genes are differentially expressed in cancer?</td>
					<td style="text-align: center; border-right: 1px #cdd4d2 solid; border-bottom: 1px #cdd4d2 solid;">http://www.ebi.ac.uk/gxa/query.tsv?condition=cancer</td>
					<td style="text-align: center; border-right: 1px #cdd4d2 solid; border-bottom: 1px #cdd4d2 solid;">A tab-delimited text file containing results for genes that are differentially expressed in experiments about cancer.</td>
				</tr>
				<tr>
					<td style="text-align: center; border-right: 1px #cdd4d2 solid;">Show me comparisons where <a href="http://www.ebi.ac.uk/interpro/entry/IPR007087">zinc finger</a> genes are differentially expressed in mice.</td>
					<td style="text-align: center; border-right: 1px #cdd4d2 solid;">http://www.ebi.ac.uk/gxa/query.tsv?geneQuery=%22zinc+finger%22&amp;condition=%22mus+musculus%22</td>
					<td style="text-align: center; border-right: 1px #cdd4d2 solid;">A tab-delimited text file containing results of comparisons where genes annotated as having zinc finger domains are differentially expressed, in <i>Mus musculus</i>.</td>
				</tr>
			</tbody>
		</table>
		</p>




		<h2><a name="results">Results</a></h2>


		<h3><a name="deResults">When I search all of Expression Atlas, what do the &quot;Differential Expression&quot; results show?</a></h3>

		<p>The heatmap table in the <strong>Differential Expression</strong> section
		shows all genes matching your query that are differentially expressed with
		adjusted <i>p</i>-value &lt; 0.05. If your search included an experimental
		condition then the table displays comparisons with a sample property or
		experimental variable that matched that condition.</p>

		<p>There is one row for each gene and comparison in which it showed differential
		expression. The genes are ordered so that the one with the largest
		log<sub>2</sub> fold-change is at the top. If some genes have identical
		log<sub>2</sub> fold-changes, the one with the lower adjusted <i>p</i>-value
		goes first. In microarray experiments, some genes may be targeted by more than
		one probe set. For each experiment, the table shows the values for the probe
		set that had the largest absolute log<sub>2</sub> fold-change only. You can see
		details of all probe sets for a given gene on the <a href="FAQ.html#DEexptPage">experiment
		page</a>. Click on the comparison name to find out more information about a
		particular gene/comparison combination.</p>

		<p>Please see the <a
			href="help/differential-atlas.html">Differential Atlas help page</a> for
		more details.</p>


		<h3><a name="designElement">What is a &quot;design element&quot;?</a></h3>

		<p>On microarray experiment pages, you will see the <strong>design
			element</strong> name alongside the gene name. A design element is also
		known as a probe or probe set. This is the oligonucleotide probe (or group
		thereof) on the microarray that targets that gene.</p>


		<h3><a name="comparison">What is a &quot;comparison&quot;?</a></h3>

		<p>A <strong>comparison</strong> is where two groups of samples are compared in
		a differential expression experiment. An example of a comparison is
		<strong>&quot;dicer knock-out&quot; <i>vs.</i> &quot;wild
			type&quot;</strong>. For each gene, the mean expression level of the
		<strong>test</strong> group (e.g. dicer knock-out) is compared with the
		mean expression level of the <strong>reference</strong> group (e.g. wild type),
		and a statistical test is performed to decide whether the two means are
		significantly different.</p>


		<h3><a name="pvalue">Where do the <i>p</i>-values come from?</a></h3>

		<p>In a microarray experiment, each gene's mean expression level in the
		<strong>test</strong> group is compared with its mean expression level in the
		<strong>reference</strong> group using a moderated <i>t</i>-test. This is done
		using the <a
			href="http://www.bioconductor.org/packages/2.13/bioc/html/limma.html">limma</a>
		package from <a href="http://www.bioconductor.org/">Bioconductor</a>.</p>

		<p>In an RNA-seq experiment, each gene's mean expression level in the
		<strong>test</strong> group is compared with its mean expression level in the
		<strong>reference</strong> group using a conditioned test based on the negative
		binomial distribution, analogous to Fisher's exact test. This is done using the
		<a
			href="http://www.bioconductor.org/packages/2.13/bioc/html/DESeq.html">DESeq</a>
		package in <a href="http://www.bioconductor.org/">Bioconductor</a>.</p>

		<p>Because the same test is done on thousands of genes at once, <i>p</i>-values
		are adjusted for <strong>multiple testing</strong> using the <a
			href="http://www.jstor.org/stable/2346101">Benjamini and
		Hochberg (1995)</a> false discovery rate
		(FDR) correction. This is what is meant by <strong>adjusted
			<i>p</i>-value</strong>.</p>

		<h3><a name="gseaPlots">How are the gene set enrichment plots created?</a></h3>

		<p>Gene set enrichment analysis is performed using the <a
		href="http://www.bioconductor.org/packages/release/bioc/html/piano.html">Piano</a>
		package from <a href="http://www.bioconductor.org/">Bioconductor</a>. For each
		comparison, enrichment of terms from <a
		href="http://www.geneontology.org/">GO</a>, <a
		href="http://www.ebi.ac.uk/interpro/">InterPro</a>, and <a
		href="http://www.reactome.org/">Reactome</a> is tested for within the set of
		differentially expressed genes, using a variation on <a
		href="http://en.wikipedia.org/wiki/Fisher%27s_exact_test">Fisher's exact
		test</a>.  Gene set enrichment plots are only shown when statistically
		significant enrichment of terms was detected. This means that for some
		experiments, the menu will not display plots for all three of the
		aforementioned resources. Please see the <a
		href="http://www.bioconductor.org/packages/release/bioc/html/piano.html">Piano</a>
		documentation for more details.</p>


		<h3><a name="deHeatmapColours">In Differential Atlas, what do the red and blue colours in the table mean?</a></h3>

		<p>A red box indicates
		that the gene was <strong>up-regulated</strong> in the test condition. A blue
		box means that the gene was <strong>down-regulated</strong> in the test
		condition.</p>

		<p>The colour intensity of filled boxes in the table
		represents how large the log<sub>2</sub> fold-change is for each gene. The larger the
		log<sub>2</sub> fold-change, the more intense the red or blue colour.</p>



		<h3><a name="displayLogFoldChanges">How do I see the actual log<sub>2</sub> fold-changes in the results table?</a></h3>

		<p>Click the <strong>Display log<sub>2</sub> fold-change</strong> button to the top left of the table.</p>


		<h3><a name="deBars">What are the blue and red bars above the results table?</a></h3>

		<p>The two bars show the red and blue colour intensities for the <strong>top
			50</strong> genes shown on the page. The colour intensities represent the
		log<sub>2</sub> fold-changes for the genes shown. To see the actual
		log<sub>2</sub> fold-changes, click the &quot;Display log<sub>2</sub>
		fold-change&quot; button to the top left of the table.</p>

		<h3><a name="DEexptPage">What does the differential expression experiment page show?</a></h3>

		<p>Each experiment in Expression Atlas has its own experiment page, e.g. <a
			href="experiments/E-MTAB-1066"><i>D. melanogaster</i> CDK8 and CycC
			mutants</a>. Here you can see the results of analysis for each comparison in
		a single experiment. Please see the <a
			href="help/differential-atlas.html">Differential Atlas help page</a> for
		more details.</p>


		<h3><a name="downloadAnalytics">Can I download the differential expression results?</a></h3>

		<p>Yes, click on the <img src="resources/images/help/dlAna_btn.png"
		alt="analytics download button" align="bottom" /> button to the top right of
		any differential experiment page, e.g. <a
			href="experiments/E-MTAB-1066"><i>D. melanogaster</i> CDK8 and CycC
			mutants</a>. For more details about the differential expression experiment pages,
		please see the <a href="help/differential-atlas.html">Differential Atlas help
			page</a>.</p>

		<h3><a name="downloadDEdata">Can I download the data used for differential expression analysis?</a></h3>

		<p>Yes, please use the download buttons to the top right of any experiment
		page. For single-channel microarray experiments you can download the normalized
		intensities using the <img src="resources/images/help/dlNorm_btn.png"
		alt="normalized data download button" align="bottom" /> button. For two-channel
		microarray experiments you can download the log<sub>2</sub> fold-changes using
		the <img src="resources/images/help/dlLogFC_btn.png"
		alt="download log2 fold changes button" align="bottom" /> button. For RNA-seq
		experiments you can download the raw counts generated by <a
			href="http://www-huber.embl.de/users/anders/HTSeq/doc/count.html">htseq-count</a>
		using the <img src="resources/images/help/dlRawCounts_btn.png" alt="download
		raw RNA-seq counts" align="bottom" /> button. For more details about the differential expression experiment pages,
		please see the <a href="help/differential-atlas.html">Differential Atlas help
			page</a>.</p>

		<h3><a name="blResults">When I search all of Expression Atlas, what do the &quot;Baseline Expression&quot; results show?</a></h3>

		<p>The table in the <strong>Baseline Expression</strong> section of the
		results shows all experiments that matched your search.  </p>

		<p>If your search was for an Experimental condition then it includes all experiments with a
		sample property or experimental variable that matched that
		condition. For example you might be interested in experiments that are
		from 'liver' samples and so you could search for this term.</p>

		<p>If you searched for a particular gene then the experiments returned
		are ones in which that gene was expressed above the default minimum
		expression level (FPKM - Fragments Per Kilobase of transcript per
		Million mapped reads) which is 0.5.</p>

		<p>Click on the name of the
		experiment to view more information about the experiment. </p>

		<p>Please see the <a href="help/baseline-atlas.html">Baseline Atlas help page</a> for
		more details.</p>


		<h3><a name="BLexptPage">What does the baseline expression
		experiment page show?</a></h3>
		<p>Each experiment in Expression Atlas has its own experiment page, e.g. <a
			href="experiments/E-MTAB-599">RNA-seq of mouse DBA/2J x
		C57BL/6J heart, hippocampus, liver, lung, spleen and thymus</a>. Here
		you can see a heat map showing the 50 most specifically expressed
		genes across all conditions studied.  You can further refine the query
		by narrowing the search to particular genes, or gene sets, or by
		limiting which organism parts are searched over. Please see the <a href="help/baseline-atlas.html">Baseline Atlas help page</a> for
		more details.</p>

		<h3><a name="blHeatmapColours">In Baseline Atlas, what do the different shades of blue
		in the heat map mean?</a></h3>

		<p> The heatmap shows expression levels by colour intensity, according to
		the gradient bar displayed above the heatmap. The intensities correspond to expression levels for the top 50 genes
		currently displayed (rather than all FPKM values for all genes returned by the query).</p>

		<h3><a name="blActualValues">How do I see the actual expression levels
		in the Baseline Atlas heat map?</a></h3>

		<p>Click on the <strong>Display levels</strong> button in the top left corner of the
		heat map to show the actual FPKM values.</p>

		<h3><a name="BLdownloadAnalytics">Can I download the baseline
		expression results?</a></h3>

		<p>Yes, click on the <img alt="Download button"
		src="resources/images/help/download_button.png" align="bottom" />
		button  to download the full results of your query in tab&ndash;delimited format with no ordering.<p>

		<h3><a name="ensembl">Can I view the results in the Ensembl browser?</a></h3>

		<p>Yes. From the heatmap, select a gene name, and either a comparison (differential) or
		a condition (baseline) from the column headings. Next, click the Ensembl Genome
		Browser <strong>Open</strong> button, to the left of the heatmap.</p>

		<h3><a name="similarExpression">How are similarly expressed genes computed?</a></h3>

		<p>Similarly expressed genes across conditions in baseline experiments (e.g. tissues, developmental stages) are shown where available. They are computed on per-experiment basis, for experiments with three or more conditions. The method for estimating them involves two steps. The first one involves k-means clustering of the expression of each gene across (e.g.) tissues. The second step compares the clusters on a gene to gene basis and outputs a ranked list of genes with decreasing similarity of expression patterns for each gene. The top 50 similarly expressed genes can be explored on our interface. Lowly expressed genes are filtered out from the calculation. A manuscript describing the methods is currently in preparation and the method will be soon available as an R package.</p>

	</div>
</section>
<!-- End of main text section -->

<!-- This part is for the navigation section at the right-hand side -->
<aside class="grid_6">
	<h3>Jump to...</h3>
	<h4><a href="FAQ.html#general">General</a></h4>
	<ul>
		<li><a href="FAQ.html#whatIsAtlas">What is the Expression Atlas</a></li>
		<li><a href="FAQ.html#exptChosen">How are experiments chosen to be in the Expression Atlas?</a></li>
		<li><a href="FAQ.html#getOrigData">How can I obtain the original data for an experiment?</a></li>
		<li><a href="FAQ.html#wholeExpt">It doesn't look like the whole of the original experiment is in the Expression Atlas, why not?</a></li>
		<li><a href="FAQ.html#arrayQuality">How is microarray data quality controlled?</a></li>
		<li><a href="FAQ.html#rnaseqQuality">How is RNA-seq data quality controlled?</a></li>
		<li><a href="FAQ.html#microarrayAnalysis">How is microarray data analysed?</a></li>
		<li><a href="FAQ.html#rnaseqAnalysis">How is RNA-seq data analysed?</a></li>
		<li><a href="FAQ.html#analysisMethods">How do I see how an individual experiment was analysed?</a></li>
		<li><a href="FAQ.html#contact">How can I contact you?</a></li>
		<li><a href="FAQ.html#mailingList">How can I keep up with the latest Expression Atlas news?</a></li>
	</ul>
	<h4><a href="FAQ.html#searching">Searching</a></h4>
	<ul>
		<li><a href="FAQ.html#tissueSearch">How do I find out which genes are expressed in my favourite tissue?</a></li>
		<li><a href="FAQ.html#geneSearch">How do I search for my favourite gene?</a></li>
		<li><a href="FAQ.html#multiGeneQuery">How do I search for multiple genes at once?</a></li>
		<li><a href="FAQ.html#geneIDs">What gene identifiers can I use to search?</a></li>
		<li><a href="FAQ.html#exptLink">How do I find a particular experiment?</a></li>
		<li><a href="FAQ.html#api">Can I search Expression Atlas programmatically?</a></li>
	</ul>
	<h4><a href="FAQ.html#results">Results</a></h4>
	<ul>
		<li><a href="FAQ.html#deResults">When I search all of Expression Atlas, what do the &quot;Differential Expression&quot; results show?</a></li>
		<li><a href="FAQ.html#designElement">What is a &quot;design element&quot;?</a></li>
		<li><a href="FAQ.html#comparison">What is a &quot;comparison&quot;?</a></li>
		<li><a href="FAQ.html#pvalue">Where do the <i>p</i>-values come from?</a></li>
		<li><a href="FAQ.html#gseaPlots">How are the gene set enrichment plots created?</a></li>
		<li><a href="FAQ.html#deHeatmapColours">In Differential Atlas, what do the red and blue colours in the table mean?</a></li>
		<li><a href="FAQ.html#displayLogFoldChanges">How do I see the actual <i>p</i>-values in the results table?</a></li>
		<li><a href="FAQ.html#deBars">What are the blue and red bars above the results table?</a></li>
		<li><a href="FAQ.html#DEexptPage">What does the differential expression experiment page show?</a></li>
		<li><a href="FAQ.html#downloadAnalytics">Can I download the differential expression results?</a></li>
		<li><a href="FAQ.html#downloadDEdata">Can I download
		the data used for differential expression
		analysis?</a></li>
		<li><a href="FAQ.html#blResults">When I search all of
		Expression Atlas, what do the &quot;Baseline
		Expression&quot; results show?</a></li>
                <li><a href="FAQ.html#BLexptPage">What does the
		baseline expression experiment page show?</a></li>
		<li><a href="FAQ.html#blHeatmapColours">In Baseline
		Atlas, what do the
		different shades of blue in the heat map mean?</a></li>
		<li><a href="FAQ.html#blActualValues">How do I see
		the actual expression levels in the Baseline Atlas heat map?</a></li>
        <li><a href="FAQ.html#BLdownloadAnalytics">Can I
		download the baseline expression results?</a></li>
		<li><a href="FAQ.html#ensembl">Can I view the results in the Ensembl browser?</a></li>
		<li><a href="FAQ.html#similarExpression">How are similarly expressed genes computed?</a></li>

		
		<!-- <li><a href="FAQ.html#"></a></li> -->
	</ul>
</aside>
