<!DOCTYPE html>

<!--
  ~ Copyright 2008-2012 Microarray Informatics Team, EMBL-European Bioinformatics Institute
  ~
  ~ Licensed under the Apache License, Version 2.0 (the "License");
  ~ you may not use this file except in compliance with the License.
  ~ You may obtain a copy of the License at
  ~
  ~ http://www.apache.org/licenses/LICENSE-2.0
  ~
  ~ Unless required by applicable law or agreed to in writing, software
  ~ distributed under the License is distributed on an "AS IS" BASIS,
  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  ~ See the License for the specific language governing permissions and
  ~ limitations under the License.
  ~
  ~
  ~ For further details of the Gene Expression Atlas project, including source code,
  ~ downloads and documentation, please see:
  ~
  ~ http://gxa.github.com/gxa
  -->

<html>
<head>
    <META HTTP-EQUIV="CACHE-CONTROL" CONTENT="NO-CACHE">
    <META HTTP-EQUIV="PRAGMA" CONTENT="NO-CACHE">
    <title></title>
</head>
<body>

<!-- request requestPreferences -->
<div id="geneSearch">Enter Ensembl gene ID(s), gene symbol(s) or keywords. Press tab before entering another term. Leave blank for all genes.
</div>

<div id="experimentalConditions">Enter sample attributes or experimental variable(s), e.g. tissues, genotypes, treatments.
</div>

<div id="filterBy">Select experimental variable(s) to limit results by.
</div>

<div id="specific">Check to see genes specific to experimental variable(s) selected, with most specific genes first. Uncheck to sort by highest expressed genes.
</div>

<div id="specific-differential">Check to see genes specific to comparison(s) selected, with most specific genes first. Uncheck to sort by highest absolute log<sub>2</sub> fold-change.
</div>

<div id="factorSearch">Select experimental variable(s), e.g. different tissues or cell types.
</div>

<div id="factorSearch-differential">Select comparison(s). A comparison compares two groups of assays: the test to the reference. Each group is described by a set of properties/values that are constant across all assays in that group.
</div>

<div id="cutoff">Enter a minimum expression level (FPKM). Only genes expressed above this are displayed. Set to 0 to retrieve all genes.
</div>

<div id="cutoff-proteomics">Enter a minimum expression level. Only genes expressed above this are displayed. Set to 0 to retrieve all genes.
</div>

<div id="cutoff-differential">Enter value from 0 to 1 to use as the maximum <i>p</i>-value cutoff.
    Only genes that show differential expression with an adjusted <i>p</i>-value at or below this cutoff will be shown.
</div>

<div id="foldChangeCutOff">Enter value equal or greater than 0 to use as the minimum absolute log<sub>2</sub>-fold change.
    Only genes that show absolute log<sub>2</sub>-fold change at or greater than this cutoff will be shown.
</div>

<div id="gene-distribution">The histogram shows number of genes (y-axis)
	expressed above each expression level (x-axis). Click a bar to select its expression level as the minimum.
</div>

<div id="toggleHistogramButton">Toggle display of histogram showing distribution of expression levels.
</div>

<div id="slider">Choose a minimum expression level.
</div>

<!--experiment -->
<div id="gradient-base">Expression level: greater colour saturation means higher expression.
</div>

<div id="gradient-base-crossexp">Expression level: greater colour saturation means higher expression.
  Please note that expression values are not directly comparable across experiments.
</div>

<div id="gradient-differential-crossexp">Greater colour saturation means higher absolute log<sub>2</sub>-fold change value.
    Blue indicates the gene is down–regulated and red means up–regulated.
    Please note that log2fold changes are not directly comparable across experiments.
</div>


<div id="gradient-differential">Greater colour saturation means higher absolute log<sub>2</sub>-fold change value.
    Blue indicates the gene is down–regulated and red means up–regulated.
</div>

<div id="heatMapTableCellInfo-baseline">Filled boxes indicate genes expressed. Click "Display levels" to show expression levels.
</div>

<div id="heatMapTableCellInfo-multiexperiment">Filled boxes indicate genes expressed. Click "Display levels" to show expression levels.
    An 'NA' in a box means that the experimental variable has not been studied in that experiment.
</div>

<div id="heatMapTableCellInfo-proteomics">Filled boxes indicate genes expressed. Click "Display levels" to show expression levels.
</div>

<div id="heatMapTableCellInfo-differential">Filled boxes indicate genes differentially expressed. Mouseover a box to see adjusted <i>p</i>-value and log<sub>2</sub> fold-change. Click "Display log<sub>2</sub> fold-change" to show exact log<sub>2</sub> fold-changes.
</div>

<div id="heatMapTableUnknownCell">Unknown expression level. Gene may or may not be expressed.
</div>

<div id="heatMapTableGeneSetUnknownCell">Unknown expression level because at least one of the constituent genes may or may not be expressed.
</div>

</body>
</html>
