<div class="row">
    <div class="columns small-10">

        <h1>Frequently Asked Questions</h1>

        <!-- Beginning of main text section -->
        <h2 id="general">General questions</h2>
        <h3 id="whatIsAtlas">What is Expression Atlas</h3>
        <p>
            Expression Atlas is a resource to query gene and protein expression data across species and biological
            conditions and to visualise down-stream analysis results to explore co-expression. Queries can be either in
            a <strong>baseline context</strong>, e.g.
            <a href="/gxa/search?organism=macaca%20mulatta&conditionQuery=%5B%7B%22value%22%3A%22brain%22%7D%5D">find
            genes expressed in the macaque brain</a>, or in a <strong>differential context</strong>, e.g.
            <a href="/gxa/search?geneQuery=%5B%5D&organism=Arabidopsis%20thaliana&conditionQuery=%5B%7B%22value%22%3A%22auxin%22%7D%5D&ds=%7B%22kingdom%22%3A%5B%22plants%22%5D%7D#differential">
            find genes that are up or downregulated in response to auxin in Arabidopsis</a>. Expression Atlas contains
            thousands of selected microarray and RNA-sequencing data that are manually curated and annotated with
            ontology terms, checked for high quality and re-analysed using standardised methods.
        </p>


        <h3 id="exptChosen">How are experiments chosen to be in Expression Atlas?</h3>
        <p>
            Experiments in Expression Atlas are selected from
            <a href="https://www.ebi.ac.uk/arrayexpress">ArrayExpress</a> that is an archive of functional genomics
            data. Different criteria are used for including experiments in Baseline Atlas or in Differential Atlas.
            All experiments in <strong>Baseline Atlas</strong> are RNA-seq data with good quality raw files and a broad
            coverage across conditions (e.g. tissues, cell types, developmental stages). A good quality reference
            genome build available is also required for RNA-seq data. All experiments in <strong>Differential
            Atlas</strong> have enough number of replicates to allow robust statistical analysis. Apart from that,
            microarray data must be gene expression data (rather than ChiP-chip or CGH for example) and it should be
            possible to re-annotate the array design against <a href="https://www.ensembl.org/index.html">Ensembl</a>.
        </p>

        <h3 id="getOrigData">How can I obtain the original data for an experiment?</h3>
        <p>
            The original raw and processed data files for experiments in Expression Atlas can be found by viewing the
            experiment in <a href="https://www.ebi.ac.uk/arrayexpress">ArrayExpress</a>. On any <strong>Experiment
            page</strong>, e.g. <a href="/gxa/experiments/E-MTAB-2770">RNA-seq of 934 human cancer cell lines from the
            Cancer Cell Line Encyclopedia</a> click on the
            <img alt="array express button" src="resources/images/help/ae_btn.png" align="bottom"> button in the
            <strong>Resources</strong> tab to view the experiment in ArrayExpress. The original submitted data files
            can be downloaded as zip archives, and sample annotation is available in MAGE-TAB format text files. See
            the online tutorial
            <a href="https://www.ebi.ac.uk/training/online/course/arrayexpress-discover-functional-genomics-data-qui/files-and-download">
            ArrayExpress: Discover functional genomics data quickly and easily</a> for further information.
        </p>


        <h3 id="wholeExpt">It doesn't look like the whole of the original experiment is in Expression Atlas, why not?</h3>
        <p>
            We sometimes only include part of an experiment in Expression Atlas because (1) there are not sufficient
            replicates of all the sample groups within an experiment, or (2) the hybridization or sequencing was not of
            high enough quality. If there are still enough assays in the experiment after the removal of those with too
            few replicates or low quality then we continue processing the experiment for Expression Atlas.
        </p>


        <h3 id="arrayQuality">How is microarray data quality controlled?</h3>
        <p>
            Microarray data quality is assessed using the
            <a href="https://www.bioconductor.org/packages/release/bioc/html/arrayQualityMetrics.html">
            arrayQualityMetrics</a> package in <a href="https://www.r-project.org/">R</a>. Outlier arrays are detected
            using distance measures, boxplots, and MA plots. If an array is classed as an outlier by all three methods,
            it is excluded from further analysis. Please see the arrayQualityMetrics
            <a href="https://www.bioconductor.org/packages/release/bioc/vignettes/arrayQualityMetrics/inst/doc/arrayQualityMetrics.pdf">
            documentation</a> for more details on the methods used.
        </p>


        <h3 id="rnaseqQuality">How is RNA-seq data quality controlled?</h3>
        <p>
            RNA-seq reads are discarded based on several criteria. First, reads with quality scores less than Q10 are
            removed. Second, the reads are mapped against a contamination reference genome (<i>E. coli</i> for animal
            data, fungal and microbial non-redundant reference for plants). Any reads that map to the contamination
            reference are removed. Third, reads with "uncalled" characters (i.e. "N"s) are discarded. Lastly, for
            paired-end libraries, any reads whose mate was lost in the previous three steps are also discarded. Please
            see the <a href="https://nunofonseca.github.io/irap/">iRAP</a>
            <a href="http://biorxiv.org/content/early/2014/06/06/005991">documentation</a> for more details on the
            methods used.
        </p>


        <h3 id="microarrayAnalysis">How is microarray data analysed?</h3>
        <p>
            Raw single-channel microarray intensities are normalized using
            <a href="https://biostatistics.oxfordjournals.org/content/4/2/249.long">RMA</a> via the
            <a href="https://www.bioconductor.org/packages/release/bioc/html/oligo.html">oligo</a> package from
            <a href="https://www.bioconductor.org/">Bioconductor</a> (<a href="https://www.affymetrix.com/">
            Affymetrix</a> data) or using quantile normalization via the
            <a href="https://www.bioconductor.org/packages/release/bioc/html/limma.html">limma</a> package
            (<a href="https://www.genomics.agilent.com/">Agilent</a> data). Two-channel
            <a href="https://www.genomics.agilent.com/">Agilent</a> data is normalized using LOESS via the
            <a href="https://www.bioconductor.org/packages/release/bioc/html/limma.html">limma</a> package. Pairwise
            comparisons are performed using a moderated <i>t</i>-test for each gene using
            <a href="https://www.bioconductor.org/packages/release/bioc/html/limma.html">limma</a>.</p>


        <h3 id="rnaseqAnalysis">How is RNA-seq data analysed?</h3>
        <p>
            RNA-seq data is analysed using the <a href="https://nunofonseca.github.io/irap/">iRAP</a> pipeline.
            Quality-filtered reads are aligned to the latest version of the reference genome from
            <a href="https://www.ensembl.org/index.html">Ensembl</a> using
            <a href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2013-14-4-r36">TopHat2</a>. Raw counts
            (number of mapped reads summarized and aggregated over each gene) are generated using
            <a href="http://htseq.readthedocs.io/">htseq-count</a>. Then, FPKM (fragments per kilobase of exon model
            per million mapped reads) and TPM (transcripts per million) are calculated. Pairwise comparisons are
            performed using a conditioned test based on the negative binomial distribution, using
            <a href="https://www.bioconductor.org/packages/release/bioc/html/DESeq.html">DESeq</a>.
        </p>


        <h3 id="rnaseqUnits">What are FPKM and TPM?</h3>
        <p>
            FPKM (fragments per kilobase of exon model per million reads mapped) and TPM (transcripts per million) are
            the most common units reported to estimate gene expression based on RNA-seq data. Both units are calculated
            from the number of reads that mapped to each particular gene sequence and both units are calculated taking
            into account two important factors in RNA-seq:
        </p>
        <ol>
            <li>The number of reads from a gene depends on its length. One expects more reads to be produced from
                longer genes.</li>
            <li>The number of reads from a gene depends on the sequencing depth that is the total number of reads you
                sequenced. One expects more reads to be produced from the sample that has been sequenced to a greater
                depth.</li>
        </ol>
        <p>
            FPKM (introduced by <a href="http://www.nature.com/nbt/journal/v28/n5/full/nbt.1621.html">Trapnell et al,
            2010</a>) are calculated with the following formula:
        </p>
        <p style="text-align:center;">
            <img  alt="fpkm formula" src="resources/images/help/fpkm.png">
        </p>
        <p>
            where <i>q<sub>i</sub></i> are raw counts (number of reads that mapped for each gene), <i>l<sub>i</sub></i>
            is gene length and
            <img alt="total number mapped reads" src="resources/images/help/total_number_mapped_reads.png" align="bottom">
            is the total number of mapped reads. The interpretation of FPKM is that if you sequence your RNA sample
            again, you expect to see for gene <i>i</i>, <i>FPKM<sub>i</sub></i> reads divided by gene <i>i</i> length
            over a thousand and divided by the total number of reads mapped over a million.
        </p>
        <p>
            <a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-12-323">
            Li and Dewey, 2011</a> introduced the unit TPM and
            <a href="https://arxiv.org/abs/1104.3889">Pachter, 2011</a> established the relationship between both
            units. It is possible to compute TPM from FPKM as follows:
        </p>
        <p style="text-align:center;">
            <img alt="tpm formula" src="resources/images/help/tpm.png">
        </p>


        <h3 id="analysisMethods">How do I see how an individual experiment was analysed?</h3>
        <p>
            On any <strong>Experiment page</strong>, e.g.<a href="/gxa/experiments/E-MTAB-2809">RNA-seq of coding RNA
            of eight barley tissues from different developmental stages</a> you can see a breakdown of the analysis
            steps from raw data to the results you see in Expression Atlas by selecting the <i>Experiment design</i> or
            <i>Supplemmentary information</i> tab on the top of the experiment section.
        </p>


        <h3 id="contact">How can I contact you?</h3>
        <p>
            If you have any questions, problems or suggestions we would love to hear from you. Please email us at
            <a href="mailto:arrayexpress-atlas@ebi.ac.uk">arrayexpress-atlas@ebi.ac.uk</a>.
        </p>


        <h3 id="mailingList">How can I keep up with the latest Expression Atlas news?</h3>
        <p>
            If you would like to stay up-to-date with news about our latest releases and developments, please subscribe
            to the <a href="https://listserver.ebi.ac.uk/mailman/listinfo/arrayexpress-atlas">Expression Atlas mailing
            list</a>.
        </p>


        <h3 id="anatomogram">Can I use Expression Atlas anatomograms on my website?</h3>
        <p>
            Yes! The anatomical diagrams (anatomograms) that you see alongside the baseline data are available from
            <a href="https://github.com/gxa/anatomogram/">GitHub</a>. The anatomograms are licenced under
            <a href="https://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution (CC BY)</a>; if you’d
            like to use them, all we ask is that you attribute <a href="/gxa">Expression Atlas</a> when you do.</p>



        <h2 id="searching">Searching</h2>

        <h3 id="conditionSearch">How do I find out what genes are expressed in a particular condition (e.g. in my
            favourite tissue, cell line, developmental stage)?</h3>
        <p>
            Use the <strong>Condition query</strong> search box on the <a href="/gxa">home page</a> to search for the
            condition you are interested in e.g.
            <a href="/gxa/search?conditionQuery=%5B%7B%22value%22%3A%22kidney%22%7D%5D">kidney</a>. Click on
            <strong>Show anatomogram</strong> in the top left corner if you want to see the corresponding anatomogram
            for each of the species displayed. Your query is expanded using the Experimental Factor Ontology
            <a href="https://www.ebi.ac.uk/efo/">(EFO)</a>, so that this search will also returns matching synonyms and
            child terms of <a href="https://www.ebi.ac.uk/ols/search?q=kidney&submit1=1&ontology=efo">kidney</a> in
            EFO. You will see both baseline expression and differential expression results in the condition (organism
            part in our example) you searched for.
        </p>
        <p>
            Select one experiment from the Baseline multi-experiment page (e.g.
            <a href="/gxa/experiments/E-MTAB-5214">GTEx</a>) to see the results in that particular
            experiment. Use the <i>Select</i> button under <strong>Organism parts</strong> on the left sidebar to find
            <a href="/gxa/experiments/E-MTAB-5214/Results?specific=true&geneQuery=%255B%255D&filterFactors=%257B%2522ORGANISM_PART%2522%253A%255B%2522cortex%2520of%2520kidney%2522%255D%257D&cutoff=%257B%2522value%2522%253A0.5%257D&unit=%2522TPM%2522">
            what genes are expressed in kidney in GTEx</a>.
        </p>


        <h3 id="multiConditionQuery">How do I search for multiple conditions at once?</h3>
        <p>
            Use the <strong>Condition query</strong> search box on the <a href="/gxa">home page</a> to search for as
            many conditions as you want. You will need to type each condition, click enter and when you are done, just
            search. For example, searching with
            <img alt="liver_heart" src="resources/images/help/liver_heart.png" align="bottom"> will find
            <a href="/gxa/search?conditionQuery=%5B%7B%22value%22%3A%22liver%22%7D%2C%7B%22value%22%3A%22heart%22%7D%5D">
            all experiments in which both liver and heart are studied as well as the ones analysing either liver or
            heart</a>.
        </p>


        <h3 id="geneSearch">How do I find out in which conditions my favourite gene is expressed?</h3>
        <p>
            Use the <strong>Gene query</strong> search box on the <a href="/gxa">home page</a> to search for your
            favourite gene (e.g.
            <a href="/gxa/search?geneQuery=%5B%7B%22value%22%3A%22SFTPC%22%2C%22category%22%3A%22symbol%22%7D%5D">
            SFTPC</a>). Click on <strong>Show anatomogram</strong> in the top left corner if you want to see the
            corresponding anatomogram for each of the species displayed. You will see <strong>Baseline
            expression</strong> results in different conditions (e.g. organism part, cell line, cell type,
            developmental stage) and <strong>Differential expression</strong> results for biologically meaningful
            pairwise comparisons.
        </p>
        <p>
            Select one experiment from the <strong>Baseline expression</strong> results (e.g.
            <a href="/gxa/experiments/E-MTAB-5214">GTEx</a>) to see the results in that particular
            experiment. Use the <strong>Gene query</strong> box (e.g. SFTPC in our example) to find
            <a href="/gxa/experiments/E-MTAB-5214/Results?specific=true&geneQuery=%255B%257B%2522value%2522%253A%2522SFTPC%2522%252C%2522category%2522%253A%2522symbol%2522%257D%255D&filterFactors=%257B%257D&cutoff=%257B%2522value%2522%253A0.5%257D&unit=%2522TPM%2522">
            in which tissues from GTEx gene SFTPC is expressed</a>.
        </p>


        <h3 id="multiGeneQuery">How do I search for multiple genes at once?</h3>
        <p>
            Use the <strong>Gene query</strong> search box on the <a href="/gxa">home page</a> to search for as many
            genes as you want. You will need to type each gene, click enter and when you are done, just search.
        </p>


        <h3 id="geneIDs">What gene identifiers can I use to search?</h3>
        <p>
            You may use the following identifiers to search using the <strong>Gene query</strong> box:
        </p>
        <ul>
            <li>
                Gene name symbol, e.g. <a href='/gxa/search?geneQuery=[{"value":"SFTPC"}]'>SFTPC</a>
            </li>
            <li>
                <a href="https://www.ensembl.org/index.html">Ensembl</a> gene ID, e.g.
                <a href="/gxa/genes/ENSG00000168484">ENSG00000168484</a>
            </li>
            <li>
                <a href="https://www.uniprot.org/">UniProt</a> ID, e.g.
                <a href="/gxa/genesets/A0A0S2Z4Q0">A0A0S2Z4Q0</a>
            </li>
            <li>
                <a href="https://www.ebi.ac.uk/interpro/">Interpro</a> ID, e.g.
                <a href="/gxa/genesets/IPR001729">IPR001729</a></li>
            <li>
                <a href="http://www.geneontology.org/">Gene Ontology</a> ID, e.g.
                <a href="/gxa/genesets/GO:0007585">GO:0007585</a>
            </li>
            <li>
                <a href="http://www.geneontology.org/">Gene Ontology</a> term, e.g.
                <a href='/gxa/search?geneQuery=[{"value":"respiratory gaseous exchange"}]'>respiratory gaseous
                exchange</a></li>
        </ul>


        <h3 id="exptLink">How do I find a particular experiment?</h3>
        <p>You can easily see all experiments in Expression Atlas by clicking in <strong>All experiments</strong> on
            the <a href="/gxa">home page</a>. You can narrow down the list of experiments by selecting Baseline or
            Differential in the first column at the bottom of the table. You can also select Plants or Animal and Fungi
            experiments (using the second column), experiments from a particular organism (fifth column) or experiments
            involving a particular variable (sixth column). Click on the Experiment title to see the experiment in
            Expression Atlas.
        </p>
        <p>
            If you know the <a href="https://www.ebi.ac.uk/arrayexpress">ArrayExpress</a> accession of the experiment
            you want to see (e.g. <a href="https://www.ebi.ac.uk/arrayexpress/experiments/E-MTAB-4202/">E-MTAB-4202
            </a>), you can link to the experiment in Expression Atlas using the following format:
            <code>https://www.ebi.ac.uk/gxa/experiments/<strong>&lt;ArrayExpress accession&gt;</strong></code>
        </p>
        <p>e.g. <a href="/gxa/experiments/E-MTAB-4202">https://www.ebi.ac.uk/gxa/experiments/E-MTAB-4202</a></p>


        <h3 id="api">Are there any alternative ways of searching in Atlas?</h3>
        <p>
            It's best if you contact us and we will be able to advise you on your specific use case.
        </p>
        <p>
            In the meantime, you can construct queries using URLs like the ones in the table below. Please be aware
            that the format of these URLs is subject to change. If your queries stop working, please check back here
            for the latest standard or email us at
            <a href="mailto:arrayexpress-atlas@ebi.ac.uk">arrayexpress-atlas@ebi.ac.uk</a>.
        </p>
        <table style="text-align: center;">
            <thead style="background-color: #E1EEE9;">
            <tr>
                <th style="text-align: center;">Query</th>
                <th style="text-align: center;">URL</th>
            </tr>
            </thead>
            <tbody>
            <tr>
                <td>In what conditions is <a href="/gxa/genes/ENSG00000066279">ASPM</a> differentially expressed?</td>
                <td>https://www.ebi.ac.uk/gxa/search?geneQuery=[{"value":"ASPM"}]#differential</td>
            </tr>
            <tr>
                <td>What genes are differentially expressed in cancer?</td>
                <td>https://www.ebi.ac.uk/gxa/search?conditionQuery=[{"value":"cancer"}]#differential</td>
            </tr>
            <tr>
                <td>Show me comparisons where <a href="https://www.ebi.ac.uk/interpro/entry/IPR007087">zinc finger</a> genes are differentially expressed in mice.</td>
                <td>https://www.ebi.ac.uk/gxa/search?geneQuery=[{"value":"zinc finger"}]&organism=Mus musculus#differential</td>
            </tr>
            </tbody>
        </table>



        <h2 ID="results">Results</h2>

        <h3 id="blResults">When I search in Expression Atlas, what do the Baseline Expression results show?</h3>
        <p>
            If you search for a particular gene (e.g. <strong>CFHR2</strong>), the <strong>Baseline Expression</strong>
            results will display all organisms and conditions in which CFHR2 is expressed above the default minimum
            expression level of 0.5 FPKM or 0.5 TPM.
        </p>
        <p>
            By default, we display expression data for different tissues, e.g
            <a href='/gxa/search?geneQuery=[{"value":"CFHR2", "category":"symbol"}]'>in which tissues is CFHR2
            expressed?</a> but you can also use the filters to find gene expression in other conditions, such as
            <a href="/gxa/genes/ENSMUSG00000033898?bs=%7B%22mus%20musculus%22%3A%5B%22CELL_TYPE%22%5D%7D&ds=%7B%22kingdom%22%3A%5B%22animals%22%5D%7D#baseline">
            in which mouse cell types is CFHR2 expressed?</a>
        </p>
        <p>
            If you search for a particular condition (e. g.
            <a href='/gxa/search?conditionQuery=[{"value":"liver"}]#baseline'>liver</a>), in the <strong>Baseline
            Expression</strong> results you will see all organisms and experiments matching your search.
        </p>


        <h3 id="blHeatmapColours">In Baseline expression results, how are baseline expression levels represented?</h3>
        <p>
            In <strong>Baseline expression</strong> results you will see one heatmap per species. Each heatmap shows,
            for each species, all conditions (columns) and all experiments (rows) that matched your search. Expression
            levels are displayed in the heatmap in five different colours:
        </p>
        <ol>
            <li>
                Grey box: expression level is below cutoff (0.5 FPKM or 0.5 TPM)
            </li>
            <li>
                Light blue box: expression level is low (between 0.5 to 10 FPKM or 0.5 to 10 TPM)
            </li>
            <li>
                Medium blue box: expression level is medium (between 11 to 1000 FPKM or 11 to 1000 TPM)
            </li>
            <li>
                Dark blue box: expression level is high (more than 1000 FPKM or more than 1000 TPM)
            </li>
            <li>
                White box: there is no data available
            </li>
        </ol>


        <h3 id="blExperimentPage">How can I see the results for each experiment in Baseline expression results?</h3>
        <p>
            Each Baseline expression experiment in Expression Atlas has its own <strong>Experiment page</strong>, e.g.
            <a href="/gxa/experiments/E-MTAB-4344">Strand-specific RNA-seq of 13 human tissues from Michael Snyder's
            lab for the ENCODE project</a> where you can see a heatmap showing the 50 most specifically expressed genes
            across all conditions studied.
        </p>
        <p>
            You can further refine the query by narrowing the search to a particular gene (e.g.
            <a href="/gxa/experiments/E-MTAB-4344/Results?specific=true&geneQuery=%255B%257B%2522value%2522%253A%2522Ctrb1%2522%252C%2522category%2522%253A%2522symbol%2522%257D%255D&filterFactors=%257B%257D&cutoff=%257B%2522value%2522%253A0.5%257D&unit=%2522TPM%2522">
            CTRB1</a>), or gene sets (e.g.
            <a href="/gxa/experiments/E-MTAB-4344/Results?specific=true&geneQuery=%255B%257B%2522value%2522%253A%2522CELA3A%2522%252C%2522category%2522%253A%2522symbol%2522%257D%252C%257B%2522value%2522%253A%2522CELA3B%2522%252C%2522category%2522%253A%2522symbol%2522%257D%252C%257B%2522value%2522%253A%2522CTRB1%2522%252C%2522category%2522%253A%2522symbol%2522%257D%252C%257B%2522value%2522%253A%2522CTRB2%2522%252C%2522category%2522%253A%2522symbol%2522%257D%252C%257B%2522value%2522%253A%2522PRSS1%2522%252C%2522category%2522%253A%2522symbol%2522%257D%252C%257B%2522value%2522%253A%2522PRSS2%2522%252C%2522category%2522%253A%2522symbol%2522%257D%255D&filterFactors=%257B%257D&cutoff=%257B%2522value%2522%253A0.5%257D&unit=%2522TPM%2522">
            CELA3A, CELA3B, CTRB1, CTRB2, PRSS1, PRSS2</a>), or by limiting which organism parts are searched over
            (e.g.
            <a href="/gxa/experiments/E-MTAB-4344/Results?specific=true&geneQuery=%255B%255D&filterFactors=%257B%2522ORGANISM_PART%2522%253A%255B%2522pancreas%2522%255D%257D&cutoff=%257B%2522value%2522%253A0.5%257D&unit=%2522TPM%2522">
            genes specifically expressed in pancreas</a>).
        </p>


        <h3 id="blHeatmapColoursExpPage">In the experiment page, how are baseline expression levels represented?</h3>
        <p>
            In the Experiment page, e.g. <a href="/gxa/experiments/E-MTAB-552">Transcription profiling by high
            throughput sequencing of different potato tissues (genotype RH89-039-16)</a>expression levels are
            represented in one heatmap that shows gene expression levels for the 50 most specifically expressed genes
            (rows) across all conditions studied in the experiment selected (columns).
        </p>
        <p>
            Expression levels are displayed in the heatmap by colour intensity, according to the gradient bar above the
            heatmap. Hover the mouse above a cell to show a tooltip with the numerical values corresponding to each
            colour.
        </p>


        <h3 id="blDownloadAnalytics">Can I download the baseline expression results?</h3>
        <p>
            Yes, click on the
            <img alt="Download All Results button" src="resources/images/help/downloadAllResults_btn.png" align="bottom">
            button above the heatmap to download the data corresponding to your query. For example, if you select
            'flower' in the <strong>Organism part</strong> box, use the <strong>Download all results</strong> button to
            download expression data for the
            <a href="/gxa/experiments/E-MTAB-552/Results?specific=true&geneQuery=%255B%255D&filterFactors=%257B%2522ORGANISM_PART%2522%253A%255B%2522flower%2522%252C%2522stamen%2522%255D%257D&cutoff=%257B%2522value%2522%253A0.5%257D&unit=%2522TPM%2522">
            subset of genes specifically expressed in flower</a> in tab-delimited format with no ordering.
        </p>
        <p>
            On the other hand, by clicking on the <i>Downloads tab</i> of the experiment page you will download
            expression data for all genes and all conditions studied in the experiment.</p>


        <h3 id="ensembl">Can I view the results in the Ensembl browser?</h3>
        <p>
            Yes, you can. From the heatmap of the Experiment page, e.g. <a href="/gxa/experiments/E-ERAD-475">Baseline
            expression from transcriptional profiling of zebrafish developmental stages</a> just select a gene (e.g.
            <strong>SNORD61</strong>) and a condition, developmental stage in that particular experiment (e.g.
            <strong>gastrula 50%-epiboly</strong>) and click on the
            <img alt="Ensembl Genome Browser" src="resources/images/help/ensemblGenomeBrowser_btn.png" align="bottom">
            button in the left of the heatmap. You will see
            <a href="https://www.ensembl.org/Danio_rerio/Location/View?g=ENSDARG00000083171;r=14:31520436-31520513;t=ENSDART00000116362;text=;time=1504707519">
            gene expression value for gene SNORD61 in developmental stage gastrula 50%-epiboly in the context of the
            genomic location of SNORD61</a>.
        </p>
        <p>
            For plant experiments, e.g. <a href="/gxa/experiments/E-GEOD-55482">Transcriptomes for hybrids (F1s)
            between 18 Arabidopsis thaliana parents of the Multiparent Advanced Generation Inter-Cross (MAGIC) genetic
            mapping resource</a> you can also use the
            <img alt="Gramene Genome Browser" src="resources/images/help/grameneGenomeBrowser_btn.png" align="bottom">
            button to see, for example,
            <a href="ensembl.gramene.org/Arabidopsis_thaliana/Location/View?g=AT2G27380;r=2:11713411-11715774;t=AT2G27380.1;time=1484746364">
            gene expression value for gene EPR1 in ecotype Sf-2 x Can-0 in the context of the genomic location of
            EPR1</a>.
        </p>
        <p>
            For experiments performed in <i>Caenorhabditis elegans</i> or in <i>Schistosoma mansoni</i> such as
            <a href="/gxa/experiments/E-MTAB-451">RNA-Seq of Schistosoma mansoni (flatworms) larva and adult
            individuals at different life-stages</a> you can see the results using the
            <img alt="WormBase Genome Browser" src="resources/images/help/wormbaseGenomeBrowser_btn.png" align="bottom">
            button.
        </p>


        <h3 id="similarExpression">How are similarly expressed genes computed?</h3>
        <p>
            Similarly expressed genes across conditions in baseline experiments (e.g. tissues, developmental stages)
            are shown where available. They are computed on per-experiment basis, for experiments with three or more
            conditions. The method for estimating them involves two steps. The first one involves k-means clustering
            (for all possible values of k, capped to 100) of the expression of each gene across (e.g.) tissues. The
            second step compares the clusters on a gene-to-gene basis and outputs a ranked list of genes with
            decreasing similarity of expression patterns for each gene. The top 50 similarly expressed genes can be
            explored on our interface. Lowly expressed genes are filtered out from the calculation. The method has been
            implemented within the Bioconductor package
            <a href="https://bioconductor.org/packages/devel/bioc/html/clusterSeq.html">ClusterSeq</a>.
        </p>


        <h3 id="deResults">When I search in Expression Atlas, what do the Differential Expression results show?</h3>
        <p>
            If you search for a particular gene (e.g. <strong>CFHR2</strong>), in the <strong>Differential
            expression</strong> results you will see
            <a href='/gxa/search?geneQuery=[{"value":"CFHR2"}]#differential'>all comparisons in all species in which
                CFHR2 is differentially expressed</a> (absolute value of log<sub>2</sub> fold-change &gt; 1 and
            adjusted p-value &lt; 0.05). Comparisons in which gene CFHR2 is differentially expressed are ordered so the
            one with the largest absolute value of log<sub>2</sub> fold-change is at the top. If gene CFHR2 has
            identical log<sub>2</sub> fold-change in several comparisons, then the one with the lower adjusted
            <i>p</i>-value goes first.
        </p>
        <p>
            If you search for a particular condition (e. g. <strong>liver</strong>), in the <strong>Differential
            expression</strong> results you will see
            <a href='/gxa/search?conditionQuery=[{"value":"liver"}]#differential'>all comparisons in all species that
                matched your search</a>. When several genes show differential expression you will see one row per each
            gene and comparison. Comparisons are ordered following the same criteria as explained before.
        </p>
        <p>
            In both cases, you can narrow down the results displayed using the filters in the left where you can select
            a particular species (e.g. Mus musculus), experimental variable (e.g. compound) to see which comparisons
            involve the treatment of liver with a compound in mouse.
        </p>
        <p>
            Click on the Comparison name (e.g.
            <a href="/gxa/experiments/E-MTAB-2445?geneQuery=ENSMUSG00000026839&queryFactorValues=g1_g5&specific=false">
            'CCl4; 1.6 gram per kilogram' at '24 hour' vs 'none' at '0 hour'</a>) to find out more information about a
            particular gene-comparison combination. In microarray experiments, some genes may be targeted by more than
            one probe set. The table in the <strong>Differential expression</strong> results shows the values for the
            probe set with the largest absolute log<sub>2</sub> fold-change. You can see details of all probe sets for
            a given gene on the Experiment page. For example,
            <a href="/gxa/experiments/E-MTAB-2445/Results?specific=true&geneQuery=%255B%257B%2522value%2522%253A%2522Car3%2522%252C%2522category%2522%253A%2522symbol%2522%257D%255D&filterFactors=%257B%257D&cutoff=%257B%2522foldChange%2522%253A1%252C%2522pValue%2522%253A0.05%257D&regulation=%2522UP_DOWN%2522">
            gene Car3 is represented by 4 probe sets</a> in the experiment
            <a href="/gxa/experiments/E-MTAB-2445">Transcriptional responses in liver upon acute CCl4 intoxication</a>.
        </p>


        <h3 id="designElement">What is a "design element"?</h3>
        <p>
            On microarray experiment pages, you will see the <strong>design element</strong> name alongside the gene
            name. A design element is also known as a probe or probe set. This is the oligonucleotide probe (or group
            thereof) on the microarray that targets that gene.
        </p>


        <h3 id="comparison">What is a "comparison"?</h3>
        <p>
            A <strong>comparison</strong> is where two groups of samples are compared in a differential expression
            experiment. An example of a comparison is <a href="/gxa/experiments/E-GEOD-31138/Results">'breast cancer'
            <i>vs.</i> 'normal'</a>.
        </p>
        <p>
            For each gene, the mean expression level of the <strong>test</strong> group (e.g. breast cancer) is
            compared with the mean expression level of the <strong>reference</strong> group (e.normal), and a
            statistical test is performed to decide whether the two means are significantly different.
        </p>


        <h3 id="deHeatmapColours">What do the red and blue colours mean in the Differential expression results?</h3>
        <p>
            Let's search for the condition
            <a href="/gxa/search?conditionQuery=%5B%7B%22value%22%3A%22breast%20cancer%22%7D%5D#differential">breast
            cancer in a differential context</a>. A red box indicates that the gene is <strong>up-regulated</strong> in
            the test condition while a blue box means that the gene is <strong>down-regulated</strong> in the test
            condition. The colour intensity of filled boxes in the table represents how large the log<sub>2</sub>
            fold-change is for each gene. The larger the log<sub>2</sub> fold-change, the more intense the red or blue
            colour.
        </p>
        <p>
            The two bars above the Differential expression results table show the red and blue colour intensities for
            the <strong>top 50</strong> genes shown on the page. The colour intensities represent the log<sub>2</sub>
            fold-changes for the genes shown.
        </p>


        <h3 id="displayLogFoldChanges">How do I see the actual log<sub>2</sub> fold-changes in the Differential expression results table?</h3>
        <p>
            Click on the
            <img alt="Display Log2 Fold-Change button" src="resources/images/help/displayLog2FC_btn.png" align="bottom">
            button to the top left corner of the table to see the numerical values of the log<sub>2</sub> fold-change
            per each gene in each comparison.
        </p>


        <h3 id="pvalue">Where do the <i>p</i>-values come from?</h3>
        <p>
            In a microarray experiment, each gene's mean expression level in the <strong>test</strong> group is
            compared with its mean expression level in the <strong>reference</strong> group using a moderated
            <i>t</i>-test. This is done using the
            <a href="https://www.bioconductor.org/packages/2.13/bioc/html/limma.html">limma</a> package from
            <a href="https://www.bioconductor.org/">Bioconductor</a>.
        </p>
        <p>
            In an RNA-seq experiment, each gene's mean expression level in the <strong>test</strong> group is compared
            with its mean expression level in the <strong>reference</strong> group using a conditioned test based on
            the negative binomial distribution, analogous to Fisher's exact test. This is done using the
            <a href="https://www.bioconductor.org/packages/2.13/bioc/html/DESeq.html">DESeq</a> package in
            <a href="https://www.bioconductor.org/">Bioconductor</a>.
        </p>
        <p>
            Because the same test is done on thousands of genes at once, <i>p</i>-values are adjusted for
            <strong>multiple testing</strong> using the <a href="https://www.jstor.org/stable/2346101">Benjamini and
            Hochberg (1995)</a> false discovery rate (FDR) correction. This is what is meant by <strong>adjusted
            <i>p</i>-value</strong>.
        </p>


        <h3 id="gseaPlots">How are the gene set enrichment plots created?</h3>
        <p>
            Gene set enrichment analysis is performed using the
            <a href="https://www.bioconductor.org/packages/release/bioc/html/piano.html">Piano</a> package from
            <a href="https://www.bioconductor.org/">Bioconductor</a>. For each comparison, enrichment of terms from
            <a href="http://www.geneontology.org/">GO</a>, <a href="https://www.ebi.ac.uk/interpro/">InterPro</a>, and
            <a href="http://www.reactome.org/">Reactome</a> is tested for within the set of differentially expressed
            genes, using a variation on <a href="https://en.wikipedia.org/wiki/Fisher%27s_exact_test">Fisher's exact
            test</a>.  Gene set enrichment plots are only shown when statistically significant enrichment of terms was
            detected. This means that for some experiments, the menu will not display plots for all three of the
            aforementioned resources.
        </p>


        <h3 id="deDownloadAnalytics">Can I download the differential expression results?</h3>
        <p>
            Yes, click on the
            <img alt="Download All Results button" src="resources/images/help/downloadAllResults_btn.png" align="bottom">
            button above the heatmap to download the data corresponding to your query. For example, in
            <a href="/gxa/experiments/E-GEOD-73131">Transcription profiling by array of pancreatic islets from Sgpp2
            knockout mice after high fat diet</a> if you select just one comparison, e.g. 'high fat diet' vs 'normal'
            in 'Sgpp2 knockout' from the <strong>Comparison</strong> box and click Apply, by clicking on the
            <strong>Download all results</strong> button you will download
            <a href="/gxa/experiments/E-GEOD-73131/Results?specific=true&geneQuery=%255B%255D&filterFactors=%257B%2522COMPARISON_NAME%2522%253A%255B%2522%27high%2520fat%2520diet%27%2520vs%2520%27normal%27%2520in%2520%27Sgpp2%2520knockout%27%2522%255D%257D&cutoff=%257B%2522foldChange%2522%253A1%252C%2522pValue%2522%253A0.05%257D&regulation=%2522UP_DOWN%2522">
            expression data for the subset of genes differentially expressed in that particular comparison</a> in
            tab-delimited format with no ordering.
        </p>
        <p>
            On the other hand, by clicking on the <i>Downloads</i> tab of the experiment page you will download
            expression data for all genes in all comparisons studied in the experiment.
        </p>

    </div>

    <!-- This part is for the navigation section at the right-hand side -->
    <div class="columns small-2">
        <h4>Jump to...</h4>
        <h5><a href="FAQ.html#general">General</a></h5>
        <ul class="menu vertical">
            <li><a href="FAQ.html#whatIsAtlas">What is Expression Atlas</a></li>
            <li><a href="FAQ.html#exptChosen">How are experiments chosen to be in Expression Atlas?</a></li>
            <li><a href="FAQ.html#getOrigData">How can I obtain the original data for an experiment?</a></li>
            <li><a href="FAQ.html#wholeExpt">It doesn't look like the whole of the original experiment is in Expression Atlas, why not?</a></li>
            <li><a href="FAQ.html#arrayQuality">How is microarray data quality controlled?</a></li>
            <li><a href="FAQ.html#rnaseqQuality">How is RNA-seq data quality controlled?</a></li>
            <li><a href="FAQ.html#microarrayAnalysis">How is microarray data analysed?</a></li>
            <li><a href="FAQ.html#rnaseqAnalysis">How is RNA-seq data analysed?</a></li>
            <li><a href="FAQ.html#rnaseqUnits">What are FPKM and TPM?</a></li>
            <li><a href="FAQ.html#analysisMethods">How do I see how an individual experiment was analysed?</a></li>
            <li><a href="FAQ.html#contact">How can I contact you?</a></li>
            <li><a href="FAQ.html#mailingList">How can I keep up with the latest Expression Atlas news?</a></li>
            <li><a href="FAQ.html#anatomogram">Can I use Expression Atlas anatomograms on my website?</a></li>

        </ul>
        <h5><a href="FAQ.html#searching">Searching</a></h5>
        <ul class="menu vertical">
            <li><a href="FAQ.html#conditionSearch">How do I find out what genes are expressed in a particular condition (e.g. in my favourite tissue, cell line, developmental stage)?</a></li>
            <li><a href="FAQ.html#multiConditionQuery">How do I search for multiple conditions at once?</a></li>
            <li><a href="FAQ.html#geneSearch">How do I find out in which conditions my favourite gene is expressed?</a></li>
            <li><a href="FAQ.html#multiGeneQuery">How do I search for multiple genes at once?</a></li>
            <li><a href="FAQ.html#geneIDs">What gene identifiers can I use to search?</a></li>
            <li><a href="FAQ.html#exptLink">How do I find a particular experiment?</a></li>
            <li><a href="FAQ.html#api">Can I search Expression Atlas programmatically?</a></li>
        </ul>
        <h5><a href="FAQ.html#results">Results</a></h5>
        <ul class="menu vertical">
            <li><a href="FAQ.html#blResults">When I search in Expression Atlas, what do the Baseline Expression results show?</a></li>
            <li><a href="FAQ.html#blHeatmapColours">In Baseline expression results, how are baseline expression levels represented?</a></li>
            <li><a href="FAQ.html#blExperimentPage">How can I see the results for each experiment in Baseline expression results?</a></li>
            <li><a href="FAQ.html#blHeatmapColoursExpPage">In the experiment page, how are baseline expression levels represented?</a></li>
            <li><a href="FAQ.html#blDownloadAnalytics">Can I download the baseline expression results?</a></li>
            <li><a href="FAQ.html#ensembl">Can I view the results in the Ensembl browser?</a></li>
            <li><a href="FAQ.html#similarExpression">How are similarly expressed genes computed?</a></li>

            <li><a href="FAQ.html#deResults">When I search in Expression Atlas, what do the Differential Expression results show?</a></li>
            <li><a href="FAQ.html#designElement">What is a &quot;design element&quot;?</a></li>
            <li><a href="FAQ.html#comparison">What is a &quot;comparison&quot;?</a></li>
            <li><a href="FAQ.html#deHeatmapColours">What do the red and blue colours mean in the Differential expression results?</a></li>
            <li><a href="FAQ.html#displayLogFoldChanges">How do I see the actual log<sub>2</sub> fold-changes in the Differential expression results table?</a></li>
            <li><a href="FAQ.html#pvalue">Where do the <i>p</i>-values come from?</a></li>
            <li><a href="FAQ.html#gseaPlots">How are the gene set enrichment plots created?</a></li>
            <li><a href="FAQ.html#deDownloadAnalytics">Can I download the differential expression results?</a></li>
        </ul>
    </div>
</div>

<script>
    document.addEventListener("DOMContentLoaded", function(event) {
        document.getElementById("local-nav-faq").className += ' active';
    });
</script>

