<div class="row">
    <div class="columns small-12">
        <h2>Release notes</h2>

        <div class="margin-bottom-xxlarge">
            <h3>
                <img src="/gxa/resources/images/release-notes/atlas-release.png" alt="Atlas release" style="height: 64px"/>
                30 August 2018
            </h3>

            <h4>Data statistics</h4>
            <ul>
                <li>
                    Ensembl <b>92</b> / Ensembl Genomes <b>39</b> / WormBase ParaSite <b>10</b> gene annotations and
                    array design probe set mappings.
                </li>
                <li>
                    This release contains <b>3,388</b> datasets (<strong>101,247</strong> assays), in particular:
                    <ul>
                        <li>
                            <b>691</b> RNA-seq and <b>10</b> proteomics datasets, of which <b>150</b> report
                            <a href="/gxa/baseline/experiments">baseline</a> gene expression, across a total of <b>40</b>
                            different organisms;
                        </li>
                        <li>
                            <b>791</b> datasets reporting expression in<a href="/gxa/plant/experiments">plants</a>;
                        </li>
                        <li>
                            <strong>3,238</strong> datasets studying samples in <strong>9,254</strong>
                            <a href="/gxa/help/index.html#differential-experiment">differential</a> comparisons across
                            <strong>45</strong> different organisms.
                        </li>
                    </ul>
                </li>
            </ul>
        </div>


        <div class="margin-bottom-xxlarge">
            <h3>
                <img src="/gxa/resources/images/release-notes/atlas-release.png" alt="Atlas release" style="height: 64px"/>
                4 April 2018
            </h3>

            <h4>Data statistics</h4>
            <ul>
                <li>
                    Ensembl <b>91</b> / Ensembl Genomes <b>38</b> / WormBase ParaSite <b>9</b> gene annotations and
                    array design probe set mappings.
                </li>
                <li>
                    This release contains <b>3,328</b> datasets (<b>119,472</b> assays), in particular:
                    <ul>
                        <li>
                            <b>653</b> RNA-seq and <b>12</b> proteomics datasets, of which <b>154</b> report
                            <a href="/gxa/baseline/experiments">baseline</a> gene expression, across a total of
                            <b>42</b> different organisms;
                        </li>
                        <li>
                            <b>780</b> datasets reporting expression in <a href="/gxa/plant/experiments">plants</a>;
                        </li>
                        <li>
                            <b>3,177</b> datasets studying samples in <b>9,007</b>
                            <a href="/gxa/help/index.html#differential-expression">differential</a> comparisons across
                            <b>46</b> different organisms.
                        </li>
                    </ul>
                </li>
            </ul>

            <h4>New experiments</h4>
            <ul>
                <li>New plant species
                    <ul>
                        <li>
                            <a href="/gxa/experiments/E-ENAD-2">RNA-seq of a E-type and a Z-type cultivar of sugar beet
                            at five different developmental stages (Beetroot)</a>
                        </li>
                        <li>
                            <a href="/gxa/experiments/E-GEOD-77637">RNA-seq of Brassica napus seeds at different
                            developmental stages (Rapeseed) </a>
                        </li>
                        <li>
                            <a href="/gxa/experiments/E-GEOD-84798">RNA-seq of Brassica napus in response to PA23 in
                            the presence or absence of Sclerotinia sclerotiorum (Rapeseed)</a>
                        </li>
                        <li>
                            <a href="/gxa/experiments/E-GEOD-62671">RNA-seq of Chlamydomonas reinhardtii under diel
                            conditions to identify cycling genes (Green algae)</a>
                        </li>
                        <li>
                            <a href="/gxa/experiments/E-ENAD-12">RNA-seq of Chlamydomonas reinhardtii response to
                            nitrogen deprivation in cht7 mutant plants (Green algae)</a>
                        </li>
                        <li>
                            <a href="/gxa/experiments/E-ENAD-8">RNA-seq of banana in vitro roots under mild osmotic
                            stress (Banana)</a>
                        </li>
                        <li>
                            <a href="/gxa/experiments/E-ENAD-11">RNA-seq of banana fruit development (Banana)</a>
                        </li>
                        <li>
                            <a href="/gxa/experiments/E-ENAD-3">RNA-seq of Prunus persica roots and leaves under
                            drought stress (Peach)</a></li>
                        <li>
                            <a href="/gxa/experiments/E-GEOD-36391">Transcriptome sequencing of Foxtail millet Setaria
                            italica for different tissues (Foxtail millet)</a>
                        </li>
                        <li>
                            <a href="/gxa/experiments/E-MTAB-5117">RNA-seq of two Trifolium pratense genotypes, Tripo
                            and Lasang, from early to late developmental stages of flower development (Red clover)</a>
                        </li>
                    </ul>
                </li>
            </ul>

            <ul>
                <li>New proteomics experiments
                    <ul>
                        <li>
                            <a href="/gxa/experiments/E-PROT-10">Deep Proteomics of Mouse Skeletal Muscle</a>
                        </li>
                        <li>
                            <a href="/gxa/experiments/E-PROT-7">Cell-Type-Resolved Quantitative Proteomics of Murine
                            Liver</a>
                        </li>
                        <li>
                            <a href="/gxa/experiments/E-PROT-11">Initial quantitative proteomic map of 28 mouse tissues
                            using the SILAC mouse</a>
                        </li>
                        <li>
                            <a href="/gxa/experiments/E-PROT-13">A tissue-specific atlas of mouse protein
                            phosphorylation and expression</a>
                        </li>
                        <li>
                            <a href="/gxa/experiments/E-PROT-15">Murine protein tissue atlas of heavily vascularized
                            organs and cells adjacent to the blood plasma</a>
                        </li>
                        <li>
                            <a href="/gxa/experiments/E-PROT-16">Atxn2-Knock-Out mice show branched chain amino acids
                            and fatty acids pathway alterations</a>
                        </li>
                        <li>
                            <a href="/gxa/experiments/E-PROT-17">mESC shotgun and positional proteomics based on deep
                            proteome sequence database (derived from RIBOseq data)</a>
                        </li>
                        <li>
                            <a href="/gxa/experiments/E-PROT-9">Fractionation-dependent improvements in proteome
                            resolution in the mouse hippocampus by IEF LC-MS/MS</a>
                        </li>
                    </ul>
                </li>
            </ul>

            <h4>New features</h4>
            <ul>
                <li>
                    Single gene results in an experiment page now include a new transcript expression visualisation.
                    Please read <a href="/gxa/help/index.html#transcripts">the help section on <i>Transcript
                    quantification</i></a> to know more.
                </li>
            </ul>
        </div>


        <div class="margin-bottom-xxlarge">
            <h3>
                <img src="/gxa/resources/images/release-notes/atlas-release.png" alt="Atlas release" style="height: 64px"/>
                31 October 2017
            </h3>

            <h4>Data statistics</h4>
            <ul>
                <li>
                    Ensembl <b>90</b> / Ensembl Genomes <b>37</b> / WormBase ParaSite <b>9</b> gene annotations and
                    array design probe set mappings.
                </li>
                <li>
                    This release contains <b>3,166</b> datasets (<b>118,209</b> assays), in particular:
                    <ul>
                        <li>
                            <b>565</b> RNA-seq and <b>4</b> proteomics datasets, of which <b>134</b> report
                            <a href="/gxa/baseline/experiments">baseline</a> gene expression, across a total of
                            <b>32</b> different organisms;
                        </li>
                        <li>
                            <b>734</b> datasets reporting expression in <a href="/gxa/plant/experiments">plants</a>;
                        </li>
                        <li>
                            <b>3,035</b> datasets studying samples in <b>8,488</b>
                            <a href="/gxa/help/index.html#differential-expression">differential</a> comparisons across
                            <b>32</b> different organisms.
                        </li>
                    </ul>
                </li>
            </ul>

            <h4>New experiments</h4>
            <ul>
                <li>
                    New cancer data set from the Sanger Institute
                    <a href="http://www.sanger.ac.uk/science/tools/gdsc-genomics-drug-sensitivity-cancer">Genomics of
                    Drug Sensitivity in Cancer (GDSC)</a>:
                    <ul>
                        <li>
                            <a href="/gxa/experiments/E-MTAB-3983">Sanger Genomics of Drug Sensitivity in Cancer
                            Project (GDSC) RNA-seq cancer cell line gene expression data</a>
                        </li>
                    </ul>
                </li>

                <li>
                    New non–small-cell lung cancer data:
                    <ul>
                        <li>
                            <a href="/gxa/experiments/E-GEOD-81089">RNA-seq of 199 non-small cell lung carcinoma
                                patients and 19 paired normal lung tissues</a>
                        </li>
                        <li>
                            <a href="/gxa/experiments/E-GEOD-60052">RNA-seq of primary tumors from 79 small cell lung
                            cancer patients and 7 normal lung tissue</a>
                        </li>
                    </ul>
                </li>
            </ul>

            <h4>New features</h4>
            <ul>
                <li>Improved search with special characters. You can now search for conditions like
                    <a href="/gxa/search?geneQuery=%5B%5D&organism=&conditionQuery=%5B%7B%22value%22%3A%222-(4-carboxyphenyl)-4%2C4%2C5%2C5-tetramethylimidazoline-1-oxyl-3-oxide%22%7D%5D">
                    2-(4-carboxyphenyl)-4,4,5,5-tetramethylimidazoline-1-oxyl-3-oxide</a>.
                </li>
            </ul>
        </div>


        <div class="margin-bottom-xxlarge">
            <h3>
                <img src="/gxa/resources/images/release-notes/atlas-release.png" alt="Atlas release" style="height: 64px"/>
                14 August 2017
            </h3>

            <h4>Data statistics</h4>
            <ul>
                <li>
                    Ensembl <b>89</b> / Ensembl Genomes <b>36</b> / WormBase ParaSite <b>9</b> gene annotations and
                    array design probe set mappings.
                </li>
                <li>
                    This release contains <b>3,126</b> datasets (<b>113,148</b> assays), in particular:
                    <ul>
                        <li>
                            <b>537</b> RNA-seq and <b>4</b> proteomics datasets, of which <b>131</b> report
                            <a href="/gxa/baseline/experiments">baseline</a> gene expression, across a total of
                            <b>32</b> different organisms;
                        </li>
                        <li>
                            <b>731</b> datasets reporting expression in <a href="/gxa/plant/experiments">plants</a>;
                        </li>
                        <li>
                            <b>2,998</b> datasets studying samples in <b>8,391</b>
                            <a href="/gxa/help/index.html#differential-expression">differential</a> comparisons across
                            <b>33</b> different organisms.
                        </li>
                    </ul>
                </li>
            </ul>

            <h4>New experiments</h4>
            <ul>
                <li>
                    First pig baseline RNA-seq data (<i>Sus scrofa</i>):
                    <ul>
                        <li>
                            <a href="/gxa/experiments/E-MTAB-5895">RNA-seq of different tissues from Sus scrofa</a>
                        </li>
                    </ul>
                </li>
            </ul>

            <h4>New features</h4>
            <ul>
                <li>
                    All RNA-Seq baseline experiments are displayed using TPMs by default where available. There is a
                    new radio button to switch to FPKMs if desired.
                    <ul>
                        <li>
                            Example: <a href="/gxa/experiments/E-MTAB-4101/Results">Open Targets RNA-seq analysis of
                            six cell lines in TPM</a> cf.
                            <a href="/gxa/experiments/E-MTAB-4101/Results?unit=%2522FPKM%2522">the same experiment in
                            FPKM</a>.
                        </li>
                    </ul>
                </li>
                <li>
                    You can download results from the Gene Set Enrichment results page by clicking on the <i>Download
                    results</i> button on the top right.
                </li>
                <li>
                    Now you can choose what data to download in the experiment page: the highest expressed genes as
                    displayed in the heatmap, or all results matching your query.
                    <ul>
                        <li>
                            Example: click on the down-pointing arrow in the <i>Download</i> button, located to the
                            right above the heatmap, to download e.g.
                            <a href="/gxa/experiments/E-MTAB-513/Results?specific=true&geneQuery=%255B%255D&filterFactors=%257B%257D&cutoff=%257B%2522value%2522%253A10%257D&unit=%2522TPM%2522">
                            all genes with expression level above 10 TPM in <i>Illumina Body Map</i>, <i>Download all
                            data</i> or only the top 50 genes sorted by expression level</a>, <i>Download table
                            content</i>.
                        </li>
                    </ul>
                </li>
            </ul>
        </div>


        <div class="margin-bottom-xxlarge">
            <h3>
                <img src="/gxa/resources/images/release-notes/atlas-release.png" alt="Atlas release" style="height: 64px"/>
                18 May 2017
            </h3>

            <h4>Data statistics</h4>
            <ul>
                <li>
                    Ensembl <b>88</b> / Ensembl Genomes <b>35</b> / WormBase ParaSite <b>9</b> gene annotations and
                    array design probe set mappings.
                </li>
                <li>
                    This release contains <b>3,116</b> datasets (<b>104,386</b> assays), in particular:
                    <ul>
                        <li>
                            <b>529</b> RNA-seq and <b>3</b> proteomics datasets, of which <b>127</b> report
                            <a href="/gxa/baseline/experiments">baseline</a> gene expression, across a total of
                            <b>31</b> different organisms;
                        </li>
                        <li>
                            <b>729</b> datasets reporting expression in <a href="/gxa/plant/experiments">plants</a>;
                        </li>
                        <li>
                            <b>2,992</b> datasets studying samples in <b>8,368</b>
                            <a href="/gxa/help/index.html#differential-expression">differential</a> comparisons across
                            <b>33</b> different organisms.
                        </li>
                    </ul>
                </li>
            </ul>

            <h4>New experiments</h4>
            <ul>
                <li>
                    New baseline RNA-seq data from the <a href="http://www.hipsci.org/">Human Induced Pluripotent Stem
                    Cells Initiative (HipSci)</a>:
                    <ul>
                        <li>
                            <a href="/gxa/experiments/E-MTAB-4748">RNA-seq of coding RNA in human fibroblasts,
                            Peripheral Blood Mononuclear Cells (PBMCs) and induced pluripotent stem cells (iPS cells)
                            </a>
                        </li>
                    </ul>
                </li>
                <li>
                    Updated version of Genotype-Tissue Expression (GTEx) Project with 18,736 assays:
                    <ul>
                        <li>
                            <a href="/gxa/experiments/E-MTAB-5214">RNA-seq from 52 human tissue samples from the
                            Genotype-Tissue Expression (GTEx) Project</a>
                        </li>
                    </ul>
                </li>
            </ul>

            <h4>New Home Page</h4>
            <ul>
                <li>
                    We have updated our <a href="https://www.ebi.ac.uk/gxa">home page</a> with more useful links to
                    species-specific experiments or other large scale projects.
                </li>
            </ul>

            <h4>On-the-fly gene set overlap analysis</h4>
            <ul>
                <li>
                    Our <a href="https://www.ebi.ac.uk/~rpetry/geteam/gsa/gsa_apispec.pdf">GeneSet Enrichment API</a>
                    now has a front page. Click on "Gene Set Enrichment" next to "Search" on our
                    <a href="https://www.ebi.ac.uk/gxa">home page</a> and input a set of gene identifiers.
                    <ul>
                        <li>
                            Example result:
                            <a href="genesetenrichment?query=AT1G48030%20AT1G53240%20AT2G17130%20AT2G20420%20AT2G44350%20AT2G47510%20AT3G09810%20AT3G15020%20AT3G17240%20AT3G27380%20AT3G55410%20AT3G60100%20AT4G26910%20AT4G35260%20AT4G35650%20AT4G35830%20AT5G03290%20AT5G08300%20AT5G23250%20AT5G40650%20AT5G50950%20AT5G55070%20AT5G65165%20AT5G65750%20AT5G66760">
                            Arabidopsis thaliana</a>
                        </li>
                    </ul>
                </li>
            </ul>

            <h4>New Experiment Page</h4>
            <ul>
                <li>
                    Our experiment pages have been re-organised and updated with a new style of heatmap. See
                    <a href="/gxa/help/index.html">help pages</a> for more details while exploring an example:
                    <ul>
                        <li>
                            <a href="/gxa/experiments/E-MTAB-5214">RNA-seq from 52 human tissue samples from the
                            Genotype-Tissue Expression (GTEx) Project</a>
                        </li>
                    </ul>
                </li>
            </ul>
        </div>

        <div class="margin-bottom-xxlarge">
            <h3>
                <img src="/gxa/resources/images/release-notes/atlas-release.png" alt="Atlas release" style="height: 64px"/>
                1 February 2017
            </h3>

            <h4>Data statistics</h4>
            <ul>
                <li>
                    Ensembl <b>87</b> / Ensembl Genomes <b>34</b> gene annotations and array design probe set mappings.
                </li>
                <li>
                    This release contains <b>3,063</b> datasets (<b>125,327</b> assays), in particular:
                    <ul>
                        <li>
                            <b>494</b> RNA-seq and <b>3</b> proteomics datasets, of which <b>122</b> report
                            <a href="/gxa/baseline/experiments">baseline</a> gene expression, across a total of
                            <b>30</b> different organisms;
                        </li>
                        <li>
                            <b>703</b> datasets reporting expression in <a href="/gxa/plant/experiments">plants</a>;
                        </li>
                        <li>
                            <b>2,941</b> datasets studying samples in <b>8,246</b>
                            <a href="/gxa/help/index.html#differential-expression">differential</a> comparisons across
                            <b>33</b> different organisms.
                        </li>
                    </ul>
                </li>
            </ul>

            <h4>New experiments</h4>
            <ul>
                <li>
                    New mouse embryo data from the Mouse Genetics Project/<a href="http://dmdd.org.uk">DMDD</a>
                    project:
                    <ul>
                        <li>
                            <a href="/gxa/experiments/E-MTAB-5224">Strand-specific RNA-seq of somite-staged whole
                            embryos comparing 11 homozygous mutants against wild type</a>
                        </li>
                    </ul>
                </li>
                <li>
                    New human baseline expression on prenatal brain tissues from the Human Developmental Biology
                    Resource <a href="http://www.hdbr.org">HDBR</a>:
                    <ul>
                        <li>
                            <a href="/gxa/experiments/E-MTAB-4840">RNA-seq from 23 time points and 21 different
                            embryonic brain tissues</a>
                        </li>
                    </ul>
                </li>
            </ul>

            <h4>New Features</h4>
            <ul>
                <li>
                    Data on cross-experiment, organism part heatmaps can now be <b>filtered</b> by
                    <b>anatomical systems</b> and <b>gene expression</b>. Explore example:
                </li>
                <ul>
                    <li><a href="/gxa/genes/ENSG00000172023">REG1B (human)</a></li>
                </ul>
            </ul>
        </div>


        <div class="margin-bottom-xxlarge">
            <h3>
                <img src="/gxa/resources/images/release-notes/atlas-release.png" alt="Atlas release" style="height: 64px"/>
                2 December 2016
            </h3>

            <h4>Data statistics</h4>
            <ul>
                <li>
                    Ensembl <b>86</b> / Ensembl Genomes <b>33</b> gene annotations and array design probe set mappings.
                </li>
                <li>
                    This release contains <b>3,025</b> datasets (<b>123,194</b> assays), in particular:
                    <ul>
                        <li>
                            <b>485</b> RNA-seq and <b>3</b> proteomics datasets, of which <b>121</b> report
                            <a href="/gxa/baseline/experiments">baseline</a> gene expression, across a total of
                            <b>30</b> different organisms;
                        </li>
                        <li>
                            <b>698</b> datasets reporting expression in <a href="/gxa/plant/experiments">plants</a>;
                        </li>
                        <li>
                            <b>2,904</b> datasets studying samples in <b>8,117</b>
                            <a href="/gxa/help/index.html#differential-expression">differential</a> comparisons across
                            <b>31</b> different organisms.
                        </li>
                    </ul>
                </li>
            </ul>

            <h4>New experiments</h4>
            <ul>
                <li>
                    New proteomics data from the <a href="http://www.hipsci.org/">Human Induced Pluripotent Stem Cells
                    Initiative (HipSci)</a>:
                    <ul>
                        <li>
                            <a href="/gxa/experiments/E-PROT-5">Baseline protein expression for 18 IPS cell lines</a>
                        </li>
                    </ul>
                </li>
                <li>
                    Updated version of Genotype-Tissue Expression (GTEx) Project with 9,347 assays:
                    <ul>
                        <li>
                            <a href="/gxa/experiments/E-MTAB-5214">RNA-seq from 52 human tissue samples from the
                            Genotype-Tissue Expression (GTEx) Project</a>
                        </li>
                    </ul>
                </li>
                <li>
                    First experiment on green monkey (Chlorocebus aethiops sabaeus):
                    <ul>
                        <li>
                            <a href="/gxa/experiments/E-MTAB-4404">RNA-seq of six tissues at different ages</a>
                        </li>
                    </ul>
                </li>
                <li>
                    New plant species:
                    <ul>
                        <li>
                            <a href="/gxa/experiments/E-GEOD-42891">Transcription profiling of seven tissues of
                            Brassica oleracea (wild cabbage)</a>
                        </li>
                        <li>
                            <a href="/gxa/experiments/E-MTAB-552">Transcription profiling of 15 tissues of Solanum
                            tuberosum (potato)</a>
                        </li>
                    </ul>
                </li>
            </ul>
        </div>


        <div class="margin-bottom-xxlarge">
            <h3>
                <img src="/gxa/resources/images/release-notes/atlas-release.png" alt="Atlas release" style="height: 64px"/>
                22 September 2016
            </h3>
            <h4>Data statistics</h4>

            <ul>
                <li>
                    Ensembl <b>85</b> / Ensembl Genomes <b>32</b> gene annotations and array design probe set mappings.
                </li>
                <li>
                    This release contains <b>2,968</b> datasets (<b>110,249</b> assays), in particular:
                    <ul>
                        <li>
                            <b>444</b> RNA-seq and <b>2</b> proteomics datasets, of which <b>108</b> report
                            <a href="/gxa/baseline/experiments">baseline</a> gene expression, across a total of
                            <b>25</b> different organisms;
                        </li>
                        <li>
                            <b>632</b> datasets reporting expression in <a href="/gxa/plant/experiments">plants</a>;
                        </li>
                        <li>
                            <b>2,860</b> datasets studying samples in <b>8,015</b>
                            <a href="/gxa/help/index.html#differential-expression">differential</a> comparisons across
                            <b>30</b> different organisms.
                        </li>
                    </ul>
                </li>
            </ul>

            <h4>New Features</h4>
            <ul>
                <li>
                    Data on heatmaps can now be <b>sorted</b> by <b>gene</b> and <b>protein expression rank</b>.
                    Explore example:
                    <ul>
                        <li>
                            <a href="/gxa/genes/ENSG00000172023">REG1B (human)</a>
                        </li>
                    </ul>
                </li>
                <li>
                    Faceted search is now available for keyword-based gene or sample property queries. Explore example
                    queries:
                    <ul>
                        <li>
                            <a href="/gxa/query?geneQuery=%5B%5D&organism=&conditionQuery=%5B%7B%22value%22%3A%22liver%22%7D%5D&bs=%7B%22homo%20sapiens%22%3A%5B%22ORGANISM_PART%22%5D%2C%22bos%20taurus%22%3A%5B%22ORGANISM_PART%22%5D%2C%22gallus%20gallus%22%3A%5B%22ORGANISM_PART%22%5D%2C%22macaca%20mulatta%22%3A%5B%22ORGANISM_PART%22%5D%2C%22monodelphis%20domestica%22%3A%5B%22ORGANISM_PART%22%5D%2C%22mus%20musculus%22%3A%5B%22ORGANISM_PART%22%5D%2C%22ovis%20aries%22%3A%5B%22ORGANISM_PART%22%5D%2C%22papio%20anubis%22%3A%5B%22ORGANISM_PART%22%5D%2C%22rattus%20norvegicus%22%3A%5B%22ORGANISM_PART%22%5D%2C%22xenopus%20tropicalis%22%3A%5B%22ORGANISM_PART%22%5D%7D&ds=%7B%22kingdom%22%3A%5B%22animals%22%5D%7D#baseline">
                            liver (any organism)</a>
                        </li>
                        <li>
                            <a href="/gxa/query?geneQuery=%5B%7B%22value%22%3A%22HSP18.1%22%7D%5D&organism=&conditionQuery=%5B%7B%22value%22%3A%22leaf%22%7D%5D&bs=%7B%22arabidopsis%20thaliana%22%3A%5B%22ORGANISM_PART%22%5D%2C%22oryza%20sativa%22%3A%5B%22ORGANISM_PART%22%5D%7D&ds=%7B%22kingdom%22%3A%5B%22plants%22%5D%7D#baseline">
                            HSP18.1 gene and leaf (any organism)</a>
                        </li>
                    </ul>
                </li>
            </ul>

            <h4>New experiments</h4>
            <ul>
                <li>
                    Mouse embryo data from the <a href="http://dmdd.org.uk">DMDD</a> project:
                    <ul>
                        <li>
                            <a href="/gxa/experiments/E-ERAD-401">Strand-specific RNA-seq of somite-staged second
                            generation genotypically wild-type embryos of mixed G0 lineage from the Mouse Genetics
                            Project/DMDD</a>
                        </li>
                        <li>
                            <a href="/gxa/experiments/E-ERAD-499">Strand-specific RNA-seq of somite-staged pure-bred
                            wild-type mouse embryos from the Mouse Genetics Project/ DMDD</a>
                        </li>
                    </ul>
                </li>
                <li>
                    Zebrafish embryo data from the
                    <a href="http://www.sanger.ac.uk/science/groups/vertebrate-genetics-and-genomics">Vertebrate
                    Genetics and Genomics Group</a> at the Wellcome Trust Sanger Institute:
                    <ul>
                        <li>
                            <a href="/gxa/experiments/E-ERAD-475">Baseline expression from transcriptional profiling of
                            zebrafish developmental stages</a>
                        </li>
                    </ul>
                </li>
            </ul>
        </div>


        <div class="margin-bottom-xxlarge">
            <h3>
                <img src="/gxa/resources/images/release-notes/atlas-release.png" alt="Atlas release" style="height: 64px"/>
                22 June 2016
            </h3>

            <h4>Data statistics</h4>
            <ul>
                <li>
                    Ensembl <b>84</b> / Ensembl Genomes <b>31</b> gene annotations and array design probe set mappings.
                </li>
                <li>
                    This release contains <b>2,914</b> datasets (<b>106,062</b> assays), in particular:
                    <ul>
                        <li>
                            <b>401</b> RNA-seq and <b>2</b> proteomics datasets, of which <b>96</b> report
                            <a href="/gxa/baseline/experiments">baseline</a> gene expression, across a total of
                            <b>23</b> different organisms;
                        </li>
                        <li>
                            <b>671</b> datasets reporting expression in <a href="/gxa/plant/experiments">plants</a>;
                        </li>
                        <li>
                            <b>2,818</b> datasets studying samples in <b>7,863</b>
                            <a href="/gxa/help/index.html#differential-expression">differential</a> comparisons across
                            <b>30</b> different organisms.
                        </li>
                    </ul>
                </li>
            </ul>

            <h4>New heatmap</h4>
            <ul>
                <li>
                    This release launches the new style of <b>heatmap</b> on the <b>gene/gene-set page</b>. The changes
                    include:
                    <ul>
                        <li>resizable, zoom-able display</li>
                        <li>assigning expression levels to <b>low</b>, <b>medium</b>, <b>high</b></li>
                    </ul>
                </li>
                <li>
                    Examples:
                    <ul>
                        <li>
                            <a href="/gxa/genes/ENSMUSG00000048756">Foxo3 (mouse)</a>
                        </li>
                        <li>
                            <a href="/gxa/genesets/GO:0007585">Respiratory gaseous exchange</a>
                        </li>
                    </ul>
                </li>
            </ul>

            <h4>New feature: similar expression patterns</h4>
            <ul>
                <li>
                    This release includes a new feature for exploring genes with similar expression patterns across
                    conditions in baseline experiments. Explore examples:
                    <ul>
                        <li>
                            In
                            <a href="/gxa/experiments/E-MTAB-4220?accessKey=&serializedFilterFactors=&queryFactorType=DEVELOPMENTAL_STAGE&rootContext=&heatmapMatrixSize=50&displayLevels=false&displayGeneDistribution=false&geneQuery=VIT_01s0010g00060&_queryFactorValues=1&specific=true&_specific=on&cutoff=0.5">
                                <i>Vitis vinifera</i> development</a>
                        </li>
                        <li>
                            In
                            <a href="/gxa/experiments/E-MTAB-2037?accessKey=&serializedFilterFactors=&queryFactorType=ORGANISM_PART&rootContext=&heatmapMatrixSize=50&displayLevels=false&displayGeneDistribution=false&geneQuery=EPlOSAG00000008047&_queryFactorValues=1&specific=true&_specific=on&cutoff=0.5">
                                <i>Oryza sativa</i> tissues</a>
                        </li>
                    </ul>
                </li>

                <li>
                    For more information please see <a href="FAQ.html#similarExpression">FAQ: How are similarly
                    expressed genes computed?</a>
                </li>
            </ul>

            <h4>New experiments</h4>
            <ul>
                <li>
                    This release of Expression Atlas includes datasets on <b>tomato</b> (<i>Solanum lycopersicum</i>),
                    a new species for us! We now have <b>four baseline experiments</b>, studying fruit development and
                    profiling various tissues:
                    <ul>
                        <li>
                            <a href="/gxa/experiments/E-MTAB-4765">Transcription profiling by high throughput
                            sequencing of the five principal tissues of the pericarp from tomato fruits at their
                            maximal growth phase</a>
                        </li>
                        <li>
                            <a href="/gxa/experiments/E-MTAB-4812">Transcription profiling by high throughput
                            sequencing of tomato root, leaf, flower (2 stages) and fruit (6 stages)</a>
                        </li>
                        <li>
                            <a href="/gxa/experiments/E-MTAB-4813">Transcription profiling by high throughput
                            sequencing along three longitudinal sections of six stages during tomato fruit
                            development</a>
                        </li>
                        <li>
                            <a href="/gxa/experiments/E-MTAB-4818">Transcription profiling by high throughput
                            sequencing of tomato fruit at four developmental stages</a>
                        </li>
                    </ul>
                </li>

                <li>
                    As well as the above baseline datasets, we can also showcase <b>seven differential expression
                    studies</b> on tomato:
                    <ul>
                        <li>
                            <a href="/gxa/experiments/E-MTAB-4789">Transcription profiling by high throughput
                            sequencing of tomato Rio Grande prf3 leaves challenged with either the flgII-28 peptide or
                            different bacterial strains</a>
                        </li>
                        <li>
                            <a href="/gxa/experiments/E-MTAB-4792">Transcription profiling by high throughput
                            sequencing of two tomato nearly isogenic lines that differ for the allele at fw3.2 in
                            Yellow Stuffer background</a>
                        </li>
                        <li>
                            <a href="/gxa/experiments/E-MTAB-4793">Transcription profiling by high throughput
                            sequencing of flower buds from two tomato nearly isogenic lines that differ for the allele
                            at fw3.2 in Yellow Stuffer background</a>
                        </li>
                        <li>
                            <a href="/gxa/experiments/E-MTAB-4796">Transcription profiling by high throughput
                            sequencing of leaves of resistant (RG-PtoR) and susceptible (RG-prf3 and RG-prf19) tomato
                            plants treated with Pseudomonas syringae pv. tomato str. DC3000</a>
                        </li>
                        <li>
                            <a href="/gxa/experiments/E-MTAB-4797">Transcription profiling by high throughput
                            sequencing of leaves of resistant (RG-PtoR) tomato plants treated with different Pst DC3000
                            mutant strains</a>
                        </li>
                        <li>
                            <a href="/gxa/experiments/E-MTAB-4819">Transcription profiling by high throughput
                            sequencing of immature green and red ripe tomato fruits from plants over-expressing SIGLK1
                            and SIGLK2</a>
                        </li>
                        <li>
                            <a href="/gxa/experiments/E-MTAB-4837">Transcription profiling by high throughput
                            sequencing of tomato SUN and wild type fruit tissues</a>
                        </li>
                    </ul>
                </li>

                <li>
                    The latest release also sees our first experiment studying baseline expression in different
                    <a href="/gxa/experiments/E-GEOD-38612">Arabidopsis tissues</a>, a
                    <a href="/gxa/experiments/E-MTAB-4754">new dataset on tonsil</a> from the
                    <a href="http://dcc.blueprint-epigenome.eu/">BLUEPRINT Epigenome project</a> as well as updates to
                    the previously-added BLUEPRINT <a href="/gxa/experiments/E-MTAB-3827">dataset on
                    haematopoiesis</a>, and <a href="/gxa/experiments/E-MTAB-4729">RNA-seq data</a> from
                    <a href="http://www.opentargets.org/">Open Targets</a> showing baseline expression of lung disease
                    cell line models.</li>
            </ul>

            <h4>Bioconductor package out now</h4>
            <ul>
                <li>
                    The <a href="http://www.bioconductor.org/packages/release/bioc/html/ExpressionAtlas.html">
                    Expression Atlas R package</a> is now available as of version 3.3 of
                    <a href="http://www.bioconductor.org/">Bioconductor</a>. Using this package you can search for
                    Atlas datasets and download the data from inside an R session or script. Please see the
                    <a href="http://www.bioconductor.org/packages/release/bioc/vignettes/ExpressionAtlas/inst/doc/ExpressionAtlas.html">
                    package vignette</a> for more information.
                </li>
            </ul>
        </div>


        <div class="margin-bottom-xxlarge">
            <h3>
                <img src="/gxa/resources/images/release-notes/atlas-release.png" alt="Atlas release" style="height: 64px"/>
                21 April 2016
            </h3>

            <h4>Data statistics</h4>
            <ul>
                <li>
                    Ensembl <b>84</b> / Ensembl Genomes <b>31</b> gene annotations and array design probe set mappings.
                </li>
                <li>
                    This release contains <b>2,738</b> datasets (<b>100,801</b> assays), in particular:
                    <ul>
                        <li>
                            <b>261</b> RNA-seq and <b>2</b> proteomics datasets, of which <b>72</b> report
                            <a href="/gxa/baseline/experiments">baseline</a> gene expression, across a total of
                            <b>22</b> different organisms;
                        </li>
                        <li>
                            <b>621</b> datasets reporting expression in <a href="/gxa/plant/experiments">plants</a>;
                        </li>
                        <li>
                            <b>2,666</b> datasets studying samples in <b>7,496</b>
                            <a href="/gxa/help/index.html#differential-expression">differential</a> comparisons across
                            <b>29</b> different organisms.
                        </li>
                    </ul>
                </li>
            </ul>
        </div>


        <div class="margin-bottom-xxlarge">
            <h3>
                <img src="/gxa/resources/images/release-notes/atlas-release.png" alt="Atlas release" style="height: 64px"/>
                4 April 2016
            </h3>

            <h4>Data statistics</h4>
            <ul>
                <li>
                    Ensembl <b>84</b> / Ensembl Genomes <b>30</b> gene annotations and array design probe set mappings.
                </li>
                <li>
                    This release contains <b>2,688</b> datasets (<b>99,162</b> assays), in particular:
                    <ul>
                        <li>
                            <b>224</b> RNA-seq and <b>2</b> proteomics datasets, of which <b>60</b> report
                            <a href="/gxa/baseline/experiments">baseline</a> gene expression, across a total of
                            <b>21</b> different organisms;
                        </li>
                        <li>
                            <b>602</b> datasets reporting expression in <a href="/gxa/plant/experiments">plants</a>;
                        </li>
                        <li>
                            <b>2,626</b> datasets studying samples in <b>7,411</b>
                            <a href="/gxa/help/index.html#differential-expression">differential</a> comparisons across
                            <b>28</b> different organisms.
                        </li>
                    </ul>
                </li>
            </ul>
        </div>


        <div class="margin-bottom-xxlarge">
            <h3>
                <img src="/gxa/resources/images/release-notes/atlas-release.png" alt="Atlas release" style="height: 64px"/>
                28 January 2016
            </h3>

            <h4>Data statistics</h4>
            <ul>
                <li>
                    Ensembl <b>83</b> / Ensembl Genomes <b>30</b> gene annotations and array design probe set mappings.
                </li>
                <li>
                    This release contains <b>2,620</b> datasets (<b>97,484</b> assays), in particular:
                    <ul>
                        <li>
                            <b>167</b> RNA-seq and <b>1</b> proteomics dataset, of which <b>47</b> report
                            <a href="/gxa/baseline/experiments">baseline</a> gene expression, across a total of
                            <b>19</b> different organisms;
                        </li>
                        <li>
                            <b>582</b> datasets reporting expression in <a href="/gxa/plant/experiments">plants</a>;
                        </li>
                        <li>
                            <b>2,573</b> datasets studying samples in <b>7,281</b>
                            <a href="/gxa/help/index.html#differential-expression">differential</a> comparisons across
                            <b>29</b> different organisms.
                        </li>
                    </ul>
                </li>
                <li>
                    <a href="https://www.r-project.org/">R</a> experiment summary objects for RNA-seq datasets now use
                    the new <i>RangedSummarizedExperiment</i> class from the
                    <a href="https://www.bioconductor.org/packages/release/bioc/html/SummarizedExperiment.html">
                    Summarized Experiment</a> <a href="https://www.bioconductor.org/">Bioconductor</a> package. See the
                    <a href="/gxa/help/r-data-objects.html">help page</a> for more details about how to use them.
                </li>
            </ul>
        </div>


        <div class="margin-bottom-xxlarge">
            <h3>
                <img src="/gxa/resources/images/release-notes/atlas-release.png" alt="Atlas release" style="height: 64px"/>
                17 December 2015
            </h3>

            <h4>Data statistics</h4>
            <ul>
                <li>
                    Ensembl <b>83</b> / Ensembl Genomes <b>29</b> gene annotations and array design probe set
                    mappings.
                </li>
                <li>
                    This release contains <b>2,574</b> datasets (<b>98,278</b> assays), in particular:
                    <ul>
                        <li>
                            <b>157</b> RNA-seq and <b>1</b> proteomics dataset, of which <b>46</b> report
                            <a href="/gxa/baseline/experiments">baseline</a> gene expression, across a total of
                            <b>19</b> different organisms;
                        </li>
                        <li>
                            <b>568</b> datasets reporting expression in <a href="/gxa/plant/experiments">plants</a>;
                        </li>
                        <li>
                            <b>2,527</b> datasets studying samples in <b>7,119</b>
                            <a href="/gxa/help/index.html#differential-expression">differential</a> comparisons across
                            <b>26</b> different organisms.
                        </li>
                    </ul>
                </li>
                <li>
                    This release launches the <b>new gene and gene-set interface</b>, available for species-specific
                    gene queries or gene-set identifier queries, via Gene Ontology (GO) or Reactome IDs, e.g.
                    <ul>
                        <li>
                            <a href="/gxa/genes/ENSMUSG00000048756">Foxo3 (mouse)</a>
                        </li>
                        <li>
                            <a href="/gxa/genesets/GO:0007585">Respiratory gaseous exchange</a>
                        </li>
                    </ul>
                    The new pages feature faceted search in <i>Differential expression</i> and <i>Baseline
                    Expression</i> tabs, including
                    <ul>
                        <li>
                            <b>Filtering differential expression</b> by Species, Experiment, Variables, up or down
                            regulation etc.
                        </li>
                        <li>
                            Ability to <b>show/hide baseline experiments</b> studying other conditions than just
                            tissues, e.g. cell lines, cell types, etc.
                        </li>
                        <li>
                            <b>Anatomograms can now be shown/hidden</b> as required.
                        </li>
                    </ul>
                    Baseline experiment types, except for cell lines, can be viewed on a heatmap.
                </li>
\            </ul>
        </div>


        <div class="margin-bottom-xxlarge">
            <h3>
                <img src="/gxa/resources/images/release-notes/atlas-release.png" alt="Atlas release" style="height: 64px"/>
                29 October 2015
            </h3>

            <h4>Data statistics</h4>
            <ul>
                <li>
                    Ensembl <b>82</b> / Ensembl Genomes <b>29</b> gene annotations and array design probe set mappings.
                </li>
                <li>
                    This release contains <b>2,373</b> datasets (<b>93,057</b> assays)
                </li>
                <li>
                    In particular, this release contains:
                    <ul>
                        <li>
                            <b>119</b> RNA-seq and <b>1</b> proteomics dataset, of which <b>43</b> report
                            <a href="/gxa/baseline/experiments">baseline</a> gene expression, across a total of
                            <b>17</b> different organisms;
                        </li>
                        <li>
                            <b>424</b> datasets reporting expression in <a href="/gxa/plant/experiments">plants</a>;
                        </li>
                        <li>
                            <b>2,330</b> datasets studying samples in <b>6,345</b>
                            <a href="/gxa/help/index.html#differential-expression">differential</a> comparisons across
                            <b>25</b> different organisms.
                        </li>
                    </ul>
                </li>
            </ul>


            <h3>
                <img src="/gxa/resources/images/release-notes/atlas-release.png" alt="Atlas release" style="height: 64px"/>
                14 October 2015
                <a href="https://www.facebook.com/ChocolatisimoShop">
                    <img src="/gxa/resources/images/release-notes/2000-experiments-cake.jpg"/>
                </a>
            </h3>

            <h4>Data statistics</h4>
            <ul>
                <li>
                    Ensembl <b>82</b> / Ensembl Genomes <b>28</b> gene annotations and array design probe set mappings.
                </li>
                <li>
                    This release contains <b>2,162</b> datasets (<b>86,980</b> assays), including:
                    <ul>
                        <li>
                            <a href="/gxa/experiments/E-MTAB-3871">19 Fetal Tissues from NIH Roadmap Epigenomics
                            Mapping Consortium (64 assays)</a>
                        </li>

                    </ul>
                </li>
                <li>
                    In particular, this release contains:
                    <ul>
                        <li>
                            <b>119</b> RNA-seq and <b>1</b> proteomics dataset, of which <b>43</b> report
                            <a href="/gxa/baseline/experiments">baseline</a> gene expression, across a total of
                            <b>17</b> different organisms;
                        </li>
                        <li>
                            <b>410</b> datasets reporting expression in <a href="/gxa/plant/experiments">plants</a>;
                        </li>
                        <li>
                            <b>2,119</b> datasets studying samples in <b>5,752</b>
                            <a href="/gxa/help/index.html#differential-expression">differential</a> comparisons across
                            <b>25</b> different organisms.
                        </li>
                    </ul>
                </li>
            </ul>
        </div>


        <div class="margin-bottom-xxlarge">
            <h3>
                <img src="/gxa/resources/images/release-notes/atlas-release.png" alt="Atlas release" style="height: 64px"/>
                14 September 2015
            </h3>

            <h4>Data statistics</h4>
            <ul>
                <li>
                    Ensembl <b>81</b> / Ensembl Genomes <b>28</b> gene annotations and array design probe set mappings.
                </li>
                <li>
                    This release contains <b>1,895</b> datasets (<b>81,255</b> assays), including:
                    <ul>
                        <li>
                            <a href="/gxa/experiments/E-PROT-1">Re-analysis of Pandey lab's draft of the human proteome
                            (30 assays)</a> - <b>our first protemics dataset!</b>
                        </li>
                        <li>
                            <a href="/gxa/experiments/E-MTAB-3838">RNA-seq of coding RNA from sheep embryonic, juvenile
                            and adult tissues (235 assays)</a>
                        </li>
                        <li>
                            <a href="/gxa/experiments/E-MTAB-3839">RNA-seq of various Sorghum bicolor (BTx623) tissues:
                            flowers, vegetative and floral meristems, embryos, roots and shoots (17 assays)</a>
                        </li>
                    </ul>
                </li>
                <li>
                    In particular, this release contains:
                    <ul>
                        <li>
                            <b>117</b> RNA-seq datasets, of which <b>42</b> of these report
                            <a href="/gxa/baseline/experiments">baseline</a> gene expression, across a total of
                            <b>15</b> different organisms;
                        </li>
                        <li>
                            <b>397</b> datasets reporting expression in <a href="/gxa/plant/experiments">plants</a>;
                        </li>
                        <li>
                            <b>1,853</b> datasets studying samples in <b>4,925</b>
                            <a href="/gxa/help/index.html#differential-expression">differential</a> comparisons across
                            <b>25</b> different organisms.
                        </li>
                    </ul>
                </li>
            </ul>
        </div>


        <div class="margin-bottom-xxlarge">
            <h3>
                <img src="/gxa/resources/images/release-notes/atlas-release.png" alt="Atlas release" style="height: 64px"/>
                11 August 2015
            </h3>

            <h4>Data statistics</h4>
            <ul>
                <li>
                    Ensembl <b>81</b> / Ensembl Genomes <b>28</b> gene annotations and array design probe set mappings.
                </li>
                <li>
                    This release contains <b>1,572</b> datasets (<b>69,239</b> assays), including:
                    <ul>
                        <li>
                            <a href="/gxa/experiments/E-MTAB-2770">RNA-seq of 934 human cancer cell lines from the
                            Cancer Cell Line Encyclopedia (30,088 assays)</a>
                        </li>
                    </ul>
                </li>
                <li>
                    In particular, this release contains:
                    <ul>
                        <li>
                            <b>96</b> RNA-seq, of which <b>38</b> of these report
                            <a href="/gxa/baseline/experiments">baseline</a> gene expression, across a total of
                            <b>15</b> different organisms;
                        </li>
                        <li>
                            <b>389</b> reporting expression in <a href="/gxa/plant/experiments">plants</a>;
                        </li>
                        <li>
                            <b>1,534</b> studying samples in <b>4,287</b>
                            <a href="/gxa/help/index.html#differential-expression">differential</a> comparisons across
                            <b>25</b> different organisms.
                        </li>
                        <li>
                            A much broader coverage of tissues in mouse anatomograms, c.f.
                            <a href="/gxa/genes/ENSMUSG00000022097">mouse SFTPA gene expression</a>
                        </li>
                    </ul>
                </li>
            </ul>
        </div>


        <div class="margin-bottom-xxlarge">
            <h3>
                <img src="/gxa/resources/images/release-notes/atlas-release.png" alt="Atlas release" style="height: 64px"/>
                2 July 2015
            </h3>

            <h4>Data statistics</h4>
            <ul>
                <li>
                    Ensembl <b>80</b> / Ensembl Genomes <b>27</b> gene annotations and array design probe set mappings.
                </li>
                <li>
                    This release contains <b>1,107</b> datasets (<b>24,048</b> assays), including:
                    <ul>
                        <li>
                            <a href="/gxa/experiments/E-MTAB-3579">RNA-Seq CAGE (Cap Analysis of Gene Expression)
                            analysis of mice tissues in RIKEN FANTOM5 project</a>
                        </li>
                    </ul>
                </li>
                <li>
                    In particular, this release contains:
                    <ul>
                        <li>
                            <b>68</b> RNA-seq, of which <b>30</b> of these report
                            <a href="/gxa/baseline/experiments">baseline</a> gene expression, across a total of
                            <b>16</b> different organisms;
                        </li>
                        <li>
                            <b>385</b> reporting expression in <a href="/gxa/plant/experiments">plants</a>;
                        </li>
                        <li>
                            <b>1,082</b> studying samples in <b>2,923</b>
                            <a href="/gxa/help/index.html#differential-expression">differential</a> comparisons across
                            <b>24</b> different organisms.
                        </li>
                        <li>
                            A much broader coverage of tissues in human anatomograms, c.f.
                            <a href="/gxa/genes/ENSG00000185303">human SFTPA2 gene expression</a>
                        </li>
                    </ul>
                </li>
            </ul>
        </div>


        <div class="margin-bottom-xxlarge">
            <h3>
                <img src="/gxa/resources/images/release-notes/atlas-release.png" alt="Atlas release" style="height: 64px"/>
                5 June 2015
                <a href="https://www.facebook.com/ChocolatisimoShop">
                    <img src="/gxa/resources/images/release-notes/1000-experiments-cake.jpeg"/>
                </a>
            </h3>

            <h4>Data statistics</h4>
            <ul>
                <li>
                    Ensembl <b>80</b> / Ensembl Genomes <b>26</b> gene annotations and array design probe set mappings.
                </li>
                <li>
                    We are proud to announce that release contains <b>1,051</b> datasets <i>(yes, we did
                    celebrate!)</i>, now including:
                    <ul>
                        <li>
                            <a href="/gxa/experiments/E-MTAB-2919">RNA-seq from 53 human tissue samples from the
                            Genotype-Tissue Expression (GTEx) Project</a>
                        </li>
                        <li>
                            <a href="/gxa/experiments/E-MTAB-2706">RNA-seq of 675 commonly used human cancer cell lines
                            (Genentech)</a>
                        </li>
                        <li>
                            <a href="/gxa/experiments/E-MTAB-3578">RNA-Seq CAGE (Cap Analysis of Gene Expression)
                            analysis of mice cells in RIKEN FANTOM5 project</a>
                        </li>
                    </ul>
                </li>
                <li>
                    In particular, this release contains:
                    <ul>
                        <li>
                            <b>67</b> RNA-seq, of which <b>29</b> of these report
                            <a href="/gxa/baseline/experiments">baseline</a> gene expression, across a total of
                            <b>16</b> different organisms;
                        </li>
                        <li>
                            <b>366</b> reporting expression in <a href="/gxa/plant/experiments">plants</a>;
                        </li>
                        <li>
                            <b>1,022</b> studying samples in <b>2,704</b>
                            <a href="/gxa/help/index.html#differential-expression">differential</a> comparisons across
                            <b>24</b> different organisms.
                        </li>
                    </ul>
                </li>
            </ul>
        </div>


        <div class="margin-bottom-xxlarge">
            <h3>
                <img src="/gxa/resources/images/release-notes/atlas-release.png" alt="Atlas release" style="height: 64px"/>
                31st March 2015
            </h3>

            <h4>Data statistics</h4>
            <ul>
                <li>
                    Ensembl <b>79</b> / Ensembl Genomes <b>26</b> gene annotations and array design probe set mappings.
                </li>
                <li>
                    This release contains <b>935</b> datasets, including:
                    <ul>
                        <li>
                            <b>56</b> RNA-seq, of which <b>26</b> of these report
                            <a href="/gxa/baseline/experiments">baseline</a> gene expression, across a total of
                            <b>16</b> different organisms;
                        </li>
                        <li>
                            <b>261</b> reporting expression in <a href="/gxa/plant/experiments">plants</a>;
                        </li>
                        <li>
                            <b>909</b> studying samples in <b>2,306</b>
                            <a href="/gxa/help/index.html#differential-expression">differential</a> comparisons across
                            <b>20</b> different organisms.
                        </li>
                    </ul>
                </li>
            </ul>
        </div>


        <div class="margin-bottom-xxlarge">
            <h3>
                <img src="/gxa/resources/images/release-notes/atlas-release.png" alt="Atlas release" style="height: 64px"/>
                20th March 2015
            </h3>

            <h4>Data statistics</h4>
            <ul>
                <li>
                    Ensembl <b>79</b> / Ensembl Genomes <b>25</b> gene annotations and array design probe set mappings.
                </li>
                <li>
                    This release contains <b>932</b> datasets, including:
                    <ul>
                        <li>
                            <b>56</b> RNA-seq, of which <b>26</b> of these report
                            <a href="/gxa/baseline/experiments">baseline</a> gene expression, across a total of
                            <b>16</b> different organisms;
                        </li>
                        <li>
                            <b>258</b> reporting expression in <a href="/gxa/plant/experiments">plants</a>;
                        </li>
                        <li>
                            <b>906</b> studying samples in <b>2,291</b>
                            <a href="/gxa/help/index.html#differential-expression">differential</a> comparisons across
                            <b>20</b> different organisms.
                        </li>
                    </ul>
                </li>
                <li>
                    New data sets of note:
                    <ul>
                        <li>
                            <a href="/gxa/experiments/E-MTAB-2980">RNA-seq of 39 human cancer cell lines that are in
                            the NCI-60 set from the Cancer Cell Line Encyclopedia</a>
                        </li>
                        <li>
                            <a href="/gxa/experiments/E-MTAB-3358">RNA-Seq CAGE (Cap Analysis of Gene Expression)
                            analysis of human tissues in RIKEN FANTOM5 project</a>
                        </li>
                    </ul>
                </li>
                <li>
                    Differential expression for RNA-seq data is now calculated using
                    <a href="http://www.bioconductor.org/packages/release/bioc/html/DESeq2.html"><b>DESeq2</b></a>
                    method.
                </li>
                <li>
                    Users searching for a single gene within a baseline experiment are now able to see how expression
                    varies across biological replicates for each condition - by selecting <b>Display variation</b>
                    option. This is currently a prototype - we welcome any feedback you may have. Please use the
                    feedback button on the top right hand side of all Atlas pages.
                    <p>
                        <img src="/gxa/resources/images/release-notes/baseline-boxplots.png" alt="Example of expression variation display"/>
                    </p>
                </li>
            </ul>
        </div>


        <div class="margin-bottom-xxlarge">
            <h3>
                <img src="/gxa/resources/images/release-notes/atlas-release.png" alt="Atlas release" style="height: 64px"/>
                27th January 2015
            </h3>

            <h4>Data statistics</h4>
            <ul>
                <li>
                    Ensembl <b>78</b> / Ensembl Genomes <b>25</b> gene annotations and array design probe set mappings.
                </li>
                <li>
                    This release contains <b>909</b> datasets, including <b>57</b> RNA-seq experiments. <b>24</b> of
                    these report <a href="/gxa/baseline/experiments">baseline</a> gene expression, across a total of
                    <b>16</b> different organisms. <b>885</b> datasets study samples in <b>2,226</b>
                    <a href="/gxa/help/index.html#differential-expression">differential</a> comparisons across
                    <b>20</b> different organisms.
                </li>
                <li>
                    <a href="/gxa/experiments/E-MTAB-2836">RNA-seq of coding RNA from tissue samples of 122 human
                    individuals representing 32 different tissues</a> data set is now available;
                </li>
            </ul>
        </div>


        <div class="margin-bottom-xxlarge">
            <h3>
                <img src="/gxa/resources/images/release-notes/atlas-release.png" alt="Atlas release" style="height: 64px"/>
                18th December 2014
            </h3>

            <h4>Data statistics</h4>
            <ul>
                <li>
                    Ensembl <b>78</b> / Ensembl Genomes <b>24</b> gene annotations and array design probe set mappings.
                </li>
                <li>
                    This release contains <b>873</b> datasets, including <b>50</b> RNA-seq experiments. <b>23</b> of
                    these report <a href="/gxa/baseline/experiments">baseline</a> gene expression, across a total of
                    <b>16</b> different organisms. <b>850</b> datasets study samples in <b>2,121</b>
                    <a href="/gxa/help/index.html#differential-expression">differential</a> comparisons across
                    <b>20</b> different organisms.
                </li>
                <li>
                    Please note that splice variant expression quantification is no longer available in Atlas. We have
                    also replaced <a href="http://cole-trapnell-lab.github.io/cufflinks/">Cufflinks2</a> with
                    <a href="http://www-huber.embl.de/users/anders/HTSeq/doc/overview.html">HTSeq</a> for gene
                    expression quantification.
                </li>
                <li>
                    By clicking on the link highlighted on the image below, the experiment data can now by filtered by
                    gene/protein keywords that were mined from the supporting publication by
                    <a href="http://europepmc.org/">Europe PubMed Central</a>. Click
                    <a href="/gxa/experiments/E-GEOD-49644">here</a> to try it out.
                    <p>
                        <img src="/gxa/resources/images/release-notes/filter-by-pub-genes.png" alt="Filter experiment by publication genes link"/>
                    </p>
                </li>
            </ul>
        </div>


        <div class="margin-bottom-xxlarge">
            <h3>
                <img src="/gxa/resources/images/release-notes/atlas-release.png" alt="Atlas release" style="height: 64px"/>
                19th November 2014
            </h3>

            <h4>Data statistics</h4>
            <ul>
                <li>
                    Ensembl <b>77</b> / Ensembl Genomes <b>24</b> gene annotations and array design probe set mappings.
                </li>
                <li>
                    This release contains <b>844</b> datasets, including <b>50</b> RNA-seq experiments. <b>22</b> of
                    these report <a href="/gxa/baseline/experiments">baseline</a> gene expression, across a total of
                    <b>16</b> different organisms. <b>822</b> datasets study samples in <b>2,032</b>
                    <a href="/gxa/help/index.html#differential-expression">differential</a> comparisons across
                    <b>20</b> different organisms.
                </li>
                <li>
                    Plant experiments, for example <a href="/gxa/experiments/E-MTAB-3028">our first RNA-seq maize data
                    set</a>, now allow users to display expression in the
                    <a href="http://ensembl.gramene.org/genome_browser/index.html">
                    <img src="/gxa/resources/images/release-notes/gramene-small.png" alt="Gramene" style="vertical-align:bottom;"/>
                    Genome Browser</a>.
                </li>
            </ul>
        </div>


        <div class="margin-bottom-xxlarge">
            <h3>
                <img src="/gxa/resources/images/release-notes/atlas-release.png" alt="Atlas release" style="height: 64px"/>
                13th October 2014
            </h3>

            <h4>Data statistics</h4>
            <ul>
                <li>
                    <a href="http://www.ensembl.org/index.html">Ensembl</a> <b>77</b> gene annotations and array design
                    probe set mappings.</li>
                <li>
                    This release contains <b>802</b> datasets, including <b>47</b> RNA-seq experiments. <b>21</b> of
                    these report <a href="/gxa/baseline/experiments">baseline</a> gene expression, across a total of
                    <b>15</b> different organisms. <b>781</b> datasets study samples in <b>1,937</b>
                    <a href="/gxa/help/index.html#differential-expression">differential</a> comparisons across
                    <b>20</b> different organisms.
                </li>
                <li>
                    <a href="/gxa/genes/ENSG00000163631">Gene pages</a> now show a combined <b>comparative view</b> for
                    gene expression across multiple <a href="/gxa/help/index.html#baseline-expression">baseline</a>
                    datasets.
                    <p>
                        <img src="/gxa/resources/images/release-notes/comparative-view.png" alt="Comparative baseline gene expression view"/>
                    </p>
                </li>
                <li>
                    <b>Hierarchical clustering</b> of the top 100 most variably expressed genes is now performed for
                    each baseline dataset.
                    <p>
                        <img src="/gxa/resources/images/release-notes/E-MTAB-2809-heatmap.png" alt="E-MTAB-2809 heatmap"/>
                    </p>
                </li>
                <li>
                    Three new anatomograms are now available, for
                    <a href="/gxa/experiments/E-GEOD-41338?serializedFilterFactors=ORGANISM:Anolis%20carolinensis">
                    <i>Anolis carolinensis</i></a>,
                    <a href="/gxa/experiments/E-GEOD-41338?serializedFilterFactors=ORGANISM:Anolis%20carolinensis">
                    <i>Xenopus tropicalis</i></a>
                    and
                    <a href="/gxa/experiments/E-GEOD-41338?serializedFilterFactors=ORGANISM:Tetraodon%20nigroviridis">
                    <i>Tetraodon nigroviridis</i></a></i>.
                    <p>
                        <img src="/gxa/resources/images/release-notes/new-anatomograms-2014-10.png" alt="Image of three new anatomograms"/>
                    </p>
                </li>
            </ul>
        </div>


        <div class="margin-bottom-xxlarge">
            <h3>
                <img src="/gxa/resources/images/release-notes/atlas-release.png" alt="Atlas release" style="height: 64px"/>
                22nd August 2014
            </h3>

            <h4>Data statistics</h4>
            <ul>
                <li>
                    Ensembl <b>76</b> / Ensembl Genomes <b>23</b> gene annotations and array design probe set mappings;
                </li>
                <li>
                    <b>709</b> experiments, including <b>39</b> RNA-seq experiments (<b>13</b>
                    <a href="/gxa/baseline/experiments">baseline</a>, <b>26</b> differential), studying samples in
                    <b>1,684</b> differential comparisons and <b>24</b> organisms (see below for breakdown of
                    comparisons per organism);
                </li>
                <li>
                    All Human RNA-seq experiments have been re-analysed against the new assembly (GRCh38), e.g.
                    <a href="/gxa/experiments/E-MTAB-1733">RNA-seq of coding RNA from tissue samples of 95 human
                    individuals representing 27 different tissues</a>;
                </li>
                <li>
                    Baseline expression levels available for genes and transcripts in 75 tissues, 21 cell lines, 7
                    cellular components, 39 developmental stages and 3 RNA types;
                </li>
                <li>
                    Quality control reports, e.g.
                    <a href="/gxa/experiments/E-MTAB-1733/fastqc/homo%20sapiens/qc.html">FASTQ QC</a> and
                    <a href="/gxa/experiments/E-MTAB-1733/fastqc/homo%20sapiens/mapping/tophat2.html">Mapping</a>, are
                    now available for all RNA-seq experiments;
                </li>
                <li>
                    <a href="http://www.ebi.ac.uk/efo/">EFO</a>-based autocomplete for experimental conditions is now
                    available on the main Atlas search page. Please see 'How to search' section of
                    <a href="/gxa/help/index.html">Atlas help page</a> for more details. <a href="/gxa/">The main
                    search interface</a> also now allows users to restrict search to a single organism;
                </li>
                <li>
                    User queries in the main Atlas search interface are expanded using the latest
                    <a href="http://www.ebi.ac.uk/efo/">EFO</a>. This means that if you search for e.g. "cancer", you
                    will also see results with terms such as "carcinoma", "malignant tumour", and other synonyms and
                    children of the <a href="http://www.ebi.ac.uk/efo/EFO_0000311">cancer</a> entry in EFO;
                </li>
                <li>
                    Gene set overlap analysis is performed using the Piano package from Bioconductor. In each
                    differential experiment, for each comparison, enrichment of terms from GO, InterPro, and Reactome
                    is tested for within the set of differentially expressed genes, using Fisher's exact test. Plots
                    showing maximum top ten GO, Interpro and Reactome terms enriched within the differentially
                    expressed genes for each comparison are now available for each differential experiment in Atlas.
                    Please see <a href="/gxa/help/index.html#differential-experiment-plots">Differential Atlas help</a>
                    for more information.
                </li>
                <li>
                    For <a href="/gxa/help/index.html#differential-expression">differential expression</a> experiments
                    you can now download a file containing the expression data and meta-data that you can load
                    into <a href="http://www.r-project.org/">R</a>. Please see
                    <a href="/gxa/help/r-data-objects.html">Expression Atlas data in R help</a> for more information;
                </li>
                <li>
                    Differential and Baseline Expression can now be visualised on Ensembl and Ensembl Genomes Genome
                    Browser. Please see <a href="FAQ.html#ensembl">“Can I view the results in the Ensembl browser?” in
                    our FAQ</a> for more information.
                </li>
            </ul>
            <p>
                <img src="/gxa/resources/images/release-notes/differential-species-2014-08.png" alt="Number of differential comparisons per organism"/>
            </p>
        </div>


        <div class="margin-bottom-xxlarge">
            <h3>
                <img src="/gxa/resources/images/release-notes/atlas-release.png" alt="Atlas release" style="height: 64px"/>
                10th April 2014
            </h3>

            <h4>Data statistics</h4>
            <ul>
                <li>
                    Ensembl <b>75</b> / Ensembl Genomes <b>22</b> gene annotations and array design probe set mappings;
                </li>
                <li>
                    <b>401</b> experiments, including <b>24</b> RNA-seq experiments (8 baseline, 16 differential),
                    studying samples in 19 species and 653 differential comparisons (see below for breakdown of
                    comparisons per experimental variable type);
                </li>
                <li>
                    Average number of citations per experiment: 42 (courtesy of
                    <a href="http://europepmc.org/">Europe PubMed Central)</a>;
                </li>
                <li>
                    Baseline expression levels available for genes and transcripts in 54 tissues, 21 cell lines, 5
                    cellular components, 5 developmental stages and 3 RNA types;
                </li>
                <li>
                    <a href="/gxa/experiments/E-MTAB-1733">RNA-seq of coding RNA from tissue samples of 95 human
                    individuals representing 27 different tissues</a> data set is now available;
                </li>
                <li>
                    Automatic quality control (QC) for Microarray experiments in operation; QC reports available to
                    users, e.g. <a href="/gxa/experiments/E-GEOD-10233/qc/A-AFFY-44/index.html">Transcription profiling
                    by array of human PMA-activated MM6 cells with cyclin T1 knockdown</a>
                </li>
            </ul>
            <p>
                <img src="/gxa/resources/images/release-notes/differential-factors-2014-04.png" alt="Number of differential comparisons per experimental variable type"/>
            </p>
        </div>


        <div class="margin-bottom-xxlarge">
            <h3>
                <img src="/gxa/resources/images/release-notes/atlas-release.png" alt="Atlas release" style="height: 64px"/>
                5th December 2013
            </h3>
            <p>
                Welcome to the new Expression Atlas service! The old Expression Atlas and its underlying data will no
                longer be updated or maintained, but limited support will be offered while the process of re-curation,
                re-analysis and transferring of experiments to the new Expression Atlas is on-going.
            </p>
            <p>
                We would like to thank our users for their continuous support, and encourage them to help us shape the
                new Expression Atlas by:
            </p>
                <ul>
                    <li>
                        providing us with feedback via <a href="https://www.ebi.ac.uk/support/gxa">the EBI Support &
                        feedback form</a>;
                    </li>
                    <li>
                        contacting us if any experiments of interest are included in the old Expression Atlas, but have
                        not made their way into the new Expression Atlas yet. We will then prioritise their transfer.
                    </li>
                </ul>
            <p>
                Enjoy the new Expression Atlas!
            </p>
        </div>
    </div>
</div>

<script>
    document.addEventListener("DOMContentLoaded", function(event) {
        document.getElementById("local-nav-release-notes").className += ' active';
    });
</script>
