<div class="row">
    <div class="columns small-12">
    <h2>Release notes</h2>

    <div class="margin-bottom-xxlarge">
        <!-- Beginning of main text section -->
        <h3><img src="resources/images/help/atlas_release_icon.png" alt="Atlas release" align="bottom" />14 August 2017</h3>

        <h4>Data statistics</h4>

        <ul>
            <li><p>Ensembl <strong>89</strong> / Ensembl Genomes <strong>36</strong> / WormBase ParaSite <strong>WBPS9</strong> gene annotations and array design probe set mappings.</p></li>
            <li>This release contains <strong>3,126</strong> datasets (<strong>113,148</strong> assays), in particular:
                <ul>
                    <li><strong>537</strong> RNA-seq and <strong>3</strong> proteomics
                        datasets, of which <strong>131</strong> report <a
                                href="baseline/experiments">baseline</a> gene
                        expression, across a total of <strong>32</strong> different
                        organisms;</li>
                    <li><strong>731</strong> datasets reporting expression in <a href="plant/experiments">plants</a>;</li>
                    <li><strong>2,998</strong> datasets studying samples in <strong>8,391</strong>
                        <a href="help/differential-atlas.html">differential</a> comparisons across <strong>33</strong> different organisms.</li>
                </ul>
            </li>
        </ul>

        <h4>New experiments</h4>

        <ul>
            <li>First pig baseline RNA-seq data (<i>Sus scrofa</i>):
                <ul>
                    <li><a href="experiments/E-MTAB-5895">RNA-seq of different tissues from Sus scrofa</a></li>
                </ul>
            </li>
            <li>Updated version of Human Induced Pluripotent Stem Cells Initiative (HipSci) with 397 samples:
                <ul>
                    <li><a href="experiments/E-MTAB-4748">RNA-seq of coding RNA in human fibroblasts, Peripheral Blood Mononuclear Cells (PBMCs) and induced pluripotent stem cells (iPS cells) as part of the HipSci project</a></li>
                </ul>
            </li>
        </ul>

        <h4>New features</h4>

        <ul>
            <li>All RNA-Seq baseline experiments are displayed using TPMs by default where available. There is a new radio button to switch to FPKMs if desired.</li>
                <ul>
                    <li>Example: see, for instance <a href="experiments/E-MTAB-4101/Results">Open Targets RNA-seq analysis of six cell lines in TPM</a> cf. <a href="experiments/E-MTAB-4101/Results?unit=%2522FPKM%2522">the same experiment in FPKM</a>.</li>
                </ul>
            <li>You can download results from the Gene Set Enrichment results page by clicking on the <em>Download results</em> button on the top right.</li>
            <li>Now you can choose what data to download in the experiment page: the highest expressed genes as displayed in the heatmap, or all results matching your query.</li>
                <ul>
                    <li>Example: click on the down-pointing arrow in the <em>Download</em> button, located to the right above the heatmap, to download e.g. <a href="experiments/E-MTAB-513/Results?specific=true&geneQuery=%255B%255D&filterFactors=%257B%257D&cutoff=%257B%2522value%2522%253A10%257D&unit=%2522TPM%2522">all genes with expression level above 10 TPM in <em>Illumina Body Map</em>, or only the top 50 genes sorted by expression level</a>.</li>
                </ul>
        </ul>

    </div>


    <div class="margin-bottom-xxlarge">
        <!-- Beginning of main text section -->
        <h3><img src="resources/images/help/atlas_release_icon.png" alt="Atlas release" align="bottom" />18 May 2017</h3>

        <h4>Data statistics</h4>

        <ul>
            <li><p>Ensembl <strong>88</strong> / Ensembl Genomes <strong>35</strong> / WormBase ParaSite <strong>WBPS9</strong> gene annotations and array design probe set mappings.</p></li>
            <li>This release contains <strong>3,116</strong> datasets (<strong>104,386</strong> assays), in particular:
                <ul>
                    <li><strong>529</strong> RNA-seq and <strong>3</strong> proteomics
                        datasets, of which <strong>127</strong> report <a
                                href="https://www.ebi.ac.uk/gxa/baseline/experiments">baseline</a> gene
                        expression, across a total of <strong>31</strong> different
                        organisms;</li>
                    <li><strong>729</strong> datasets reporting expression in <a href="https://www.ebi.ac.uk/gxa/plant/experiments">plants</a>;</li>
                    <li><strong>2,992</strong> datasets studying samples in <strong>8,368</strong>
                        <a href="https://www.ebi.ac.uk/gxa/help/differential-atlas.html">differential</a> comparisons across <strong>33</strong> different organisms.</li>
                </ul>
            </li>
        </ul>

        <h4>New experiments</h4>

        <ul>
            <li>New baseline RNA-seq data from the <a href="http://www.hipsci.org/">Human Induced Pluripotent Stem Cells Initiative (HipSci)</a>:
                <ul>
                    <li><a href="https://www.ebi.ac.uk/gxa/experiments/E-MTAB-4748">RNA-seq of coding RNA in human fibroblasts, Peripheral Blood Mononuclear Cells (PBMCs) and induced pluripotent stem cells (iPS cells)</a></li>
                </ul>
            </li>
            <li>Updated version of Genotype-Tissue Expression (GTEx) Project with 18,736 samples:
                <ul>
                    <li><a href="https://www.ebi.ac.uk/gxa/experiments/E-MTAB-5214">RNA-seq from 52 human tissue samples from the Genotype-Tissue Expression (GTEx) Project</a></li>
                </ul>
            </li>
        </ul>

        <h4>New Home Page</h4>

        <ul>
            <li>We have updated our <a href="https://www.ebi.ac.uk/gxa">home page</a> with more useful links to species-specific experiments or other large scale projects.</li>
        </ul>

        <h4>On-the-fly Gene set overlap analysis</h4>

        <ul>
            <li>Our <a href="https://www.ebi.ac.uk/~rpetry/geteam/gsa/gsa_apispec.pdf">GeneSet Enrichment API</a> now has a front page. Click on "Gene Set Enrichment" next to "Search" on our <a href="https://www.ebi.ac.uk/gxa">home page</a> and input a set of gene identifiers
                <ul>
                    <li>Example result: <a href="https://www.ebi.ac.uk/gxa/genesetenrichment?query=AT1G48030%20AT1G53240%20AT2G17130%20AT2G20420%20AT2G44350%20AT2G47510%20AT3G09810%20AT3G15020%20AT3G17240%20AT3G27380%20AT3G55410%20AT3G60100%20AT4G26910%20AT4G35260%20AT4G35650%20AT4G35830%20AT5G03290%20AT5G08300%20AT5G23250%20AT5G40650%20AT5G50950%20AT5G55070%20AT5G65165%20AT5G65750%20AT5G66760">Arabidopsis thaliana</a></li>
                </ul>
            </li>
        </ul>

        <h4>New Experiment Page</h4>

        <ul>
            <li>Our experiment pages have been re-organised and updated with a new style of heatmap. See <a href="http://www.ebi.ac.uk/gxa/help/index.html">help pages</a> for more details while exploring an example:
                <ul>
                    <li><a href="https://www.ebi.ac.uk/gxa/experiments/E-MTAB-5214">RNA-seq from 52 human tissue samples from the Genotype-Tissue Expression (GTEx) Project</a></li>
                </ul>
            </li>
        </ul>
    </div>


    <h3><img src="resources/images/help/atlas_release_icon.png" alt="Atlas release" align="bottom" />1 February 2017
    </h3>

    <h4>Data statistics</h4>

    <ul>
        <li><p>Ensembl <strong>87</strong> / Ensembl Genomes <strong>34</strong> gene annotations and array design probe set mappings.</p></li>
        <li>This release contains <strong>3,063</strong> datasets (<strong>125,327</strong> assays), in particular:
        </li>
        <ul>
            <li><strong>494</strong> RNA-seq and <strong>3</strong> proteomics
                datasets, of which <strong>122</strong> report <a
                        href="https://www.ebi.ac.uk/gxa/baseline/experiments">baseline</a> gene
                expression, across a total of <strong>30</strong> different
                organisms;</li>
            <li><strong>703</strong> datasets reporting expression in <a href="https://www.ebi.ac.uk/gxa/plant/experiments">plants</a>;</li>
            <li><strong>2,941</strong> datasets studying samples in <strong>8,246</strong>
                <a href="https://www.ebi.ac.uk/gxa/help/differential-atlas.html">differential</a> comparisons across <strong>33</strong> different organisms.</li>
        </ul>
    </ul>

    <h4>New experiments</h4>

    <ul>
        <li>New mouse embryo data from the Mouse Genetics Project/<a href="http://dmdd.org.uk">DMDD</a> project:
            <ul>
                <li><a href="https://www.ebi.ac.uk/gxa/experiments/E-MTAB-5224">Strand-specific RNA-seq of somite-staged whole embryos comparing 11 homozygous mutants against wild type</a></li>
            </ul>
        </li>
        <li>New human baseline expression on prenatal brain tissues from the Human Developmental Biology Resource <a href="http://www.hdbr.org">HDBR</a>:
            <ul>
                <li><a href="https://www.ebi.ac.uk/gxa/experiments/E-MTAB-4840">RNA-seq from 23 time points and 21 different embryonic brain tissues</a></li>
            </ul>
        </li>
    </ul>

    <h4>New Features</h4>

    <ul>
        <li>Data on cross-experiment, organism part heatmaps can now be <strong>filtered</strong> by <strong>anatomical systems</strong> and <strong>gene expression</strong>. Explore example:</li>
        <ul>
            <li><a href="https://www.ebi.ac.uk/gxa/genes/ENSG00000172023">REG1B (human)</a></li>
        </ul>
    </ul>



        <h3><img src="resources/images/help/atlas_release_icon.png" alt="Atlas release" align="bottom" />2 December 2016
    </h3>

    <h4>Data statistics</h4>

    <ul>
        <li><p>Ensembl <strong>86</strong> / Ensembl Genomes <strong>33</strong> gene annotations and array design probe set mappings.</p></li>
        <li>This release contains <strong>3,025</strong> datasets (<strong>123,194</strong> assays), in particular:
        </li>
        <ul>
            <li><strong>485</strong> RNA-seq and <strong>3</strong> proteomics
                datasets, of which <strong>121</strong> report <a
                        href="https://www.ebi.ac.uk/gxa/baseline/experiments">baseline</a> gene
                expression, across a total of <strong>30</strong> different
                organisms;</li>
            <li><strong>698</strong> datasets reporting expression in <a href="https://www.ebi.ac.uk/gxa/plant/experiments">plants</a>;</li>
            <li><strong>2,904</strong> datasets studying samples in <strong>8,117</strong>
                <a href="https://www.ebi.ac.uk/gxa/help/differential-atlas.html">differential</a> comparisons across <strong>31</strong> different organisms.</li>
        </ul>
    </ul>

    <h4>New experiments</h4>

    <ul>
        <li>New proteomics data from the <a href="http://www.hipsci.org/">Human Induced Pluripotent Stem Cells Initiative (HipSci)</a>:
            <ul>
                <li><a href="https://www.ebi.ac.uk/gxa/experiments/E-PROT-5">Baseline protein expression for 18 IPS cell lines</a></li>
            </ul>
        </li>
        <li>Updated version of Genotype-Tissue Expression (GTEx) Project with 9,347 samples:
            <ul>
                <li><a href="https://www.ebi.ac.uk/gxa/experiments/E-MTAB-5214">RNA-seq from 52 human tissue samples from the Genotype-Tissue Expression (GTEx) Project</a></li>
            </ul>
        </li>
        <li>First experiment on green monkey (Chlorocebus aethiops sabaeus):
            <ul>
                <li><a href="https://www.ebi.ac.uk/gxa/experiments/E-MTAB-4404">RNA-seq of six tissues at different ages</a></li>
            </ul>
        </li>
        <li>New plant species:
            <ul>
                <li><a href="https://www.ebi.ac.uk/gxa/experiments/E-GEOD-42891">Transcription profiling of seven tissues of Brassica oleracea (wild cabbage)</a></li>
                <li><a href="https://www.ebi.ac.uk/gxa/experiments/E-MTAB-552">Transcription profiling of 15 tissues of Solanum tuberosum (potato)</a></li>
            </ul>
        </li>
    </ul>

    <h3><img src="resources/images/help/atlas_release_icon.png" alt="Atlas release" align="bottom" />22 September 2016
    </h3>

    <h4>Data statistics</h4>

    <ul>
        <li><p>Ensembl <strong>85</strong> / Ensembl Genomes <strong>32</strong> gene annotations and array design probe set mappings.</p></li>
        <li>This release contains <strong>2,968</strong> datasets (<strong>110,249</strong> assays), in particular:
        </li>
        <ul>
            <li><strong>444</strong> RNA-seq and <strong>2</strong> proteomics
                datasets, of which <strong>108</strong> report <a
                        href="https://www.ebi.ac.uk/gxa/baseline/experiments">baseline</a> gene
                expression, across a total of <strong>25</strong> different
                organisms;</li>
            <li><strong>632</strong> datasets reporting expression in <a href="https://www.ebi.ac.uk/gxa/plant/experiments">plants</a>;</li>
            <li><strong>2,860</strong> datasets studying samples in <strong>8,015</strong>
                <a href="https://www.ebi.ac.uk/gxa/help/differential-atlas.html">differential</a> comparisons across <strong>30</strong> different organisms.</li>
        </ul>
    </ul>

    <h4>New Features</h4>

    <ul>
        <li>Data on heatmaps can now be <strong>sorted</strong> by <strong>gene</strong> and <strong>protein expression rank</strong>. Explore example:</li>
        <ul>
            <li><a href="https://www.ebi.ac.uk/gxa/genes/ENSG00000172023">REG1B (human)</a></li>
        </ul>
        <li>Faceted search is now available for keyword-based gene or sample property queries. Explore example queries:</li>
        <ul>
            <li><a href="https://www.ebi.ac.uk/gxa/query?geneQuery=%5B%5D&organism=&conditionQuery=%5B%7B%22value%22%3A%22liver%22%7D%5D&bs=%7B%22homo%20sapiens%22%3A%5B%22ORGANISM_PART%22%5D%2C%22bos%20taurus%22%3A%5B%22ORGANISM_PART%22%5D%2C%22gallus%20gallus%22%3A%5B%22ORGANISM_PART%22%5D%2C%22macaca%20mulatta%22%3A%5B%22ORGANISM_PART%22%5D%2C%22monodelphis%20domestica%22%3A%5B%22ORGANISM_PART%22%5D%2C%22mus%20musculus%22%3A%5B%22ORGANISM_PART%22%5D%2C%22ovis%20aries%22%3A%5B%22ORGANISM_PART%22%5D%2C%22papio%20anubis%22%3A%5B%22ORGANISM_PART%22%5D%2C%22rattus%20norvegicus%22%3A%5B%22ORGANISM_PART%22%5D%2C%22xenopus%20tropicalis%22%3A%5B%22ORGANISM_PART%22%5D%7D&ds=%7B%22kingdom%22%3A%5B%22animals%22%5D%7D#baseline">liver (any organism)</a></li>
            <li><a href="https://www.ebi.ac.uk/gxa/query?geneQuery=%5B%7B%22value%22%3A%22HSP18.1%22%7D%5D&organism=&conditionQuery=%5B%7B%22value%22%3A%22leaf%22%7D%5D&bs=%7B%22arabidopsis%20thaliana%22%3A%5B%22ORGANISM_PART%22%5D%2C%22oryza%20sativa%22%3A%5B%22ORGANISM_PART%22%5D%7D&ds=%7B%22kingdom%22%3A%5B%22plants%22%5D%7D#baseline">HSP18.1 gene and leaf (any organism)</a></li>
        </ul>
    </ul>

    <h4>New experiments</h4>

    <ul>
        <li>Mouse embryo data from the <a href="http://dmdd.org.uk">DMDD</a> project:
            <ul>
                <li><a href="https://www.ebi.ac.uk/gxa/experiments/E-ERAD-401">Strand-specific RNA-seq of somite-staged second generation genotypically wild-type embryos of mixed G0 lineage from the Mouse Genetics Project/DMDD</a></li>
                <li><a href="https://www.ebi.ac.uk/gxa/experiments/E-ERAD-499">Strand-specific RNA-seq of somite-staged pure-bred wild-type mouse embryos from the Mouse Genetics Project/ DMDD</a></li>
            </ul>
        </li>
        <li>Zebrafish embryo data from the <a href="http://www.sanger.ac.uk/science/groups/vertebrate-genetics-and-genomics">Vertebrate Genetics and Genomics Group</a> at the Wellcome Trust Sanger Institute:
            <ul>
                <li><a href="https://www.ebi.ac.uk/gxa/experiments/E-ERAD-475">Baseline expression from transcriptional profiling of zebrafish developmental stages</a></li>
            </ul>
        </li>
    </ul>

    <h3><img src="resources/images/help/atlas_release_icon.png" alt="Atlas release" align="bottom" />22 June 2016</h3>

    <h4>Data statistics</h4>

    <ul>
        <li><p>Ensembl <strong>84</strong> / Ensembl Genomes <strong>31</strong> gene annotations and array design probe set mappings.</p></li>
        <li>This release contains <strong>2,914</strong> datasets (<strong>106,062</strong> assays), in particular:
        </li>
        <ul>
            <li><strong>401</strong> RNA-seq and <strong>2</strong> proteomics
                datasets, of which <strong>96</strong> report <a
                        href="https://www.ebi.ac.uk/gxa/baseline/experiments">baseline</a> gene
                expression, across a total of <strong>23</strong> different
                organisms;</li>
            <li><strong>671</strong> datasets reporting expression in <a href="https://www.ebi.ac.uk/gxa/plant/experiments">plants</a>;</li>
            <li><strong>2,818</strong> datasets studying samples in <strong>7,863</strong>
                <a href="https://www.ebi.ac.uk/gxa/help/differential-atlas.html">differential</a> comparisons across <strong>30</strong> different organisms.</li>
        </ul>
    </ul>

    <h4>New heatmap</h4>

    <ul>
        <li>This release launches the new style of <strong>heatmap</strong> on the <strong>gene/gene-set page</strong>. The changes include:</li>
        <ul>
            <li>resize-able, zoom-able display</li>
            <li>assigning expression levels to <strong>low</strong>, <strong>medium</strong>, <strong>high</strong></li>
        </ul>
        <li>Examples:</li>
        <ul>
            <li><a href="https://www.ebi.ac.uk/gxa/genes/ENSMUSG00000048756">Foxo3 (mouse)</a></li>
            <li><a href="https://www.ebi.ac.uk/gxa/genesets/GO:0007585">Respiratory gaseous exchange</a></li>
        </ul>
    </ul>

    <h4>New feature: similar expression patterns</h4>

    <ul>
        <li>This release includes a new feature for exploring genes with similar expression patterns across conditions in baseline experiments. Explore examples:</li>
        <ul>
            <li>In <a href="https://www.ebi.ac.uk/gxa/experiments/E-MTAB-4220?accessKey=&serializedFilterFactors=&queryFactorType=DEVELOPMENTAL_STAGE&rootContext=&heatmapMatrixSize=50&displayLevels=false&displayGeneDistribution=false&geneQuery=VIT_01s0010g00060&_queryFactorValues=1&specific=true&_specific=on&cutoff=0.5"><i>Vitis vinifera</i> development</a></li>
            <li>In <a href="https://www.ebi.ac.uk/gxa/experiments/E-MTAB-2037?accessKey=&serializedFilterFactors=&queryFactorType=ORGANISM_PART&rootContext=&heatmapMatrixSize=50&displayLevels=false&displayGeneDistribution=false&geneQuery=EPlOSAG00000008047&_queryFactorValues=1&specific=true&_specific=on&cutoff=0.5"><i>Oryza sativa</i> tissues</a></li>
        </ul>

        <li>For more information please see <a href="FAQ.html#similarExpression">FAQ: How are similarly expressed genes computed?</a></li>
    </ul>

    <h4>New experiments</h4>
    <ul>
        <li><p>This release of Expression Atlas includes datasets on <strong>tomato</strong> (<i>Solanum lycopersicum</i>), a new species for us! We now have <strong>four baseline experiments</strong>, studying fruit development and profiling various tissues:</p></li>
        <ul>
            <li><a href="https://www.ebi.ac.uk/gxa/experiments/E-MTAB-4765">Transcription profiling by high throughput sequencing of the five principal tissues of the pericarp from tomato fruits at their maximal growth phase	</a></li>
            <li><a href="https://www.ebi.ac.uk/gxa/experiments/E-MTAB-4812">Transcription profiling by high throughput sequencing of tomato root, leaf, flower (2 stages) and fruit (6 stages)</a></li>
            <li><a href="https://www.ebi.ac.uk/gxa/experiments/E-MTAB-4813">Transcription profiling by high throughput sequencing along three longitudinal sections of six stages during tomato fruit development</a></li>
            <li><a href="https://www.ebi.ac.uk/gxa/experiments/E-MTAB-4818">Transcription profiling by high throughput sequencing of tomato fruit at four developmental stages</a></li>
        </ul>

        <li><p>As well as the above baseline datasets, we can also showcase <strong>seven differential expression studies</strong> on tomato:</p></li>

        <ul>
            <li><a href="https://www.ebi.ac.uk/gxa/experiments/E-MTAB-4789">Transcription profiling by high throughput sequencing of tomato Rio Grande prf3 leaves challenged with either the flgII-28 peptide or different bacterial strains</a></li>
            <li><a href="https://www.ebi.ac.uk/gxa/experiments/E-MTAB-4792">Transcription profiling by high throughput sequencing of two tomato nearly isogenic lines that differ for the allele at fw3.2 in Yellow Stuffer background</a></li>
            <li><a href="https://www.ebi.ac.uk/gxa/experiments/E-MTAB-4793">Transcription profiling by high throughput sequencing of flower buds from two tomato nearly isogenic lines that differ for the allele at fw3.2 in Yellow Stuffer background</a></li>
            <li><a href="https://www.ebi.ac.uk/gxa/experiments/E-MTAB-4796">Transcription profiling by high throughput sequencing of leaves of resistant (RG-PtoR) and susceptible (RG-prf3 and RG-prf19) tomato plants treated with Pseudomonas syringae pv. tomato str. DC3000</a></li>
            <li><a href="https://www.ebi.ac.uk/gxa/experiments/E-MTAB-4797">Transcription profiling by high throughput sequencing of leaves of resistant (RG-PtoR) tomato plants treated with different Pst DC3000 mutant strains</a></li>
            <li><a href="https://www.ebi.ac.uk/gxa/experiments/E-MTAB-4819">Transcription profiling by high throughput sequencing of immature green and red ripe tomato fruits from plants over-expressing SIGLK1 and SIGLK2</a></li>
            <li><a href="https://www.ebi.ac.uk/gxa/experiments/E-MTAB-4837">Transcription profiling by high throughput sequencing of tomato SUN and wild type fruit tissues</a></li>
        </ul>

        <li><p>The latest release also sees our first experiment studying baseline
        expression in different <a
        href="https://www.ebi.ac.uk/gxa/experiments/E-GEOD-38612">Arabidopsis
        tissues</a>, a <a
        href="https://www.ebi.ac.uk/gxa/experiments/E-MTAB-4754">new dataset on
        tonsil</a> from the <a href="http://dcc.blueprint-epigenome.eu/">BLUEPRINT
        Epigenome project</a> as well as updates to the previously-added BLUEPRINT <a
        href="https://www.ebi.ac.uk/gxa/experiments/E-MTAB-3827">dataset on
        haematopoiesis</a>, and <a
        href="https://www.ebi.ac.uk/gxa/experiments/E-MTAB-4729">RNA-seq data</a>
        from <a href="http://www.opentargets.org/">Open Targets</a> showing baseline
        expression of lung disease cell line models.</p></li>
    </ul>

    <h4>Bioconductor package out now</h4>

    <ul>
        <li><p>The <a href="http://www.bioconductor.org/packages/release/bioc/html/ExpressionAtlas.html">ExpressionAtlas R package</a> is now available as of version 3.3 of <a href="http://www.bioconductor.org/">Bioconductor</a>. Using this package you can search for Atlas datasets and download the data from inside an R session or script. Please see the <a href="http://www.bioconductor.org/packages/release/bioc/vignettes/ExpressionAtlas/inst/doc/ExpressionAtlas.html">package vignette</a> for more information.</p></li>
    </ul>

    <h3><img src="resources/images/help/atlas_release_icon.png" alt="Atlas release" align="bottom" />21 April 2016
    </h3>
    <ul>
        <li><p>Ensembl 84 / Ensembl Genomes 31 gene annotations and array design probe set mappings.</p></li>
        <li>This release contains <strong>2,738</strong> datasets (<strong>100,801</strong> assays), in particular:
        </li>
        <ul>
            <li><strong>261</strong> RNA-seq and <strong>2</strong> proteomics
            datasets, of which <strong>72</strong> report <a
            href="https://www.ebi.ac.uk/gxa/baseline/experiments">baseline</a> gene
        expression, across a total of <strong>22</strong> different
        organisms;</li>
            <li><strong>621</strong> datasets reporting expression in <a href="https://www.ebi.ac.uk/gxa/plant/experiments">plants</a>;</li>
            <li><strong>2,666</strong> datasets studying samples in <strong>7,496</strong>
                <a href="https://www.ebi.ac.uk/gxa/help/differential-atlas.html">differential</a> comparisons across <strong>29</strong> different organisms.</li>
        </ul>
    </ul>


    <h3><img src="resources/images/help/atlas_release_icon.png" alt="Atlas release" align="bottom" />4 April 2016
    </h3>
    <ul>
        <li><p>Ensembl 84 / Ensembl Genomes 30 gene annotations and array design probe set mappings.</p></li>
        <li>This release contains <strong>2,688</strong> datasets (<strong>99,162</strong> assays), in particular:
        </li>
        <ul>
            <li><strong>224</strong> RNA-seq and <strong>2</strong> proteomics
            datasets, of which <strong>60</strong> report <a
            href="https://www.ebi.ac.uk/gxa/baseline/experiments">baseline</a> gene
        expression, across a total of <strong>21</strong> different
        organisms;</li>
            <li><strong>602</strong> datasets reporting expression in <a href="https://www.ebi.ac.uk/gxa/plant/experiments">plants</a>;</li>
            <li><strong>2,626</strong> datasets studying samples in <strong>7,411</strong>
                <a href="https://www.ebi.ac.uk/gxa/help/differential-atlas.html">differential</a> comparisons across <strong>28</strong> different organisms.</li>
        </ul>
    </ul>


    <h3><img src="resources/images/help/atlas_release_icon.png" alt="Atlas release" align="bottom" /> 28 January 2016
    </h3>
    <ul>
        <li><p>Ensembl 83 / Ensembl Genomes 30 gene annotations and array design probe set mappings.</p></li>
        <li>This release contains <strong>2,620</strong> datasets (<strong>97,484</strong> assays), in particular:
        </li>
        <ul>
            <li><strong>167</strong> RNA-seq and <strong>1</strong> proteomics
            dataset, of which <strong>47</strong> report <a
            href="https://www.ebi.ac.uk/gxa/baseline/experiments">baseline</a> gene
        expression, across a total of <strong>19</strong> different
        organisms;</li>
            <li><strong>582</strong> datasets reporting expression in <a href="https://www.ebi.ac.uk/gxa/plant/experiments">plants</a>;</li>
            <li><strong>2,573</strong> datasets studying samples in <strong>7,281</strong>
                <a href="https://www.ebi.ac.uk/gxa/help/differential-atlas.html">differential</a> comparisons across <strong>29</strong> different organisms.</li>
        </ul>
        <br />

        <li><a href="https://www.r-project.org/">R</a> experiment summary objects
    for RNA-seq datasets now use the new <i>RangedSummarizedExperiment</i> class
        from the <a
        href="https://www.bioconductor.org/packages/release/bioc/html/SummarizedExperiment.html">SummarizedExperiment</a>
        <a href="https://www.bioconductor.org/">Bioconductor</a> package. See the
        <a href="https://www.ebi.ac.uk/gxa/help/r-data-objects.html">help page</a>
        for more details about how to use them.</a></li>

    </ul>


    <h3><img src="resources/images/help/atlas_release_icon.png" alt="Atlas release" align="bottom" /> 17 December 2015
    </h3>
    <ul>
        <li><p>Ensembl 83 / Ensembl Genomes 29 gene annotations and array design probe set mappings.</p></li>
        <li>This release contains <strong>2,574</strong> datasets (<strong>98,278</strong> assays), in particular:
        </li>
        <ul>
            <li><strong>157</strong> RNA-seq and <strong>1</strong> proteomics dataset, of which <strong>46</strong> report <a href="https://www.ebi.ac.uk/gxa/baseline/experiments">baseline</a> gene expression, across a total of
                <strong>19</strong> different organisms;</li>
            <li><strong>568</strong> datasets reporting expression in <a href="https://www.ebi.ac.uk/gxa/plant/experiments">plants</a>;</li>
            <li><strong>2,527</strong> datasets studying samples in <strong>7,119</strong>
                <a href="https://www.ebi.ac.uk/gxa/help/differential-atlas.html">differential</a> comparisons across <strong>26</strong> different organisms.</li>
        </ul>
        <br />
        <li>This release launches the <b>new gene and gene-set interface</b>, available for species-specific gene queries or gene-set identifier queries, via Gene Ontology (GO) or Reactome IDs, e.g.
            <ul>
             <li><a href="https://www.ebi.ac.uk/gxa/genes/ENSMUSG00000048756">Foxo3 (mouse)</a></li>
             <li><a href="https://www.ebi.ac.uk/gxa/genesets/GO:0007585">Respiratory gaseous exchange</a></li>
            </ul>
            <br />
            The new pages feature faceted search in <i>Differential expression</i> and <i>Baseline Expression</i> tabs, including</li>
        <ul>
            <li><b>Filtering differential expression</b> by Species, Experiment, Variables, up or down regulation etc.</li>
            <li>Ability to <b>show/hide baseline experiments</b> studying other conditions than just tissues, e.g. cell lines, cell types, etc.</li>
            <li><b>Anatomograms can now be shown/hidden</b> as required.</li>
        </ul>
        Baseline experiment types, except for cell lines, can be viewed on a heatmap.

        </li>
    </ul>

    <h3><img src="resources/images/help/atlas_release_icon.png" alt="Atlas release" align="bottom" /> 29 October 2015
    </h3>
    <ul>
        <li><p>Ensembl 82 / Ensembl Genomes 29 gene annotations and array design probe set mappings.</p></li>
        <li>This release contains <strong>2,373</strong> datasets (<strong>93,057</strong> assays)
        </li>
        <br />
        <li>In particular, this release contains:</li>
        <ul>
            <li><strong>119</strong> RNA-seq and <strong>1</strong> proteomics dataset, of which <strong>43</strong> report <a href="https://www.ebi.ac.uk/gxa/baseline/experiments">baseline</a> gene expression, across a total of
                <strong>17</strong> different organisms;</li>
            <li><strong>424</strong> datasets reporting expression in <a href="https://www.ebi.ac.uk/gxa/plant/experiments">plants</a>;</li>
            <li><strong>2,330</strong> datasets studying samples in <strong>6,345</strong>
                <a href="https://www.ebi.ac.uk/gxa/help/differential-atlas.html">differential</a> comparisons across <strong>25</strong> different organisms.</li>
        </ul>
        </li>
    </ul>

    <h3><img src="resources/images/help/atlas_release_icon.png" alt="Atlas release" align="bottom" /> 14 October 2015
        <a href="https://www.facebook.com/ChocolatisimoShop"><img src="resources/images/help/2000exps_web.jpg" /></a></h3>
    <ul>
        <li><p>Ensembl 82 / Ensembl Genomes 28 gene annotations and array design probe set mappings.</p></li>
        <li>This release contains <strong>2,162</strong> datasets (<strong>86,980</strong> assays), including:
            <ul>
                <li><a href="https://www.ebi.ac.uk/gxa/experiments/E-MTAB-3871">19 Fetal Tissues from NIH Roadmap Epigenomics Mapping Consortium
                    (64 assays)</a></li>

            </ul>
        </li>
        <br />
        <li>In particular, this release contains:</li>
        <ul>
            <li><strong>119</strong> RNA-seq and <strong>1</strong> proteomics dataset, of which <strong>43</strong> report <a href="https://www.ebi.ac.uk/gxa/baseline/experiments">baseline</a> gene expression, across a total of
                <strong>17</strong> different organisms;</li>
            <li><strong>410</strong> datasets reporting expression in <a href="https://www.ebi.ac.uk/gxa/plant/experiments">plants</a>;</li>
            <li><strong>2,119</strong> datasets studying samples in <strong>5,752</strong>
                <a href="https://www.ebi.ac.uk/gxa/help/differential-atlas.html">differential</a> comparisons across <strong>25</strong> different organisms.</li>
        </ul>
        </li>
    </ul>

    <h3><img src="resources/images/help/atlas_release_icon.png" alt="Atlas release" align="bottom" />14 September 2015</h3>
    <ul>
        <li><p>Ensembl 81 / Ensembl Genomes 28 gene annotations and array design probe set mappings.</p></li>
        <li>This release contains <strong>1,895</strong> datasets (<strong>81,255</strong> assays), including:
            <ul>
                <li><a href="https://www.ebi.ac.uk/gxa/experiments/E-PROT-1">Re-analysis of Pandey lab's draft of the human proteome (30 assays)</a> - <b>out first protemics dataset!</b></li>
                <li><a href="https://www.ebi.ac.uk/gxa/experiments/E-MTAB-3838">RNA-seq of coding RNA from sheep embryonic, juvenile and adult tissues (235 assays)</a></li>
                <li><a href="https://www.ebi.ac.uk/gxa/experiments/E-MTAB-3839">RNA-seq of various Sorghum bicolor (BTx623) tissues: flowers, vegetative and floral meristems, embryos, roots and shoots (17 assays)</a></li>

            </ul>
        </li>
        <br />
        <li>In particular, this release contains:</li>
        <ul>
            <li><strong>117</strong> RNA-seq datasets, of which <strong>42</strong> of these report <a href="https://www.ebi.ac.uk/gxa/baseline/experiments">baseline</a> gene expression, across a total of
                <strong>15</strong> different organisms;</li>
            <li><strong>397</strong> datasets reporting expression in <a href="https://www.ebi.ac.uk/gxa/plant/experiments">plants</a>;</li>
            <li><strong>1,853</strong> datasets studying samples in <strong>4,925</strong>
                <a href="https://www.ebi.ac.uk/gxa/help/differential-atlas.html">differential</a> comparisons across <strong>25</strong> different organisms.</li>
        </ul>
        </li>
    </ul>

    <h3><img src="resources/images/help/atlas_release_icon.png" alt="Atlas release" align="bottom" />11 August 2015</h3>
    <ul>
        <li><p>Ensembl 81 / Ensembl Genomes 28 gene annotations and array design probe set mappings.</p></li>
        <li>This release contains <strong>1,572</strong> datasets (<strong>69,239</strong> assays), including:
            <ul>
                <li><a href="https://www.ebi.ac.uk/gxa/experiments/E-MTAB-2770">RNA-seq of 934 human cancer cell lines from the Cancer Cell Line Encyclopedia (30,088 assays)</a></li>
            </ul>
        </li>
        <br />
        <li>In particular, this release contains:</li>
        <ul>
            <li><strong>96</strong> RNA-seq, of which <strong>38</strong> of these report <a href="https://www.ebi.ac.uk/gxa/baseline/experiments">baseline</a> gene expression, across a total of
                <strong>15</strong> different organisms;</li>
            <li><strong>389</strong> reporting expression in <a href="https://www.ebi.ac.uk/gxa/plant/experiments">plants</a>;</li>
            <li><strong>1,534</strong> studying samples in <strong>4,287</strong>
                <a href="https://www.ebi.ac.uk/gxa/help/differential-atlas.html">differential</a> comparisons across <strong>25</strong> different organisms.</li>
            <li>A much broader coverage of tissues in mouse anatomograms, c.f. <a href="https://www.ebi.ac.uk/gxa/genes/ENSMUSG00000022097">mouse SFTPA gene expression</a></li>
        </ul>
        </li>
    </ul>


    <h3><img src="resources/images/help/atlas_release_icon.png" alt="Atlas release" align="bottom" />2 July 2015</h3>

    <ul>
        <li><p>Ensembl 80 / Ensembl Genomes 27 gene annotations and array design probe set mappings.</p></li>
        <li>This release contains <strong>1,107</strong> datasets (<strong>24,048</strong> assays), including:
            <ul>
                <li><a href="https://www.ebi.ac.uk/gxa/experiments/E-MTAB-3579">RNA-Seq CAGE (Cap Analysis of Gene Expression) analysis of mice tissues in RIKEN FANTOM5 project</a></li>
            </ul>
        </li>
        <br />
        <li>In particular, this release contains:</li>
        <ul>
            <li><strong>68</strong> RNA-seq, of which <strong>30</strong> of these report <a href="https://www.ebi.ac.uk/gxa/baseline/experiments">baseline</a> gene expression, across a total of
                <strong>16</strong> different organisms;</li>
            <li><strong>385</strong> reporting expression in <a href="https://www.ebi.ac.uk/gxa/plant/experiments">plants</a>;</li>
            <li><strong>1,082</strong> studying samples in <strong>2,923</strong>
                <a href="https://www.ebi.ac.uk/gxa/help/differential-atlas.html">differential</a> comparisons across <strong>24</strong> different organisms.</li>
            <li>A much broader coverage of tissues in human anatomograms, c.f. <a href="https://www.ebi.ac.uk/gxa/genes/ENSG00000185303">human SFTPA2 gene expression</a></li>
        </ul>
        </li>
    </ul>

    <h3><img src="resources/images/help/atlas_release_icon.png" alt="Atlas release" align="bottom" />5 June 2015
        <a href="http://www.chocolatisimo.es"><img src="resources/images/help/1000exps_web.jpeg" /></a></h3>

    <ul>
        <li><p>Ensembl 80 / Ensembl Genomes 26 gene annotations and array design probe set mappings.</p></li>
        <li><p>We are proud to announce that release contains <strong>1,051</strong> datasets <i>(yes, we did celebrate!)</i>, now including:</p>
            <ul>
                <li><a href="https://www.ebi.ac.uk/gxa/experiments/E-MTAB-2919">RNA-seq from 53 human tissue samples from the Genotype-Tissue Expression (GTEx) Project</a></li>
                <li><a href="https://www.ebi.ac.uk/gxa/experiments/E-MTAB-2706">RNA-seq of 675 commonly used human cancer cell lines (Genentech)</a></li>
                <li><a href="https://www.ebi.ac.uk/gxa/experiments/E-MTAB-3578">RNA-Seq CAGE (Cap Analysis of Gene Expression) analysis of mice cells in RIKEN FANTOM5 project</a></li>
            </ul>
        </li>
        <br />
        <li>In particular, this release contains:</li>
            <ul>
                <li><strong>67</strong> RNA-seq, of which <strong>29</strong> of these report <a href="https://www.ebi.ac.uk/gxa/baseline/experiments">baseline</a> gene expression, across a total of
                    <strong>16</strong> different organisms;</li>
                <li><strong>366</strong> reporting expression in <a href="https://www.ebi.ac.uk/gxa/plant/experiments">plants</a>;</li>
                <li><strong>1,022</strong> studying samples in <strong>2,704</strong>
                    <a href="https://www.ebi.ac.uk/gxa/help/differential-atlas.html">differential</a> comparisons across <strong>24</strong> different organisms.</p></li>
            </ul>
        </li>
    </ul>

    <h3><img src="resources/images/help/atlas_release_icon.png" alt="Atlas release" align="bottom" />31st March 2015</h3>
    <ul>
        <li><p>Ensembl 79 / Ensembl Genomes 26 gene annotations and array design probe set mappings.</p></li>
        <li>This release contains <strong>935</strong> datasets, including:
            <ul>
                <li><strong>56</strong> RNA-seq, of which <strong>26</strong> of these report <a href="https://www.ebi.ac.uk/gxa/baseline/experiments">baseline</a> gene expression, across a total of
                    <strong>16</strong> different organisms;</li>
                <li><strong>261</strong> reporting expression in <a href="https://www.ebi.ac.uk/gxa/plant/experiments">plants</a>;</li>
                <li><strong>909</strong> studying samples in <strong>2,306</strong>
                    <a href="https://www.ebi.ac.uk/gxa/help/differential-atlas.html">differential</a> comparisons across <strong>20</strong> different organisms.</li>
            </ul>
            </li>
    </ul>

    <h3><img src="resources/images/help/atlas_release_icon.png" alt="Atlas release" align="bottom" />20th March 2015</h3>
    <ul>
        <li><p>Ensembl 79 / Ensembl Genomes 25 gene annotations and array design probe set mappings.</p></li>
        <li>This release contains <strong>932</strong> datasets, including:
            <ul>
            <li><strong>56</strong> RNA-seq, of which <strong>26</strong> of these report <a href="https://www.ebi.ac.uk/gxa/baseline/experiments">baseline</a> gene expression, across a total of
            <strong>16</strong> different organisms;</li>
            <li><strong>258</strong> reporting expression in <a href="https://www.ebi.ac.uk/gxa/plant/experiments">plants</a>;</li>
            <li><strong>906</strong> studying samples in <strong>2,291</strong>
            <a href="https://www.ebi.ac.uk/gxa/help/differential-atlas.html">differential</a> comparisons across <strong>20</strong> different organisms.</li>
            </ul>
        </li>
        <li>New data sets of note:
        <ul>
            <li><a href="https://www.ebi.ac.uk/gxa/experiments/E-MTAB-2980">RNA-seq of 39 human cancer cell lines that are in the NCI-60 set from the Cancer Cell Line Encyclopedia</a></li>
            <li><a href="https://www.ebi.ac.uk/gxa/experiments/E-MTAB-3358">RNA-Seq CAGE (Cap Analysis of Gene Expression) analysis of human tissues in RIKEN FANTOM5 project</a></li>
        </ul><p/></li>
        <li><p>Differential expression for RNA-seq data is now calculated using <a href="http://www.bioconductor.org/packages/release/bioc/html/DESeq2.html"><strong>DESeq2</strong></a> method.</p></li>
        <li><p>Users searching for a single gene within a baseline experiment are now able to see how expression varies across biological replicates for each condition - by selecting <strong>Display variation</strong> option. For an example click <a href="https://www.ebi.ac.uk/gxa/experiments/E-MTAB-2836?geneQuery=ENSG00000197249">here</a>.
            This is currently a prototype - we welcome any feedback you may have. Please use the feedback button on the top right hand side of all Atlas pages.
            <img src="resources/images/help/baselineBoxplots.png" alt="Example of expression variation display" align="bottom" /></p></li>
    </ul>

    <h3><img src="resources/images/help/atlas_release_icon.png" alt="Atlas release" align="bottom" />27th January 2015</h3>
    <ul>
        <li><p>Ensembl 78 / Ensembl Genomes 25 gene annotations and array design probe set mappings</p></li>
        <li><p>This release contains <strong>909</strong> datasets, including <strong>57</strong> RNA-seq experiments. <strong>24</strong> of these report <a href="https://www.ebi.ac.uk/gxa/baseline/experiments">baseline</a> gene expression, across a total of <strong>16</strong> different organisms. <strong>885</strong> datasets study samples in <strong>2,226</strong> <a href="https://www.ebi.ac.uk/gxa/help/differential-atlas.html">differential</a> comparisons across <strong>20</strong> different organisms.</p></li>
        <li><a href="https://www.ebi.ac.uk/gxa/experiments/E-MTAB-2836">RNA-seq of coding RNA from tissue samples of 122 human individuals representing 32 different tissues</a> data set is now available;</li>

    </ul>

    <h3><img src="resources/images/help/atlas_release_icon.png" alt="Atlas release" align="bottom" />18th December 2014</h3>
    <ul>
        <li><p>Ensembl 78 / Ensembl Genomes 24 gene annotations and array design probe set mappings</p></li>
        <li><p>This release contains <strong>873</strong> datasets, including <strong>50</strong> RNA-seq experiments. <strong>23</strong> of these report <a href="https://www.ebi.ac.uk/gxa/baseline/experiments">baseline</a> gene expression, across a total of <strong>16</strong> different organisms. <strong>850</strong> datasets study samples in <strong>2,121</strong> <a href="https://www.ebi.ac.uk/gxa/help/differential-atlas.html">differential</a> comparisons across <strong>20</strong> different organisms.</p></li>
        <li><p>Please note that splice variant expression quantification is no longer available in Atlas. We have also replaced <a href="http://cole-trapnell-lab.github.io/cufflinks/">Cufflinks2</a> with <a href="http://www-huber.embl.de/users/anders/HTSeq/doc/overview.html">HTSeq</a> for gene expression quantification.</p></li>
        <li><p>By clicking on the link highlighted on the image below, the experiment data can now by filtered by gene/protein keywords that were mined from the supporting publication by <a href="http://europepmc.org/">Europe PubMed Central</a>.
    Click <a href="https://www.ebi.ac.uk/gxa/experiments/E-GEOD-49644">here</a> to try it out.
            <img src="resources/images/filter_by_pub_genes.png" alt="Filter experiment by publication genes link" align="top" />
        </p></li>
    </ul>

    <h3><img src="resources/images/help/atlas_release_icon.png" alt="Atlas release" align="bottom" />19th November 2014</h3>
    <ul>
        <li><p>Ensembl 77 / Ensembl Genomes 24 gene annotations and array design probe set mappings</p></li>
        <li><p>This release contains <strong>844</strong> datasets, including <strong>50</strong> RNA-seq experiments. <strong>22</strong> of these report <a href="https://www.ebi.ac.uk/gxa/baseline/experiments">baseline</a> gene expression, across a total of <strong>16</strong> different organisms. <strong>822</strong> datasets study samples in <strong>2,032</strong> <a href="https://www.ebi.ac.uk/gxa/help/differential-atlas.html">differential</a> comparisons across <strong>20</strong> different organisms.</p></li>
        <li><p>Plant experiments, for example <a href="https://www.ebi.ac.uk/gxa/experiments/E-MTAB-3028">our first RNA-seq maize data set</a>, now allow users to display expression in the <img src="resources/images/help/gramene_full.png" alt="Gramene"/>&nbsp;&nbsp;<a href="http://ensembl.gramene.org/genome_browser/index.html">genome browser</a>. </p></li>
    </ul>

    <h3><img src="resources/images/help/atlas_release_icon.png" alt="Atlas release" align="bottom" />13th October 2014</h3>

    <ul>
        <li><p><a href="http://www.ensembl.org/index.html">Ensembl</a> <strong>77</strong> gene annotations and array design probe set mappings.</p></li>
        <li><p>This release contains <strong>802</strong> datasets, including <strong>47</strong> RNA-seq experiments. <strong>21</strong> of these report <a href="https://www.ebi.ac.uk/gxa/baseline/experiments">baseline</a> gene expression, across a total of <strong>15</strong> different organisms. <strong>781</strong> datasets study samples in <strong>1,937</strong> <a href="https://www.ebi.ac.uk/gxa/help/differential-atlas.html">differential</a> comparisons across <strong>20</strong> different organisms.</p></li>
        <li><p><a href="https://www.ebi.ac.uk/gxa/genes/ENSG00000163631">Gene pages</a> now show a combined <strong>comparative view</strong> for gene expression across multiple <a href="https://www.ebi.ac.uk/gxa/help/baseline-atlas.html">baseline</a> datasets.</p>
        <p><img src="resources/images/help/comparative_view.png" alt="Comparative baseline gene expression view" /></p></li>
        <li><p><img src="resources/images/help/cluster_button_4help.png" alt="hierarchical clustering button" align="bottom" /> <strong>Hierarchical clustering</strong> of the top 100 most variably expressed genes is now performed for each baseline dataset.</p>
        <p><img src="resources/images/help/E-MTAB-2809-heatmap.png" alt="E-MTAB-2809 heatmap" /></p></li>
        <li><p>Three new anatomograms are now available, for <i><a href="https://www.ebi.ac.uk/gxa/experiments/E-GEOD-41338?serializedFilterFactors=ORGANISM:Anolis%20carolinensis">Anolis carolinensis</a></i>, <i><a href="https://www.ebi.ac.uk/gxa/experiments/E-GEOD-41338?serializedFilterFactors=ORGANISM:Anolis%20carolinensis">Xenopus tropicalis</a></i> and <i><a href="https://www.ebi.ac.uk/gxa/experiments/E-GEOD-41338?serializedFilterFactors=ORGANISM:Tetraodon%20nigroviridis">Tetraodon nigroviridis</a></i>. <img src="resources/images/help/three_new_anatomograms_oct2014.png" alt="Image of three new anatomograms" align="middle" /></p>
    </ul>



    <h3><img src="resources/images/help/atlas_release_icon.png" alt="Atlas release" align="bottom" />22nd August 2014</h3>
    <ul>
        <li>Ensembl 76 / Ensembl Genomes 23 gene annotations and array design probe set mappings;</li>
        <li>709 experiments, including 39 RNA-seq experiments (13 <a href="https://www.ebi.ac.uk/gxa/baseline/experiments">baseline</a>, 26 differential), studying samples in 1,684 differential comparisons and 24 organisms (see below for breakdown of comparisons per organism);</li>
        <li>All Human RNA-seq experiments have been re-analysed against the new assembly (GRCh38), e.g. <a href="https://www.ebi.ac.uk/gxa/experiments/E-MTAB-1733">RNA-seq of coding RNA from tissue samples of 95 human individuals representing 27 different tissues</a>;</li>
        <li>Baseline expression levels available for genes and transcripts in 75 tissues, 21 cell lines, 7 cellular components, 39 developmental stages and 3 RNA types;</li>
        <li><img src="resources/images/qc_v15.png" alt="Quality control report button" align="bottom" /> Quality control reports, e.g. <a href="https://www.ebi.ac.uk/gxa/experiments/E-MTAB-1733/fastqc/homo%20sapiens/qc.html">FASTQ QC</a> and <a href="https://www.ebi.ac.uk/gxa/experiments/E-MTAB-1733/fastqc/homo%20sapiens/mapping/tophat2.html">Mapping</a>, are now available for all RNA-seq experiments;</li>
        <li><a href="http://www.ebi.ac.uk/efo/">EFO</a>-based autocomplete for experimental conditions is now available on the main Atlas search page. Please see 'How to search' section of <a href="https://www.ebi.ac.uk/gxa/help/index.html">Atlas help page</a> for more details. <a href="https://www.ebi.ac.uk/gxa/">The main search interface</a> also now allows users to restrict search to a single organism;</li>
        <li>User queries in the main Atlas search interface are expanded using the latest <a href="http://www.ebi.ac.uk/efo/">EFO</a>. This means that if you search for e.g. "cancer", you will also see results with terms such as "carcinoma", "malignant tumour", and other synonyms and children of the <a href="http://www.ebi.ac.uk/efo/EFO_0000311">cancer</a> entry in EFO; </li>
        <li>Gene set overlap analysis is performed using the Piano package from Bioconductor. In each differential experiment, for each comparison, enrichment of terms from GO, InterPro, and Reactome is tested for within the set of differentially expressed genes, using Fisher's exact test. Plots showing maximum top ten GO, Interpro and Reactome terms enriched within the differentially expressed genes for each comparison are now available for each differential experiment in Atlas. Please see <a href="https://www.ebi.ac.uk/gxa/help/differential-atlas.html#plots">Differential Atlas help</a> for more information. </li>
        <li><img src="resources/images/r-button.png" alt="R object download button" align="bottom" /> For <a href="help/differential-atlas.html">differential expression</a> experiments in Expression Atlas, you can now download a file containing the expression data and meta-data that you can load into <a href="http://www.r-project.org/">R</a>. Please see <a href="https://www.ebi.ac.uk/gxa/help/r-data-objects.html">Expression Atlas data in R help</a> for more information;</li>
        <li>Differential and Baseline Expression can now be visualised on Ensembl and Ensembl Genomes Genome Browser. Please see <a href="https://www.ebi.ac.uk/gxa/help/baseline-atlas.html#ensembl">Visualising baseline expression at Ensembl</a> and <a href="https://www.ebi.ac.uk/gxa/help/differential-atlas.html#ensembl">Visualising differential expression at Ensembl</a> for more information.</li>
    </ul>
    <img src="resources/images/help/differential_species_aug2014.png" alt="Number of differential comparisons per organism" align="middle" />

    <h3><img src="resources/images/help/atlas_release_icon.png" alt="Atlas release" align="bottom" />10th April 2014</h3>
    <ul>
        <li>Ensembl 75 / Ensembl Genomes 22 gene annotations and array design probe set mappings;</li>
        <li>401 experiments, including 24 RNA-seq experiments (8 baseline, 16 differential), studying samples in 19 species and 653 differential comparisons (see below for breakdown of comparisons per experimental variable type);</li>
        <li>Average number of citations per experiment: 42 (courtesy of <a href="http://europepmc.org/">Europe PubMed Central)</a>;</li>
        <li>Baseline expression levels available for genes and transcripts in 54 tissues, 21 cell lines, 5 cellular components, 5 developmental stages and 3 RNA types;</li>
        <li><a href="https://www.ebi.ac.uk/gxa/experiments/E-MTAB-1733">RNA-seq of coding RNA from tissue samples of 95 human individuals representing 27 different tissues</a> data set is now available;</li>
        <li>Automatic quality control (QC) for Microarray experiments in operation; QC reports available to users, e.g. <a href="https://www.ebi.ac.uk/gxa/experiments/E-GEOD-10233/qc/A-AFFY-44/index.html">Transcription profiling by array of human PMA-activated MM6 cells with cyclin T1 knockdown</a></li>
    </ul>
    <img src="resources/images/help/differential_factors_apr2014.png" alt="Number of differential comparisons per experimental variable type" align="middle" />

    <h3><img src="resources/images/help/atlas_release_icon.png" alt="Atlas release" align="bottom" />5th December 2013</h3>
    <p>Welcome to the new Expression Atlas service! <a href="https://www.ebi.ac.uk/gxa" title="https://www.ebi.ac.uk/gxa">The old Expression Atlas</a> and its underlying data will no longer be updated or maintained, but limited support will be offered while the process of re-curation, re-analysis and transferring of experiments to the new Expression Atlas is on-going.</p>
    <p>
    We would like to thank our users for their continuous support, and encourage them to help us shape the new Expression Atlas by:
        <ul>
            <li>providing us with feedback via <a href="mailto:atlas-feedback@ebi.ac.uk" rel="nofollow" title="mailto:atlas-feedback@ebi.ac.uk">mailto:atlas-feedback@ebi.ac.uk</a> or via the <img src="resources/images/help/feedback_button.png" alt="Atlas feedback" align="bottom"> button on the top right hand side of each page; </li>
            <li>contacting us if any experiments of interest are included in the old Expression Atlas, but have not made their way into the new Expression Atlas yet. We will then prioritise their transfer.</li>
        </ul></p>
        <p>Enjoy the new Expression Atlas!</p>
    </div>
</div>